{
 "cells": [
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "import pandas as pd\n",
    "import matplotlib.pyplot as plt\n",
    "import numpy as np\n",
    "\n",
    "%matplotlib inline\n",
    "\n",
    "\n",
    "plt.rcParams['font.size'] = 7\n",
    "plt.rcParams['font.family'] = \"sans-serif\"\n",
    "plt.rcParams['font.sans-serif'] = 'Arial'\n",
    "plt.rcParams['mathtext.fontset'] = 'custom'\n",
    "plt.rcParams['mathtext.cal'] = 'Arial' "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "cocl = pd.read_csv('../data_tables/Fitness_files/pool4_fitness_fc.tsv', sep ='\\t')\n",
    "caff = pd.read_csv('../data_tables/Fitness_files/pool7_fitness_fc.tsv',sep = '\\t')\n",
    "fluc = pd.read_csv('../data_tables/Fitness_files/pool9_fitness_fc.tsv',sep='\\t')\n",
    "\n",
    "annos = pd.read_csv('../data_tables/QTL_pool_annotations.tsv',sep='\\t')\n",
    "anno_cocl = annos.merge(cocl, 'inner', 'var_id')\n",
    "anno_caff = annos.merge(caff,'inner','var_id')\n",
    "anno_fluc = annos.merge(fluc,'inner','var_id')\n",
    "\n",
    "\n",
    "anno_caff['abscoef'] = abs(anno_caff['coef'])\n",
    "anno_cocl['abscoef'] = abs(anno_cocl['coef'])\n",
    "anno_fluc['abscoef'] = abs(anno_fluc['coef'])"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "plt.rcParams['font.family'] = 'Arial'\n",
    "plt.rcParams['font.size'] = 7"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "\n",
    "SIZE = 7\n",
    "\n",
    "plt.rc('axes', titlesize=SIZE)     # fontsize of the axes title\n",
    "plt.rc('axes', labelsize=SIZE)    # fontsize of the x and y labels\n",
    "plt.rc('xtick', labelsize=SIZE)    # fontsize of the tick labels\n",
    "plt.rc('ytick', labelsize=SIZE)    # fontsize of the tick labels\n",
    "plt.rc('legend', fontsize=SIZE)    # legend fontsize\n",
    "plt.rc('figure', titlesize=SIZE)  # fontsize of the figure title"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "fig,axes = plt.subplots(2,3, figsize = (6.4,3.5))\n",
    "fig.tight_layout()\n",
    "ax = axes[0,2]\n",
    "ax2 = axes[1,2]\n",
    "ax3 = axes[0,1]\n",
    "ax4 = axes[1,1]\n",
    "ax5 = axes[0,0]\n",
    "ax6 = axes[1,0]\n",
    "\n",
    "\n",
    "tor1 = anno_caff[(anno_caff['Gene']=='TOR1')]\n",
    "tor1[(tor1['condition']=='CAFF')].plot.scatter('SNP_chr_pos','coef', s = 3, ax= ax, color = 'lightblue')\n",
    "tor1[(tor1['condition']=='P7SC')].plot.scatter('SNP_chr_pos','coef', s = 3, ax= ax, color = 'gold')\n",
    "tor1[(tor1['condition']=='CAFF')&(tor1['padj']<.01)].plot.scatter('SNP_chr_pos','coef', s = 3, ax= ax, color = 'blue')\n",
    "try:\n",
    "    tor1[(tor1['condition']=='P7SC')&(tor1['padj']<.01)].plot.scatter('SNP_chr_pos','coef', s = 3, ax= ax, color = 'orangered')\n",
    "except:\n",
    "    pass\n",
    "# set new limits\n",
    "\n",
    "low, high = ax.get_ylim()\n",
    "bound = max(abs(low), abs(high))*1.25\n",
    "ax.set_ylim(-bound, bound)\n",
    "ysize = bound*2\n",
    "xsize = ax.get_xlim()[1]-ax.get_xlim()[0]\n",
    "ax.arrow(559416,-bound+.03*ysize,7413 ,0,  width = ysize/12,head_width = ysize/12, head_length = xsize/20,length_includes_head=True)  \n",
    "ax.annotate('TOR1', xy = (559416+7413/2,-bound+.01*ysize), color = 'white', fontstyle = 'oblique', weight = 'bold', ha = 'center')  \n",
    "ax.xaxis.set_label_text('')\n",
    "ax.yaxis.set_label_text('')\n",
    "ax.locator_params(tight = True,axis=\"x\", nbins=3)\n",
    "ax.locator_params(tight = True,axis=\"y\", nbins=5)\n",
    "ax.set_title('Chr11 TOR1 CAFF QTL')\n",
    "\n",
    "\n",
    "tor2 = anno_caff[(anno_caff['Gene']=='TOR2')]\n",
    "tor2[tor2['condition']=='CAFF'].plot.scatter('SNP_chr_pos','coef', s = 3, ax= ax2, color = 'lightblue')\n",
    "tor2[(tor2['condition']=='P7SC')].plot.scatter('SNP_chr_pos','coef', s = 3, ax= ax2, color = 'gold')\n",
    "tor2[(tor2['padj']<.01)&(tor2['condition']=='CAFF')].plot.scatter('SNP_chr_pos','coef', s = 3, ax= ax2, color = 'blue')\n",
    "\n",
    "try:\n",
    "    tor2[(tor2['adjP']<.01)&(tor2['condition']=='P7SC')].plot.scatter('SNP_chr_pos','coef', s = 3, ax= ax2, color = 'orangered')\n",
    "except:\n",
    "    pass\n",
    "# set new limits\n",
    "\n",
    "low, high = ax2.get_ylim()\n",
    "bound = max(abs(low), abs(high))*1.25\n",
    "ax2.set_ylim(-bound, bound)\n",
    "ysize = bound*2\n",
    "xsize = ax2.get_xlim()[1]-ax2.get_xlim()[0]\n",
    "ax2.arrow(63359,-bound+.03*ysize,-7425 ,0,  width = ysize/12,head_width = ysize/12, head_length = xsize/20,length_includes_head=True)  \n",
    "ax2.annotate('TOR2', xy = (63359-7425/2,-bound+.01*ysize), color = 'white', fontstyle = 'oblique', weight = 'bold', ha = 'center')   \n",
    "ax2.xaxis.set_label_text('')\n",
    "ax2.yaxis.set_label_text('')\n",
    "ax2.locator_params(tight = True,axis=\"x\", nbins=3)\n",
    "ax2.locator_params(tight = True,axis=\"y\", nbins=5)\n",
    "ax2.set_title('Chr11 TOR2 CAFF QTL')\n",
    "\n",
    "\n",
    "\n",
    "\n",
    "mam3  = anno_cocl[(anno_cocl['Gene']=='MAM3')]\n",
    "mam3[mam3['condition']=='COCL'].plot.scatter('SNP_chr_pos','coef', s = 3, ax= ax3, color = 'lightblue')\n",
    "mam3[mam3['condition']=='P4SC'].plot.scatter('SNP_chr_pos','coef', s = 3, ax= ax3, color = 'gold')\n",
    "mam3[(mam3['condition']=='COCL')&(mam3['padj']<.01)].plot.scatter('SNP_chr_pos','coef', s = 3, ax= ax3, color = 'blue')\n",
    "try:\n",
    "    mam3[(mam3['condition']=='P4SC')&(mam3['padj']<.01)].plot.scatter('SNP_chr_pos','coef', s = 3, ax= ax3, color = 'orangered')\n",
    "except:\n",
    "    pass\n",
    "\n",
    "# set new limits\n",
    "\n",
    "low, high = ax3.get_ylim()\n",
    "bound = max(abs(low), abs(high))\n",
    "ax3.set_ylim(-bound, bound)    \n",
    "ax3.xaxis.set_label_text('')\n",
    "ax3.yaxis.set_label_text('')\n",
    "ysize = bound*2    \n",
    "xsize = ax3.get_xlim()[1]-ax3.get_xlim()[0]\n",
    "ax3.arrow(216137,-bound+.03*ysize,-2121 ,0,  width = ysize/12,head_width = ysize/12, head_length = xsize/20,length_includes_head=True)  \n",
    "ax3.annotate('MAM3', xy = (216137-2121/2,-bound+.01*ysize), color = 'white', fontstyle = 'oblique', weight = 'bold', ha = 'center')   \n",
    "ax3.locator_params(tight = True,axis=\"x\", nbins=3)\n",
    "ax3.locator_params(tight = True,axis=\"y\", nbins=5)\n",
    "ax3.set_title(r'Chr15 MAM3 CoCl$_2$ QTL')\n",
    "\n",
    "\n",
    "cot1 = anno_cocl[(anno_cocl['Gene']=='COT1')]\n",
    "cot1[cot1['condition']=='COCL'].plot.scatter('SNP_chr_pos','coef', s = 3, ax= ax4, color = 'lightblue')\n",
    "cot1[cot1['condition']=='P4SC'].plot.scatter('SNP_chr_pos','coef', s = 3, ax= ax4, color = 'gold')\n",
    "cot1[(cot1['condition']=='COCL')&(cot1['padj']<.01)].plot.scatter('SNP_chr_pos','coef', s = 3, ax= ax4, color = 'blue')\n",
    "try:\n",
    "    cot1[(cot1['condition']=='P4SC')&(cot1['padj']<.01)].plot.scatter('SNP_chr_pos','coef', s = 3, ax= ax4, color = 'orangered')\n",
    "except:\n",
    "    pass\n",
    "\n",
    "low, high = ax4.get_ylim()\n",
    "bound = max(abs(low), abs(high))*1.25\n",
    "ax4.set_ylim(-bound, bound)    \n",
    "ysize = bound*2  \n",
    "xsize = ax4.get_xlim()[1]-ax4.get_xlim()[0]\n",
    "ax4.arrow(907555,-bound+.03*ysize,-1320,0,  width = ysize/12,head_width = ysize/12, head_length = xsize/20,length_includes_head=True)  \n",
    "ax4.annotate('COT1', xy = (907555-1320/2,-bound+.01*ysize), color = 'white', fontstyle = 'oblique', weight = 'bold', ha = 'center')   \n",
    "ax4.xaxis.set_label_text('')\n",
    "ax4.yaxis.set_label_text('')\n",
    "ax4.locator_params(tight = True,axis=\"x\", nbins=3)\n",
    "ax4.locator_params(tight = True,axis=\"y\", nbins=5)\n",
    "ax4.set_title(r'Chr15 COT1 CoCl$_2$ QTL')\n",
    "#every_nth = 2\n",
    "#for n, label in enumerate(ax4.xaxis.get_ticklabels()):\n",
    "#    if n % every_nth == 0:\n",
    "#        label.set_visible(False)\n",
    "\n",
    "\n",
    "pdr3 = anno_fluc[(anno_fluc['chrom']=='II')&(anno_fluc['Gene']=='PDR3')]\n",
    "pdr3[pdr3['condition']=='FLUC'].plot.scatter('SNP_chr_pos','coef', s = 3, ax= ax5, color = 'lightblue')\n",
    "pdr3[pdr3['condition']=='P9SC'].plot.scatter('SNP_chr_pos','coef', s = 3, ax= ax5, color = 'gold')\n",
    "pdr3[(pdr3['condition']=='FLUC')&(pdr3['padj']<.01)].plot.scatter('SNP_chr_pos','coef', s = 3, ax= ax5, color = 'blue')\n",
    "try:\n",
    "    pdr3[(pdr3['condition']=='P9SC')&(pdr3['padj']<.01)].plot.scatter('SNP_chr_pos','coef', s = 3, ax= ax5, color = 'orangered')\n",
    "except: \n",
    "    pass\n",
    "\n",
    "low, high = ax5.get_ylim()\n",
    "bound = max(abs(low), abs(high))*1.25\n",
    "ax5.set_ylim(-bound, bound)    \n",
    "ysize = bound*2    \n",
    "xsize = ax5.get_xlim()[1]-ax5.get_xlim()[0]\n",
    "ax5.arrow(217470,-bound+.03*ysize,2931,0,  width = ysize/12,head_width = ysize/12, head_length = xsize/20,length_includes_head=True)  \n",
    "ax5.annotate('PDR3', xy = (217470+2931/2,-bound+.01*ysize), color = 'white', fontstyle = 'oblique', weight = 'bold', ha = 'center')\n",
    "ax5.xaxis.set_label_text('')\n",
    "ax5.yaxis.set_label_text('')\n",
    "ax5.locator_params(tight = True,axis=\"x\", nbins=3)\n",
    "ax5.locator_params(tight = True,axis=\"y\", nbins=5)\n",
    "ax5.set_title('Chr2 PDR3 FLC QTL')\n",
    "\n",
    "\n",
    "sfl1 = anno_fluc[(anno_fluc['Gene'] =='SFL1')]\n",
    "sfl1[sfl1['condition']=='FLUC'].plot.scatter('SNP_chr_pos','coef', s = 3, ax= ax6, color = 'lightblue')\n",
    "sfl1[sfl1['condition']=='P9SC'].plot.scatter('SNP_chr_pos','coef', s = 3, ax= ax6, color = 'gold')\n",
    "sfl1[(sfl1['condition']=='FLUC')&(sfl1['padj']<.01)].plot.scatter('SNP_chr_pos','coef', s = 3, ax= ax6, color = 'blue')\n",
    "try:\n",
    "    sfl1[(sfl1['condition']=='P9SC')&(sfl1['padj']<.01)].plot.scatter('SNP_chr_pos','coef', s = 3, ax= ax6, color = 'orangered')\n",
    "except:\n",
    "    pass\n",
    "\n",
    "xsize = ax6.get_xlim()[1]-ax6.get_xlim()[0]\n",
    "\n",
    "ax6.xaxis.set_label_text('')\n",
    "ax6.yaxis.set_label_text('')\n",
    "\n",
    "low, high = ax6.get_ylim()\n",
    "bound = max(abs(low), abs(high))\n",
    "ax6.set_ylim(-bound, bound)  \n",
    "ysize = bound*2\n",
    "ax6.arrow(586981,-bound+.03*ysize,2301,0, width = ysize/12,head_width = ysize/12, head_length = xsize/20,length_includes_head=True)  \n",
    "ax6.annotate('SFL1', xy = (586981+2301/2,-bound+.01*ysize), color = 'white', fontstyle = 'oblique', weight = 'bold', ha = 'center')\n",
    "ax6.locator_params(tight = True,axis=\"x\", nbins=3)\n",
    "ax6.locator_params(tight = True,axis=\"y\", nbins=5)\n",
    "ax6.set_title('Chr15 SFL1 FLC QTL')\n",
    "\n",
    "\n",
    "\n",
    "plt.subplots_adjust(hspace = .5)\n",
    "fig.add_subplot(111, frameon=False)\n",
    "# hide tick and tick label of the big axis\n",
    "\n",
    "plt.xlabel('Position in Chromosome', labelpad = 25)\n",
    "plt.ylabel(\"Variant Fitness\", labelpad = 25)\n",
    "lgnd = ax3.legend(['Stress FDR\\u2265.01', 'SC FDR\\u2265.01','Stress FDR<.01', 'SC Media FDR<.01'],loc='lower left', bbox_to_anchor = (2.22,-.51))\n",
    "lgnd.legendHandles[0]._sizes = [20]\n",
    "lgnd.legendHandles[1]._sizes = [20]\n",
    "lgnd.legendHandles[2]._sizes = [20]\n",
    "lgnd.legendHandles[3]._sizes = [20]\n",
    "\n",
    "\n",
    "plt.tick_params(labelcolor='none', which='both', top=False, bottom=False, left=False, right=False,labelbottom=False,labelleft=False)\n",
    "plt.savefig('../GxE_Figures/Figure_2/figure2e_exampleQTLs_FC.svg')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "CRISPEY_GxE",
   "language": "python",
   "name": "crispey_gxe"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.9.7"
  },
  "toc": {
   "base_numbering": 1,
   "nav_menu": {},
   "number_sections": true,
   "sideBar": true,
   "skip_h1_title": false,
   "title_cell": "Table of Contents",
   "title_sidebar": "Contents",
   "toc_cell": false,
   "toc_position": {},
   "toc_section_display": true,
   "toc_window_display": false
  }
 },
 "nbformat": 4,
 "nbformat_minor": 2
}
