{
 "cells": [
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Looking at Genomic Loci with multiple edits"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 1,
   "metadata": {},
   "outputs": [],
   "source": [
    "import pandas as pd"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 2,
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/opt/modules/pkgs/anaconda/4.8/envs/CRISPEY_GxE/lib/python3.9/site-packages/IPython/core/interactiveshell.py:3457: FutureWarning: In a future version of pandas all arguments of read_csv except for the argument 'filepath_or_buffer' will be keyword-only.\n",
      "  exec(code_obj, self.user_global_ns, self.user_ns)\n",
      "/tmp/ipykernel_20449/1135356240.py:5: DtypeWarning: Columns (24,25,26,27,28,29,37) have mixed types. Specify dtype option on import or set low_memory=False.\n",
      "  oli_info = pd.read_csv(OLIGO_TABLE_PATH,'\\t')\n"
     ]
    }
   ],
   "source": [
    "annotations = pd.read_csv('../data_tables/erg_annos.tsv',sep = '\\t')\n",
    "fits = pd.read_csv('../data_tables/ergosterol_pathway_fitness.tsv',sep='\\t')\n",
    "annotated_fits =  annotations.merge(fits, 'inner', 'var_id')\n",
    "OLIGO_TABLE_PATH = '/home/shian/CRISPEY3_fraserserver/tables/oligos_nonuniq_crispey3_GG_9bp_OLIGO_with_seq_primers.txt'\n",
    "oli_info = pd.read_csv(OLIGO_TABLE_PATH,'\\t')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 3,
   "metadata": {},
   "outputs": [],
   "source": [
    "#Getting annotations for each variant in the ergosterol pool that was measured\n",
    "all_variants = annotated_fits['var_id'].unique()\n",
    "all_variant_annotations = annotations[(annotations['var_id'].isin(all_variants))]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 4,
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/tmp/ipykernel_20449/1743769777.py:2: SettingWithCopyWarning: \n",
      "A value is trying to be set on a copy of a slice from a DataFrame.\n",
      "Try using .loc[row_indexer,col_indexer] = value instead\n",
      "\n",
      "See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy\n",
      "  all_variant_annotations['edits_per_location'] = all_variant_annotations.groupby(['chrom','SNP_chr_pos'])['var_id'].transform('size')\n"
     ]
    }
   ],
   "source": [
    "#getting the number of variants/edits per genomic location\n",
    "all_variant_annotations['edits_per_location'] = all_variant_annotations.groupby(['chrom','SNP_chr_pos'])['var_id'].transform('size')\n",
    "\n",
    "#getting genomic locations with multiple edits\n",
    "multi_edits = all_variant_annotations[all_variant_annotations['edits_per_location']>1]['var_id'].unique()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 6,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "67"
      ]
     },
     "execution_count": 6,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "all_variant_annotations[all_variant_annotations['edits_per_location']>1]['SNP_chr_pos'].nunique()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 7,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "134"
      ]
     },
     "execution_count": 7,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "67*2"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### Missense Hits within sites w/ multiple edits"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 5,
   "metadata": {
    "scrolled": true
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Unnamed: 0_x</th>\n",
       "      <th>var_id</th>\n",
       "      <th>chrom</th>\n",
       "      <th>SNP_chr_pos</th>\n",
       "      <th>Gene</th>\n",
       "      <th>Consequence</th>\n",
       "      <th>ALT</th>\n",
       "      <th>CDS_position</th>\n",
       "      <th>Protein_position</th>\n",
       "      <th>Amino_acids</th>\n",
       "      <th>DISTANCE</th>\n",
       "      <th>Unnamed: 0_y</th>\n",
       "      <th>condition</th>\n",
       "      <th>coef</th>\n",
       "      <th>std err</th>\n",
       "      <th>t</th>\n",
       "      <th>pval</th>\n",
       "      <th>Conf. Int. Low</th>\n",
       "      <th>Conf. Int. Upp.</th>\n",
       "      <th>padj</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>1157</th>\n",
       "      <td>16298</td>\n",
       "      <td>ERG_00376</td>\n",
       "      <td>VII</td>\n",
       "      <td>847050</td>\n",
       "      <td>ERG1</td>\n",
       "      <td>missense_variant</td>\n",
       "      <td>A</td>\n",
       "      <td>1374</td>\n",
       "      <td>458</td>\n",
       "      <td>L/F</td>\n",
       "      <td>0</td>\n",
       "      <td>3835</td>\n",
       "      <td>TBF</td>\n",
       "      <td>0.085077</td>\n",
       "      <td>0.000861</td>\n",
       "      <td>98.760958</td>\n",
       "      <td>0.000000e+00</td>\n",
       "      <td>0.083388</td>\n",
       "      <td>0.086765</td>\n",
       "      <td>0.000000e+00</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2576</th>\n",
       "      <td>22187</td>\n",
       "      <td>EGB_00021</td>\n",
       "      <td>VIII</td>\n",
       "      <td>239187</td>\n",
       "      <td>ERG7</td>\n",
       "      <td>missense_variant</td>\n",
       "      <td>G</td>\n",
       "      <td>90</td>\n",
       "      <td>30</td>\n",
       "      <td>S/R</td>\n",
       "      <td>0</td>\n",
       "      <td>1080</td>\n",
       "      <td>COCL</td>\n",
       "      <td>0.035477</td>\n",
       "      <td>0.002511</td>\n",
       "      <td>14.130287</td>\n",
       "      <td>2.739628e-45</td>\n",
       "      <td>0.030556</td>\n",
       "      <td>0.040398</td>\n",
       "      <td>1.906484e-43</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4339</th>\n",
       "      <td>29444</td>\n",
       "      <td>EGE_00340</td>\n",
       "      <td>XII</td>\n",
       "      <td>341947</td>\n",
       "      <td>ERG27</td>\n",
       "      <td>missense_variant</td>\n",
       "      <td>C</td>\n",
       "      <td>138</td>\n",
       "      <td>46</td>\n",
       "      <td>Q/H</td>\n",
       "      <td>0</td>\n",
       "      <td>2815</td>\n",
       "      <td>LOV</td>\n",
       "      <td>-0.002613</td>\n",
       "      <td>0.000736</td>\n",
       "      <td>-3.550121</td>\n",
       "      <td>3.852356e-04</td>\n",
       "      <td>-0.004055</td>\n",
       "      <td>-0.001170</td>\n",
       "      <td>7.931789e-03</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4809</th>\n",
       "      <td>31980</td>\n",
       "      <td>EGD_00085</td>\n",
       "      <td>XII</td>\n",
       "      <td>1034989</td>\n",
       "      <td>HMG2</td>\n",
       "      <td>missense_variant</td>\n",
       "      <td>T</td>\n",
       "      <td>2363</td>\n",
       "      <td>788</td>\n",
       "      <td>C/F</td>\n",
       "      <td>0</td>\n",
       "      <td>2007</td>\n",
       "      <td>COCL</td>\n",
       "      <td>0.030071</td>\n",
       "      <td>0.001353</td>\n",
       "      <td>22.224510</td>\n",
       "      <td>3.721192e-109</td>\n",
       "      <td>0.027419</td>\n",
       "      <td>0.032723</td>\n",
       "      <td>3.884318e-107</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4810</th>\n",
       "      <td>31980</td>\n",
       "      <td>EGD_00085</td>\n",
       "      <td>XII</td>\n",
       "      <td>1034989</td>\n",
       "      <td>HMG2</td>\n",
       "      <td>missense_variant</td>\n",
       "      <td>T</td>\n",
       "      <td>2363</td>\n",
       "      <td>788</td>\n",
       "      <td>C/F</td>\n",
       "      <td>0</td>\n",
       "      <td>2008</td>\n",
       "      <td>LOV</td>\n",
       "      <td>-0.024343</td>\n",
       "      <td>0.000696</td>\n",
       "      <td>-34.994723</td>\n",
       "      <td>1.231939e-266</td>\n",
       "      <td>-0.025706</td>\n",
       "      <td>-0.022979</td>\n",
       "      <td>2.457581e-264</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4811</th>\n",
       "      <td>31980</td>\n",
       "      <td>EGD_00085</td>\n",
       "      <td>XII</td>\n",
       "      <td>1034989</td>\n",
       "      <td>HMG2</td>\n",
       "      <td>missense_variant</td>\n",
       "      <td>T</td>\n",
       "      <td>2363</td>\n",
       "      <td>788</td>\n",
       "      <td>C/F</td>\n",
       "      <td>0</td>\n",
       "      <td>2009</td>\n",
       "      <td>NACL</td>\n",
       "      <td>-0.001828</td>\n",
       "      <td>0.000429</td>\n",
       "      <td>-4.259399</td>\n",
       "      <td>2.051685e-05</td>\n",
       "      <td>-0.002670</td>\n",
       "      <td>-0.000987</td>\n",
       "      <td>5.360108e-04</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4812</th>\n",
       "      <td>31980</td>\n",
       "      <td>EGD_00085</td>\n",
       "      <td>XII</td>\n",
       "      <td>1034989</td>\n",
       "      <td>HMG2</td>\n",
       "      <td>missense_variant</td>\n",
       "      <td>T</td>\n",
       "      <td>2363</td>\n",
       "      <td>788</td>\n",
       "      <td>C/F</td>\n",
       "      <td>0</td>\n",
       "      <td>2010</td>\n",
       "      <td>SC</td>\n",
       "      <td>0.001595</td>\n",
       "      <td>0.000427</td>\n",
       "      <td>3.732065</td>\n",
       "      <td>1.900243e-04</td>\n",
       "      <td>0.000758</td>\n",
       "      <td>0.002433</td>\n",
       "      <td>4.264621e-03</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4815</th>\n",
       "      <td>31984</td>\n",
       "      <td>EGD_00086</td>\n",
       "      <td>XII</td>\n",
       "      <td>1034989</td>\n",
       "      <td>HMG2</td>\n",
       "      <td>missense_variant</td>\n",
       "      <td>A</td>\n",
       "      <td>2363</td>\n",
       "      <td>788</td>\n",
       "      <td>C/Y</td>\n",
       "      <td>0</td>\n",
       "      <td>2013</td>\n",
       "      <td>COCL</td>\n",
       "      <td>0.030950</td>\n",
       "      <td>0.001555</td>\n",
       "      <td>19.909303</td>\n",
       "      <td>5.060471e-88</td>\n",
       "      <td>0.027903</td>\n",
       "      <td>0.033997</td>\n",
       "      <td>4.683284e-86</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4816</th>\n",
       "      <td>31984</td>\n",
       "      <td>EGD_00086</td>\n",
       "      <td>XII</td>\n",
       "      <td>1034989</td>\n",
       "      <td>HMG2</td>\n",
       "      <td>missense_variant</td>\n",
       "      <td>A</td>\n",
       "      <td>2363</td>\n",
       "      <td>788</td>\n",
       "      <td>C/Y</td>\n",
       "      <td>0</td>\n",
       "      <td>2014</td>\n",
       "      <td>LOV</td>\n",
       "      <td>-0.044398</td>\n",
       "      <td>0.000955</td>\n",
       "      <td>-46.485171</td>\n",
       "      <td>0.000000e+00</td>\n",
       "      <td>-0.046270</td>\n",
       "      <td>-0.042526</td>\n",
       "      <td>0.000000e+00</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4817</th>\n",
       "      <td>31984</td>\n",
       "      <td>EGD_00086</td>\n",
       "      <td>XII</td>\n",
       "      <td>1034989</td>\n",
       "      <td>HMG2</td>\n",
       "      <td>missense_variant</td>\n",
       "      <td>A</td>\n",
       "      <td>2363</td>\n",
       "      <td>788</td>\n",
       "      <td>C/Y</td>\n",
       "      <td>0</td>\n",
       "      <td>2015</td>\n",
       "      <td>NACL</td>\n",
       "      <td>-0.003589</td>\n",
       "      <td>0.000494</td>\n",
       "      <td>-7.266212</td>\n",
       "      <td>3.724574e-13</td>\n",
       "      <td>-0.004558</td>\n",
       "      <td>-0.002621</td>\n",
       "      <td>1.519795e-11</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5337</th>\n",
       "      <td>32625</td>\n",
       "      <td>ERG_01754</td>\n",
       "      <td>XIII</td>\n",
       "      <td>115802</td>\n",
       "      <td>HMG1</td>\n",
       "      <td>missense_variant</td>\n",
       "      <td>C</td>\n",
       "      <td>3097</td>\n",
       "      <td>1033</td>\n",
       "      <td>P/A</td>\n",
       "      <td>0</td>\n",
       "      <td>6501</td>\n",
       "      <td>CAFF</td>\n",
       "      <td>0.007017</td>\n",
       "      <td>0.001589</td>\n",
       "      <td>4.414814</td>\n",
       "      <td>1.012054e-05</td>\n",
       "      <td>0.003902</td>\n",
       "      <td>0.010132</td>\n",
       "      <td>2.786875e-04</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5339</th>\n",
       "      <td>32625</td>\n",
       "      <td>ERG_01754</td>\n",
       "      <td>XIII</td>\n",
       "      <td>115802</td>\n",
       "      <td>HMG1</td>\n",
       "      <td>missense_variant</td>\n",
       "      <td>C</td>\n",
       "      <td>3097</td>\n",
       "      <td>1033</td>\n",
       "      <td>P/A</td>\n",
       "      <td>0</td>\n",
       "      <td>6503</td>\n",
       "      <td>LOV</td>\n",
       "      <td>0.020164</td>\n",
       "      <td>0.001627</td>\n",
       "      <td>12.396164</td>\n",
       "      <td>2.914492e-35</td>\n",
       "      <td>0.016976</td>\n",
       "      <td>0.023352</td>\n",
       "      <td>1.829608e-33</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5345</th>\n",
       "      <td>32631</td>\n",
       "      <td>ERG_01755</td>\n",
       "      <td>XIII</td>\n",
       "      <td>115802</td>\n",
       "      <td>HMG1</td>\n",
       "      <td>missense_variant</td>\n",
       "      <td>T</td>\n",
       "      <td>3097</td>\n",
       "      <td>1033</td>\n",
       "      <td>P/T</td>\n",
       "      <td>0</td>\n",
       "      <td>6509</td>\n",
       "      <td>LOV</td>\n",
       "      <td>-0.013972</td>\n",
       "      <td>0.003414</td>\n",
       "      <td>-4.093143</td>\n",
       "      <td>4.259076e-05</td>\n",
       "      <td>-0.020662</td>\n",
       "      <td>-0.007282</td>\n",
       "      <td>1.059106e-03</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "      Unnamed: 0_x     var_id chrom  SNP_chr_pos   Gene       Consequence ALT  \\\n",
       "1157         16298  ERG_00376   VII       847050   ERG1  missense_variant   A   \n",
       "2576         22187  EGB_00021  VIII       239187   ERG7  missense_variant   G   \n",
       "4339         29444  EGE_00340   XII       341947  ERG27  missense_variant   C   \n",
       "4809         31980  EGD_00085   XII      1034989   HMG2  missense_variant   T   \n",
       "4810         31980  EGD_00085   XII      1034989   HMG2  missense_variant   T   \n",
       "4811         31980  EGD_00085   XII      1034989   HMG2  missense_variant   T   \n",
       "4812         31980  EGD_00085   XII      1034989   HMG2  missense_variant   T   \n",
       "4815         31984  EGD_00086   XII      1034989   HMG2  missense_variant   A   \n",
       "4816         31984  EGD_00086   XII      1034989   HMG2  missense_variant   A   \n",
       "4817         31984  EGD_00086   XII      1034989   HMG2  missense_variant   A   \n",
       "5337         32625  ERG_01754  XIII       115802   HMG1  missense_variant   C   \n",
       "5339         32625  ERG_01754  XIII       115802   HMG1  missense_variant   C   \n",
       "5345         32631  ERG_01755  XIII       115802   HMG1  missense_variant   T   \n",
       "\n",
       "     CDS_position Protein_position Amino_acids  DISTANCE  Unnamed: 0_y  \\\n",
       "1157         1374              458         L/F         0          3835   \n",
       "2576           90               30         S/R         0          1080   \n",
       "4339          138               46         Q/H         0          2815   \n",
       "4809         2363              788         C/F         0          2007   \n",
       "4810         2363              788         C/F         0          2008   \n",
       "4811         2363              788         C/F         0          2009   \n",
       "4812         2363              788         C/F         0          2010   \n",
       "4815         2363              788         C/Y         0          2013   \n",
       "4816         2363              788         C/Y         0          2014   \n",
       "4817         2363              788         C/Y         0          2015   \n",
       "5337         3097             1033         P/A         0          6501   \n",
       "5339         3097             1033         P/A         0          6503   \n",
       "5345         3097             1033         P/T         0          6509   \n",
       "\n",
       "     condition      coef   std err          t           pval  Conf. Int. Low  \\\n",
       "1157       TBF  0.085077  0.000861  98.760958   0.000000e+00        0.083388   \n",
       "2576      COCL  0.035477  0.002511  14.130287   2.739628e-45        0.030556   \n",
       "4339       LOV -0.002613  0.000736  -3.550121   3.852356e-04       -0.004055   \n",
       "4809      COCL  0.030071  0.001353  22.224510  3.721192e-109        0.027419   \n",
       "4810       LOV -0.024343  0.000696 -34.994723  1.231939e-266       -0.025706   \n",
       "4811      NACL -0.001828  0.000429  -4.259399   2.051685e-05       -0.002670   \n",
       "4812        SC  0.001595  0.000427   3.732065   1.900243e-04        0.000758   \n",
       "4815      COCL  0.030950  0.001555  19.909303   5.060471e-88        0.027903   \n",
       "4816       LOV -0.044398  0.000955 -46.485171   0.000000e+00       -0.046270   \n",
       "4817      NACL -0.003589  0.000494  -7.266212   3.724574e-13       -0.004558   \n",
       "5337      CAFF  0.007017  0.001589   4.414814   1.012054e-05        0.003902   \n",
       "5339       LOV  0.020164  0.001627  12.396164   2.914492e-35        0.016976   \n",
       "5345       LOV -0.013972  0.003414  -4.093143   4.259076e-05       -0.020662   \n",
       "\n",
       "      Conf. Int. Upp.           padj  \n",
       "1157         0.086765   0.000000e+00  \n",
       "2576         0.040398   1.906484e-43  \n",
       "4339        -0.001170   7.931789e-03  \n",
       "4809         0.032723  3.884318e-107  \n",
       "4810        -0.022979  2.457581e-264  \n",
       "4811        -0.000987   5.360108e-04  \n",
       "4812         0.002433   4.264621e-03  \n",
       "4815         0.033997   4.683284e-86  \n",
       "4816        -0.042526   0.000000e+00  \n",
       "4817        -0.002621   1.519795e-11  \n",
       "5337         0.010132   2.786875e-04  \n",
       "5339         0.023352   1.829608e-33  \n",
       "5345        -0.007282   1.059106e-03  "
      ]
     },
     "execution_count": 5,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "#displaying all missense variants which are hits and target a site with multiple edits \n",
    "annotated_fits[(annotated_fits['var_id'].isin(multi_edits))&(annotated_fits['padj']<.01)&(annotated_fits['Consequence']=='missense_variant')]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 6,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Unnamed: 0_x</th>\n",
       "      <th>var_id</th>\n",
       "      <th>chrom</th>\n",
       "      <th>SNP_chr_pos</th>\n",
       "      <th>Gene</th>\n",
       "      <th>Consequence</th>\n",
       "      <th>ALT</th>\n",
       "      <th>CDS_position</th>\n",
       "      <th>Protein_position</th>\n",
       "      <th>Amino_acids</th>\n",
       "      <th>DISTANCE</th>\n",
       "      <th>Unnamed: 0_y</th>\n",
       "      <th>condition</th>\n",
       "      <th>coef</th>\n",
       "      <th>std err</th>\n",
       "      <th>t</th>\n",
       "      <th>pval</th>\n",
       "      <th>Conf. Int. Low</th>\n",
       "      <th>Conf. Int. Upp.</th>\n",
       "      <th>padj</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>4337</th>\n",
       "      <td>29444</td>\n",
       "      <td>EGE_00340</td>\n",
       "      <td>XII</td>\n",
       "      <td>341947</td>\n",
       "      <td>ERG27</td>\n",
       "      <td>missense_variant</td>\n",
       "      <td>C</td>\n",
       "      <td>138</td>\n",
       "      <td>46</td>\n",
       "      <td>Q/H</td>\n",
       "      <td>0</td>\n",
       "      <td>2813</td>\n",
       "      <td>CAFF</td>\n",
       "      <td>-0.000161</td>\n",
       "      <td>0.000540</td>\n",
       "      <td>-0.297753</td>\n",
       "      <td>0.765893</td>\n",
       "      <td>-0.001218</td>\n",
       "      <td>0.000897</td>\n",
       "      <td>0.938200</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4338</th>\n",
       "      <td>29444</td>\n",
       "      <td>EGE_00340</td>\n",
       "      <td>XII</td>\n",
       "      <td>341947</td>\n",
       "      <td>ERG27</td>\n",
       "      <td>missense_variant</td>\n",
       "      <td>C</td>\n",
       "      <td>138</td>\n",
       "      <td>46</td>\n",
       "      <td>Q/H</td>\n",
       "      <td>0</td>\n",
       "      <td>2814</td>\n",
       "      <td>COCL</td>\n",
       "      <td>0.001700</td>\n",
       "      <td>0.001801</td>\n",
       "      <td>0.943934</td>\n",
       "      <td>0.345206</td>\n",
       "      <td>-0.001830</td>\n",
       "      <td>0.005230</td>\n",
       "      <td>0.719297</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4339</th>\n",
       "      <td>29444</td>\n",
       "      <td>EGE_00340</td>\n",
       "      <td>XII</td>\n",
       "      <td>341947</td>\n",
       "      <td>ERG27</td>\n",
       "      <td>missense_variant</td>\n",
       "      <td>C</td>\n",
       "      <td>138</td>\n",
       "      <td>46</td>\n",
       "      <td>Q/H</td>\n",
       "      <td>0</td>\n",
       "      <td>2815</td>\n",
       "      <td>LOV</td>\n",
       "      <td>-0.002613</td>\n",
       "      <td>0.000736</td>\n",
       "      <td>-3.550121</td>\n",
       "      <td>0.000385</td>\n",
       "      <td>-0.004055</td>\n",
       "      <td>-0.001170</td>\n",
       "      <td>0.007932</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4340</th>\n",
       "      <td>29444</td>\n",
       "      <td>EGE_00340</td>\n",
       "      <td>XII</td>\n",
       "      <td>341947</td>\n",
       "      <td>ERG27</td>\n",
       "      <td>missense_variant</td>\n",
       "      <td>C</td>\n",
       "      <td>138</td>\n",
       "      <td>46</td>\n",
       "      <td>Q/H</td>\n",
       "      <td>0</td>\n",
       "      <td>2816</td>\n",
       "      <td>NACL</td>\n",
       "      <td>-0.001254</td>\n",
       "      <td>0.000536</td>\n",
       "      <td>-2.339868</td>\n",
       "      <td>0.019293</td>\n",
       "      <td>-0.002304</td>\n",
       "      <td>-0.000204</td>\n",
       "      <td>0.176181</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4341</th>\n",
       "      <td>29444</td>\n",
       "      <td>EGE_00340</td>\n",
       "      <td>XII</td>\n",
       "      <td>341947</td>\n",
       "      <td>ERG27</td>\n",
       "      <td>missense_variant</td>\n",
       "      <td>C</td>\n",
       "      <td>138</td>\n",
       "      <td>46</td>\n",
       "      <td>Q/H</td>\n",
       "      <td>0</td>\n",
       "      <td>2817</td>\n",
       "      <td>SC</td>\n",
       "      <td>-0.001048</td>\n",
       "      <td>0.000541</td>\n",
       "      <td>-1.937658</td>\n",
       "      <td>0.052668</td>\n",
       "      <td>-0.002107</td>\n",
       "      <td>0.000012</td>\n",
       "      <td>0.307812</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4342</th>\n",
       "      <td>29444</td>\n",
       "      <td>EGE_00340</td>\n",
       "      <td>XII</td>\n",
       "      <td>341947</td>\n",
       "      <td>ERG27</td>\n",
       "      <td>missense_variant</td>\n",
       "      <td>C</td>\n",
       "      <td>138</td>\n",
       "      <td>46</td>\n",
       "      <td>Q/H</td>\n",
       "      <td>0</td>\n",
       "      <td>2818</td>\n",
       "      <td>TBF</td>\n",
       "      <td>-0.000416</td>\n",
       "      <td>0.000381</td>\n",
       "      <td>-1.092802</td>\n",
       "      <td>0.274484</td>\n",
       "      <td>-0.001163</td>\n",
       "      <td>0.000330</td>\n",
       "      <td>0.661455</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4343</th>\n",
       "      <td>29453</td>\n",
       "      <td>EGE_00341</td>\n",
       "      <td>XII</td>\n",
       "      <td>341947</td>\n",
       "      <td>ERG27</td>\n",
       "      <td>synonymous_variant</td>\n",
       "      <td>A</td>\n",
       "      <td>138</td>\n",
       "      <td>46</td>\n",
       "      <td>Q</td>\n",
       "      <td>0</td>\n",
       "      <td>2819</td>\n",
       "      <td>CAFF</td>\n",
       "      <td>0.001806</td>\n",
       "      <td>0.000895</td>\n",
       "      <td>2.019187</td>\n",
       "      <td>0.043471</td>\n",
       "      <td>0.000053</td>\n",
       "      <td>0.003560</td>\n",
       "      <td>0.278739</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4344</th>\n",
       "      <td>29453</td>\n",
       "      <td>EGE_00341</td>\n",
       "      <td>XII</td>\n",
       "      <td>341947</td>\n",
       "      <td>ERG27</td>\n",
       "      <td>synonymous_variant</td>\n",
       "      <td>A</td>\n",
       "      <td>138</td>\n",
       "      <td>46</td>\n",
       "      <td>Q</td>\n",
       "      <td>0</td>\n",
       "      <td>2820</td>\n",
       "      <td>COCL</td>\n",
       "      <td>-0.005362</td>\n",
       "      <td>0.002492</td>\n",
       "      <td>-2.151660</td>\n",
       "      <td>0.031427</td>\n",
       "      <td>-0.010245</td>\n",
       "      <td>-0.000478</td>\n",
       "      <td>0.236402</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4345</th>\n",
       "      <td>29453</td>\n",
       "      <td>EGE_00341</td>\n",
       "      <td>XII</td>\n",
       "      <td>341947</td>\n",
       "      <td>ERG27</td>\n",
       "      <td>synonymous_variant</td>\n",
       "      <td>A</td>\n",
       "      <td>138</td>\n",
       "      <td>46</td>\n",
       "      <td>Q</td>\n",
       "      <td>0</td>\n",
       "      <td>2821</td>\n",
       "      <td>LOV</td>\n",
       "      <td>-0.001681</td>\n",
       "      <td>0.001012</td>\n",
       "      <td>-1.661501</td>\n",
       "      <td>0.096616</td>\n",
       "      <td>-0.003664</td>\n",
       "      <td>0.000302</td>\n",
       "      <td>0.418640</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4346</th>\n",
       "      <td>29453</td>\n",
       "      <td>EGE_00341</td>\n",
       "      <td>XII</td>\n",
       "      <td>341947</td>\n",
       "      <td>ERG27</td>\n",
       "      <td>synonymous_variant</td>\n",
       "      <td>A</td>\n",
       "      <td>138</td>\n",
       "      <td>46</td>\n",
       "      <td>Q</td>\n",
       "      <td>0</td>\n",
       "      <td>2822</td>\n",
       "      <td>NACL</td>\n",
       "      <td>-0.000939</td>\n",
       "      <td>0.000886</td>\n",
       "      <td>-1.059815</td>\n",
       "      <td>0.289231</td>\n",
       "      <td>-0.002675</td>\n",
       "      <td>0.000797</td>\n",
       "      <td>0.673173</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4347</th>\n",
       "      <td>29453</td>\n",
       "      <td>EGE_00341</td>\n",
       "      <td>XII</td>\n",
       "      <td>341947</td>\n",
       "      <td>ERG27</td>\n",
       "      <td>synonymous_variant</td>\n",
       "      <td>A</td>\n",
       "      <td>138</td>\n",
       "      <td>46</td>\n",
       "      <td>Q</td>\n",
       "      <td>0</td>\n",
       "      <td>2823</td>\n",
       "      <td>SC</td>\n",
       "      <td>-0.001439</td>\n",
       "      <td>0.000818</td>\n",
       "      <td>-1.759433</td>\n",
       "      <td>0.078507</td>\n",
       "      <td>-0.003041</td>\n",
       "      <td>0.000164</td>\n",
       "      <td>0.376273</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4348</th>\n",
       "      <td>29453</td>\n",
       "      <td>EGE_00341</td>\n",
       "      <td>XII</td>\n",
       "      <td>341947</td>\n",
       "      <td>ERG27</td>\n",
       "      <td>synonymous_variant</td>\n",
       "      <td>A</td>\n",
       "      <td>138</td>\n",
       "      <td>46</td>\n",
       "      <td>Q</td>\n",
       "      <td>0</td>\n",
       "      <td>2824</td>\n",
       "      <td>TBF</td>\n",
       "      <td>-0.000238</td>\n",
       "      <td>0.000619</td>\n",
       "      <td>-0.384168</td>\n",
       "      <td>0.700855</td>\n",
       "      <td>-0.001452</td>\n",
       "      <td>0.000976</td>\n",
       "      <td>0.911030</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "      Unnamed: 0_x     var_id chrom  SNP_chr_pos   Gene         Consequence  \\\n",
       "4337         29444  EGE_00340   XII       341947  ERG27    missense_variant   \n",
       "4338         29444  EGE_00340   XII       341947  ERG27    missense_variant   \n",
       "4339         29444  EGE_00340   XII       341947  ERG27    missense_variant   \n",
       "4340         29444  EGE_00340   XII       341947  ERG27    missense_variant   \n",
       "4341         29444  EGE_00340   XII       341947  ERG27    missense_variant   \n",
       "4342         29444  EGE_00340   XII       341947  ERG27    missense_variant   \n",
       "4343         29453  EGE_00341   XII       341947  ERG27  synonymous_variant   \n",
       "4344         29453  EGE_00341   XII       341947  ERG27  synonymous_variant   \n",
       "4345         29453  EGE_00341   XII       341947  ERG27  synonymous_variant   \n",
       "4346         29453  EGE_00341   XII       341947  ERG27  synonymous_variant   \n",
       "4347         29453  EGE_00341   XII       341947  ERG27  synonymous_variant   \n",
       "4348         29453  EGE_00341   XII       341947  ERG27  synonymous_variant   \n",
       "\n",
       "     ALT CDS_position Protein_position Amino_acids  DISTANCE  Unnamed: 0_y  \\\n",
       "4337   C          138               46         Q/H         0          2813   \n",
       "4338   C          138               46         Q/H         0          2814   \n",
       "4339   C          138               46         Q/H         0          2815   \n",
       "4340   C          138               46         Q/H         0          2816   \n",
       "4341   C          138               46         Q/H         0          2817   \n",
       "4342   C          138               46         Q/H         0          2818   \n",
       "4343   A          138               46           Q         0          2819   \n",
       "4344   A          138               46           Q         0          2820   \n",
       "4345   A          138               46           Q         0          2821   \n",
       "4346   A          138               46           Q         0          2822   \n",
       "4347   A          138               46           Q         0          2823   \n",
       "4348   A          138               46           Q         0          2824   \n",
       "\n",
       "     condition      coef   std err         t      pval  Conf. Int. Low  \\\n",
       "4337      CAFF -0.000161  0.000540 -0.297753  0.765893       -0.001218   \n",
       "4338      COCL  0.001700  0.001801  0.943934  0.345206       -0.001830   \n",
       "4339       LOV -0.002613  0.000736 -3.550121  0.000385       -0.004055   \n",
       "4340      NACL -0.001254  0.000536 -2.339868  0.019293       -0.002304   \n",
       "4341        SC -0.001048  0.000541 -1.937658  0.052668       -0.002107   \n",
       "4342       TBF -0.000416  0.000381 -1.092802  0.274484       -0.001163   \n",
       "4343      CAFF  0.001806  0.000895  2.019187  0.043471        0.000053   \n",
       "4344      COCL -0.005362  0.002492 -2.151660  0.031427       -0.010245   \n",
       "4345       LOV -0.001681  0.001012 -1.661501  0.096616       -0.003664   \n",
       "4346      NACL -0.000939  0.000886 -1.059815  0.289231       -0.002675   \n",
       "4347        SC -0.001439  0.000818 -1.759433  0.078507       -0.003041   \n",
       "4348       TBF -0.000238  0.000619 -0.384168  0.700855       -0.001452   \n",
       "\n",
       "      Conf. Int. Upp.      padj  \n",
       "4337         0.000897  0.938200  \n",
       "4338         0.005230  0.719297  \n",
       "4339        -0.001170  0.007932  \n",
       "4340        -0.000204  0.176181  \n",
       "4341         0.000012  0.307812  \n",
       "4342         0.000330  0.661455  \n",
       "4343         0.003560  0.278739  \n",
       "4344        -0.000478  0.236402  \n",
       "4345         0.000302  0.418640  \n",
       "4346         0.000797  0.673173  \n",
       "4347         0.000164  0.376273  \n",
       "4348         0.000976  0.911030  "
      ]
     },
     "execution_count": 6,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "#ERG27 missense variant with a significant fitness effect in lovastatin,\n",
    "#synonymous variant in same site has no significant effect\n",
    "\n",
    "chrom = 'XII'\n",
    "pos = 341947\n",
    "\n",
    "annotated_fits[(annotated_fits['chrom']==chrom)&(annotated_fits['SNP_chr_pos']==pos)]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 7,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Unnamed: 0_x</th>\n",
       "      <th>var_id</th>\n",
       "      <th>chrom</th>\n",
       "      <th>SNP_chr_pos</th>\n",
       "      <th>Gene</th>\n",
       "      <th>Consequence</th>\n",
       "      <th>ALT</th>\n",
       "      <th>CDS_position</th>\n",
       "      <th>Protein_position</th>\n",
       "      <th>Amino_acids</th>\n",
       "      <th>DISTANCE</th>\n",
       "      <th>Unnamed: 0_y</th>\n",
       "      <th>condition</th>\n",
       "      <th>coef</th>\n",
       "      <th>std err</th>\n",
       "      <th>t</th>\n",
       "      <th>pval</th>\n",
       "      <th>Conf. Int. Low</th>\n",
       "      <th>Conf. Int. Upp.</th>\n",
       "      <th>padj</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>1146</th>\n",
       "      <td>16288</td>\n",
       "      <td>ERG_00375</td>\n",
       "      <td>VII</td>\n",
       "      <td>847050</td>\n",
       "      <td>ERG1</td>\n",
       "      <td>synonymous_variant</td>\n",
       "      <td>T</td>\n",
       "      <td>1374</td>\n",
       "      <td>458</td>\n",
       "      <td>L</td>\n",
       "      <td>0</td>\n",
       "      <td>3824</td>\n",
       "      <td>CAFF</td>\n",
       "      <td>0.000210</td>\n",
       "      <td>0.002225</td>\n",
       "      <td>0.094274</td>\n",
       "      <td>0.924892</td>\n",
       "      <td>-0.004152</td>\n",
       "      <td>0.004571</td>\n",
       "      <td>0.987177</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1147</th>\n",
       "      <td>16288</td>\n",
       "      <td>ERG_00375</td>\n",
       "      <td>VII</td>\n",
       "      <td>847050</td>\n",
       "      <td>ERG1</td>\n",
       "      <td>synonymous_variant</td>\n",
       "      <td>T</td>\n",
       "      <td>1374</td>\n",
       "      <td>458</td>\n",
       "      <td>L</td>\n",
       "      <td>0</td>\n",
       "      <td>3825</td>\n",
       "      <td>COCL</td>\n",
       "      <td>-0.005909</td>\n",
       "      <td>0.006614</td>\n",
       "      <td>-0.893530</td>\n",
       "      <td>0.371576</td>\n",
       "      <td>-0.018872</td>\n",
       "      <td>0.007053</td>\n",
       "      <td>0.743566</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1148</th>\n",
       "      <td>16288</td>\n",
       "      <td>ERG_00375</td>\n",
       "      <td>VII</td>\n",
       "      <td>847050</td>\n",
       "      <td>ERG1</td>\n",
       "      <td>synonymous_variant</td>\n",
       "      <td>T</td>\n",
       "      <td>1374</td>\n",
       "      <td>458</td>\n",
       "      <td>L</td>\n",
       "      <td>0</td>\n",
       "      <td>3826</td>\n",
       "      <td>LOV</td>\n",
       "      <td>-0.000577</td>\n",
       "      <td>0.002960</td>\n",
       "      <td>-0.194876</td>\n",
       "      <td>0.845490</td>\n",
       "      <td>-0.006379</td>\n",
       "      <td>0.005226</td>\n",
       "      <td>0.967488</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1149</th>\n",
       "      <td>16288</td>\n",
       "      <td>ERG_00375</td>\n",
       "      <td>VII</td>\n",
       "      <td>847050</td>\n",
       "      <td>ERG1</td>\n",
       "      <td>synonymous_variant</td>\n",
       "      <td>T</td>\n",
       "      <td>1374</td>\n",
       "      <td>458</td>\n",
       "      <td>L</td>\n",
       "      <td>0</td>\n",
       "      <td>3827</td>\n",
       "      <td>NACL</td>\n",
       "      <td>-0.003936</td>\n",
       "      <td>0.002384</td>\n",
       "      <td>-1.651079</td>\n",
       "      <td>0.098726</td>\n",
       "      <td>-0.008609</td>\n",
       "      <td>0.000736</td>\n",
       "      <td>0.421828</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1150</th>\n",
       "      <td>16288</td>\n",
       "      <td>ERG_00375</td>\n",
       "      <td>VII</td>\n",
       "      <td>847050</td>\n",
       "      <td>ERG1</td>\n",
       "      <td>synonymous_variant</td>\n",
       "      <td>T</td>\n",
       "      <td>1374</td>\n",
       "      <td>458</td>\n",
       "      <td>L</td>\n",
       "      <td>0</td>\n",
       "      <td>3828</td>\n",
       "      <td>SC</td>\n",
       "      <td>-0.001373</td>\n",
       "      <td>0.002327</td>\n",
       "      <td>-0.589969</td>\n",
       "      <td>0.555213</td>\n",
       "      <td>-0.005934</td>\n",
       "      <td>0.003188</td>\n",
       "      <td>0.848076</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1151</th>\n",
       "      <td>16288</td>\n",
       "      <td>ERG_00375</td>\n",
       "      <td>VII</td>\n",
       "      <td>847050</td>\n",
       "      <td>ERG1</td>\n",
       "      <td>synonymous_variant</td>\n",
       "      <td>T</td>\n",
       "      <td>1374</td>\n",
       "      <td>458</td>\n",
       "      <td>L</td>\n",
       "      <td>0</td>\n",
       "      <td>3829</td>\n",
       "      <td>TBF</td>\n",
       "      <td>-0.000805</td>\n",
       "      <td>0.001966</td>\n",
       "      <td>-0.409466</td>\n",
       "      <td>0.682198</td>\n",
       "      <td>-0.004658</td>\n",
       "      <td>0.003048</td>\n",
       "      <td>0.904974</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1152</th>\n",
       "      <td>16298</td>\n",
       "      <td>ERG_00376</td>\n",
       "      <td>VII</td>\n",
       "      <td>847050</td>\n",
       "      <td>ERG1</td>\n",
       "      <td>missense_variant</td>\n",
       "      <td>A</td>\n",
       "      <td>1374</td>\n",
       "      <td>458</td>\n",
       "      <td>L/F</td>\n",
       "      <td>0</td>\n",
       "      <td>3830</td>\n",
       "      <td>CAFF</td>\n",
       "      <td>0.001629</td>\n",
       "      <td>0.002467</td>\n",
       "      <td>0.660253</td>\n",
       "      <td>0.509093</td>\n",
       "      <td>-0.003206</td>\n",
       "      <td>0.006464</td>\n",
       "      <td>0.828994</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1153</th>\n",
       "      <td>16298</td>\n",
       "      <td>ERG_00376</td>\n",
       "      <td>VII</td>\n",
       "      <td>847050</td>\n",
       "      <td>ERG1</td>\n",
       "      <td>missense_variant</td>\n",
       "      <td>A</td>\n",
       "      <td>1374</td>\n",
       "      <td>458</td>\n",
       "      <td>L/F</td>\n",
       "      <td>0</td>\n",
       "      <td>3831</td>\n",
       "      <td>COCL</td>\n",
       "      <td>-0.015084</td>\n",
       "      <td>0.006839</td>\n",
       "      <td>-2.205619</td>\n",
       "      <td>0.027413</td>\n",
       "      <td>-0.028488</td>\n",
       "      <td>-0.001680</td>\n",
       "      <td>0.218162</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1154</th>\n",
       "      <td>16298</td>\n",
       "      <td>ERG_00376</td>\n",
       "      <td>VII</td>\n",
       "      <td>847050</td>\n",
       "      <td>ERG1</td>\n",
       "      <td>missense_variant</td>\n",
       "      <td>A</td>\n",
       "      <td>1374</td>\n",
       "      <td>458</td>\n",
       "      <td>L/F</td>\n",
       "      <td>0</td>\n",
       "      <td>3832</td>\n",
       "      <td>LOV</td>\n",
       "      <td>0.007815</td>\n",
       "      <td>0.002675</td>\n",
       "      <td>2.921551</td>\n",
       "      <td>0.003484</td>\n",
       "      <td>0.002572</td>\n",
       "      <td>0.013058</td>\n",
       "      <td>0.050934</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1155</th>\n",
       "      <td>16298</td>\n",
       "      <td>ERG_00376</td>\n",
       "      <td>VII</td>\n",
       "      <td>847050</td>\n",
       "      <td>ERG1</td>\n",
       "      <td>missense_variant</td>\n",
       "      <td>A</td>\n",
       "      <td>1374</td>\n",
       "      <td>458</td>\n",
       "      <td>L/F</td>\n",
       "      <td>0</td>\n",
       "      <td>3833</td>\n",
       "      <td>NACL</td>\n",
       "      <td>0.002963</td>\n",
       "      <td>0.002400</td>\n",
       "      <td>1.234558</td>\n",
       "      <td>0.216998</td>\n",
       "      <td>-0.001741</td>\n",
       "      <td>0.007667</td>\n",
       "      <td>0.599014</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1156</th>\n",
       "      <td>16298</td>\n",
       "      <td>ERG_00376</td>\n",
       "      <td>VII</td>\n",
       "      <td>847050</td>\n",
       "      <td>ERG1</td>\n",
       "      <td>missense_variant</td>\n",
       "      <td>A</td>\n",
       "      <td>1374</td>\n",
       "      <td>458</td>\n",
       "      <td>L/F</td>\n",
       "      <td>0</td>\n",
       "      <td>3834</td>\n",
       "      <td>SC</td>\n",
       "      <td>-0.001063</td>\n",
       "      <td>0.002404</td>\n",
       "      <td>-0.442296</td>\n",
       "      <td>0.658276</td>\n",
       "      <td>-0.005775</td>\n",
       "      <td>0.003649</td>\n",
       "      <td>0.896217</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1157</th>\n",
       "      <td>16298</td>\n",
       "      <td>ERG_00376</td>\n",
       "      <td>VII</td>\n",
       "      <td>847050</td>\n",
       "      <td>ERG1</td>\n",
       "      <td>missense_variant</td>\n",
       "      <td>A</td>\n",
       "      <td>1374</td>\n",
       "      <td>458</td>\n",
       "      <td>L/F</td>\n",
       "      <td>0</td>\n",
       "      <td>3835</td>\n",
       "      <td>TBF</td>\n",
       "      <td>0.085077</td>\n",
       "      <td>0.000861</td>\n",
       "      <td>98.760958</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.083388</td>\n",
       "      <td>0.086765</td>\n",
       "      <td>0.000000</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "      Unnamed: 0_x     var_id chrom  SNP_chr_pos  Gene         Consequence  \\\n",
       "1146         16288  ERG_00375   VII       847050  ERG1  synonymous_variant   \n",
       "1147         16288  ERG_00375   VII       847050  ERG1  synonymous_variant   \n",
       "1148         16288  ERG_00375   VII       847050  ERG1  synonymous_variant   \n",
       "1149         16288  ERG_00375   VII       847050  ERG1  synonymous_variant   \n",
       "1150         16288  ERG_00375   VII       847050  ERG1  synonymous_variant   \n",
       "1151         16288  ERG_00375   VII       847050  ERG1  synonymous_variant   \n",
       "1152         16298  ERG_00376   VII       847050  ERG1    missense_variant   \n",
       "1153         16298  ERG_00376   VII       847050  ERG1    missense_variant   \n",
       "1154         16298  ERG_00376   VII       847050  ERG1    missense_variant   \n",
       "1155         16298  ERG_00376   VII       847050  ERG1    missense_variant   \n",
       "1156         16298  ERG_00376   VII       847050  ERG1    missense_variant   \n",
       "1157         16298  ERG_00376   VII       847050  ERG1    missense_variant   \n",
       "\n",
       "     ALT CDS_position Protein_position Amino_acids  DISTANCE  Unnamed: 0_y  \\\n",
       "1146   T         1374              458           L         0          3824   \n",
       "1147   T         1374              458           L         0          3825   \n",
       "1148   T         1374              458           L         0          3826   \n",
       "1149   T         1374              458           L         0          3827   \n",
       "1150   T         1374              458           L         0          3828   \n",
       "1151   T         1374              458           L         0          3829   \n",
       "1152   A         1374              458         L/F         0          3830   \n",
       "1153   A         1374              458         L/F         0          3831   \n",
       "1154   A         1374              458         L/F         0          3832   \n",
       "1155   A         1374              458         L/F         0          3833   \n",
       "1156   A         1374              458         L/F         0          3834   \n",
       "1157   A         1374              458         L/F         0          3835   \n",
       "\n",
       "     condition      coef   std err          t      pval  Conf. Int. Low  \\\n",
       "1146      CAFF  0.000210  0.002225   0.094274  0.924892       -0.004152   \n",
       "1147      COCL -0.005909  0.006614  -0.893530  0.371576       -0.018872   \n",
       "1148       LOV -0.000577  0.002960  -0.194876  0.845490       -0.006379   \n",
       "1149      NACL -0.003936  0.002384  -1.651079  0.098726       -0.008609   \n",
       "1150        SC -0.001373  0.002327  -0.589969  0.555213       -0.005934   \n",
       "1151       TBF -0.000805  0.001966  -0.409466  0.682198       -0.004658   \n",
       "1152      CAFF  0.001629  0.002467   0.660253  0.509093       -0.003206   \n",
       "1153      COCL -0.015084  0.006839  -2.205619  0.027413       -0.028488   \n",
       "1154       LOV  0.007815  0.002675   2.921551  0.003484        0.002572   \n",
       "1155      NACL  0.002963  0.002400   1.234558  0.216998       -0.001741   \n",
       "1156        SC -0.001063  0.002404  -0.442296  0.658276       -0.005775   \n",
       "1157       TBF  0.085077  0.000861  98.760958  0.000000        0.083388   \n",
       "\n",
       "      Conf. Int. Upp.      padj  \n",
       "1146         0.004571  0.987177  \n",
       "1147         0.007053  0.743566  \n",
       "1148         0.005226  0.967488  \n",
       "1149         0.000736  0.421828  \n",
       "1150         0.003188  0.848076  \n",
       "1151         0.003048  0.904974  \n",
       "1152         0.006464  0.828994  \n",
       "1153        -0.001680  0.218162  \n",
       "1154         0.013058  0.050934  \n",
       "1155         0.007667  0.599014  \n",
       "1156         0.003649  0.896217  \n",
       "1157         0.086765  0.000000  "
      ]
     },
     "execution_count": 7,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "#ERG1 missense variant with a significant fitness effect in terbinafine,\n",
    "#synonymous variant in same site has no significant effect\n",
    "\n",
    "chrom = 'VII'\n",
    "pos = 847050\n",
    "\n",
    "annotated_fits[(annotated_fits['chrom']==chrom)&(annotated_fits['SNP_chr_pos']==pos)]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 8,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Unnamed: 0_x</th>\n",
       "      <th>var_id</th>\n",
       "      <th>chrom</th>\n",
       "      <th>SNP_chr_pos</th>\n",
       "      <th>Gene</th>\n",
       "      <th>Consequence</th>\n",
       "      <th>ALT</th>\n",
       "      <th>CDS_position</th>\n",
       "      <th>Protein_position</th>\n",
       "      <th>Amino_acids</th>\n",
       "      <th>DISTANCE</th>\n",
       "      <th>Unnamed: 0_y</th>\n",
       "      <th>condition</th>\n",
       "      <th>coef</th>\n",
       "      <th>std err</th>\n",
       "      <th>t</th>\n",
       "      <th>pval</th>\n",
       "      <th>Conf. Int. Low</th>\n",
       "      <th>Conf. Int. Upp.</th>\n",
       "      <th>padj</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>2575</th>\n",
       "      <td>22187</td>\n",
       "      <td>EGB_00021</td>\n",
       "      <td>VIII</td>\n",
       "      <td>239187</td>\n",
       "      <td>ERG7</td>\n",
       "      <td>missense_variant</td>\n",
       "      <td>G</td>\n",
       "      <td>90</td>\n",
       "      <td>30</td>\n",
       "      <td>S/R</td>\n",
       "      <td>0</td>\n",
       "      <td>1079</td>\n",
       "      <td>CAFF</td>\n",
       "      <td>0.001218</td>\n",
       "      <td>0.000800</td>\n",
       "      <td>1.522078</td>\n",
       "      <td>1.279929e-01</td>\n",
       "      <td>-0.000351</td>\n",
       "      <td>0.002787</td>\n",
       "      <td>4.777514e-01</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2576</th>\n",
       "      <td>22187</td>\n",
       "      <td>EGB_00021</td>\n",
       "      <td>VIII</td>\n",
       "      <td>239187</td>\n",
       "      <td>ERG7</td>\n",
       "      <td>missense_variant</td>\n",
       "      <td>G</td>\n",
       "      <td>90</td>\n",
       "      <td>30</td>\n",
       "      <td>S/R</td>\n",
       "      <td>0</td>\n",
       "      <td>1080</td>\n",
       "      <td>COCL</td>\n",
       "      <td>0.035477</td>\n",
       "      <td>0.002511</td>\n",
       "      <td>14.130287</td>\n",
       "      <td>2.739628e-45</td>\n",
       "      <td>0.030556</td>\n",
       "      <td>0.040398</td>\n",
       "      <td>1.906484e-43</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2577</th>\n",
       "      <td>22187</td>\n",
       "      <td>EGB_00021</td>\n",
       "      <td>VIII</td>\n",
       "      <td>239187</td>\n",
       "      <td>ERG7</td>\n",
       "      <td>missense_variant</td>\n",
       "      <td>G</td>\n",
       "      <td>90</td>\n",
       "      <td>30</td>\n",
       "      <td>S/R</td>\n",
       "      <td>0</td>\n",
       "      <td>1081</td>\n",
       "      <td>LOV</td>\n",
       "      <td>0.000653</td>\n",
       "      <td>0.001091</td>\n",
       "      <td>0.598598</td>\n",
       "      <td>5.494424e-01</td>\n",
       "      <td>-0.001485</td>\n",
       "      <td>0.002791</td>\n",
       "      <td>8.454451e-01</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2578</th>\n",
       "      <td>22187</td>\n",
       "      <td>EGB_00021</td>\n",
       "      <td>VIII</td>\n",
       "      <td>239187</td>\n",
       "      <td>ERG7</td>\n",
       "      <td>missense_variant</td>\n",
       "      <td>G</td>\n",
       "      <td>90</td>\n",
       "      <td>30</td>\n",
       "      <td>S/R</td>\n",
       "      <td>0</td>\n",
       "      <td>1082</td>\n",
       "      <td>NACL</td>\n",
       "      <td>0.001682</td>\n",
       "      <td>0.000783</td>\n",
       "      <td>2.147460</td>\n",
       "      <td>3.175912e-02</td>\n",
       "      <td>0.000147</td>\n",
       "      <td>0.003218</td>\n",
       "      <td>2.373869e-01</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2579</th>\n",
       "      <td>22187</td>\n",
       "      <td>EGB_00021</td>\n",
       "      <td>VIII</td>\n",
       "      <td>239187</td>\n",
       "      <td>ERG7</td>\n",
       "      <td>missense_variant</td>\n",
       "      <td>G</td>\n",
       "      <td>90</td>\n",
       "      <td>30</td>\n",
       "      <td>S/R</td>\n",
       "      <td>0</td>\n",
       "      <td>1083</td>\n",
       "      <td>SC</td>\n",
       "      <td>-0.001976</td>\n",
       "      <td>0.000770</td>\n",
       "      <td>-2.564535</td>\n",
       "      <td>1.033290e-02</td>\n",
       "      <td>-0.003485</td>\n",
       "      <td>-0.000466</td>\n",
       "      <td>1.148001e-01</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2580</th>\n",
       "      <td>22187</td>\n",
       "      <td>EGB_00021</td>\n",
       "      <td>VIII</td>\n",
       "      <td>239187</td>\n",
       "      <td>ERG7</td>\n",
       "      <td>missense_variant</td>\n",
       "      <td>G</td>\n",
       "      <td>90</td>\n",
       "      <td>30</td>\n",
       "      <td>S/R</td>\n",
       "      <td>0</td>\n",
       "      <td>1084</td>\n",
       "      <td>TBF</td>\n",
       "      <td>0.000183</td>\n",
       "      <td>0.000588</td>\n",
       "      <td>0.311725</td>\n",
       "      <td>7.552502e-01</td>\n",
       "      <td>-0.000970</td>\n",
       "      <td>0.001337</td>\n",
       "      <td>9.325007e-01</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2581</th>\n",
       "      <td>22198</td>\n",
       "      <td>EGB_00022</td>\n",
       "      <td>VIII</td>\n",
       "      <td>239187</td>\n",
       "      <td>ERG7</td>\n",
       "      <td>synonymous_variant</td>\n",
       "      <td>T</td>\n",
       "      <td>90</td>\n",
       "      <td>30</td>\n",
       "      <td>S</td>\n",
       "      <td>0</td>\n",
       "      <td>1085</td>\n",
       "      <td>CAFF</td>\n",
       "      <td>0.000766</td>\n",
       "      <td>0.000896</td>\n",
       "      <td>0.854975</td>\n",
       "      <td>3.925672e-01</td>\n",
       "      <td>-0.000990</td>\n",
       "      <td>0.002521</td>\n",
       "      <td>7.579107e-01</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2582</th>\n",
       "      <td>22198</td>\n",
       "      <td>EGB_00022</td>\n",
       "      <td>VIII</td>\n",
       "      <td>239187</td>\n",
       "      <td>ERG7</td>\n",
       "      <td>synonymous_variant</td>\n",
       "      <td>T</td>\n",
       "      <td>90</td>\n",
       "      <td>30</td>\n",
       "      <td>S</td>\n",
       "      <td>0</td>\n",
       "      <td>1086</td>\n",
       "      <td>COCL</td>\n",
       "      <td>0.007269</td>\n",
       "      <td>0.003179</td>\n",
       "      <td>2.286125</td>\n",
       "      <td>2.224909e-02</td>\n",
       "      <td>0.001037</td>\n",
       "      <td>0.013500</td>\n",
       "      <td>1.927896e-01</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2583</th>\n",
       "      <td>22198</td>\n",
       "      <td>EGB_00022</td>\n",
       "      <td>VIII</td>\n",
       "      <td>239187</td>\n",
       "      <td>ERG7</td>\n",
       "      <td>synonymous_variant</td>\n",
       "      <td>T</td>\n",
       "      <td>90</td>\n",
       "      <td>30</td>\n",
       "      <td>S</td>\n",
       "      <td>0</td>\n",
       "      <td>1087</td>\n",
       "      <td>LOV</td>\n",
       "      <td>0.001486</td>\n",
       "      <td>0.001135</td>\n",
       "      <td>1.309072</td>\n",
       "      <td>1.905130e-01</td>\n",
       "      <td>-0.000739</td>\n",
       "      <td>0.003712</td>\n",
       "      <td>5.644341e-01</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2584</th>\n",
       "      <td>22198</td>\n",
       "      <td>EGB_00022</td>\n",
       "      <td>VIII</td>\n",
       "      <td>239187</td>\n",
       "      <td>ERG7</td>\n",
       "      <td>synonymous_variant</td>\n",
       "      <td>T</td>\n",
       "      <td>90</td>\n",
       "      <td>30</td>\n",
       "      <td>S</td>\n",
       "      <td>0</td>\n",
       "      <td>1088</td>\n",
       "      <td>NACL</td>\n",
       "      <td>0.000711</td>\n",
       "      <td>0.000895</td>\n",
       "      <td>0.794056</td>\n",
       "      <td>4.271646e-01</td>\n",
       "      <td>-0.001044</td>\n",
       "      <td>0.002465</td>\n",
       "      <td>7.823484e-01</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2585</th>\n",
       "      <td>22198</td>\n",
       "      <td>EGB_00022</td>\n",
       "      <td>VIII</td>\n",
       "      <td>239187</td>\n",
       "      <td>ERG7</td>\n",
       "      <td>synonymous_variant</td>\n",
       "      <td>T</td>\n",
       "      <td>90</td>\n",
       "      <td>30</td>\n",
       "      <td>S</td>\n",
       "      <td>0</td>\n",
       "      <td>1089</td>\n",
       "      <td>SC</td>\n",
       "      <td>-0.001180</td>\n",
       "      <td>0.000878</td>\n",
       "      <td>-1.343732</td>\n",
       "      <td>1.790381e-01</td>\n",
       "      <td>-0.002902</td>\n",
       "      <td>0.000541</td>\n",
       "      <td>5.504194e-01</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2586</th>\n",
       "      <td>22198</td>\n",
       "      <td>EGB_00022</td>\n",
       "      <td>VIII</td>\n",
       "      <td>239187</td>\n",
       "      <td>ERG7</td>\n",
       "      <td>synonymous_variant</td>\n",
       "      <td>T</td>\n",
       "      <td>90</td>\n",
       "      <td>30</td>\n",
       "      <td>S</td>\n",
       "      <td>0</td>\n",
       "      <td>1090</td>\n",
       "      <td>TBF</td>\n",
       "      <td>-0.000294</td>\n",
       "      <td>0.000671</td>\n",
       "      <td>-0.438439</td>\n",
       "      <td>6.610690e-01</td>\n",
       "      <td>-0.001610</td>\n",
       "      <td>0.001021</td>\n",
       "      <td>8.973865e-01</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "      Unnamed: 0_x     var_id chrom  SNP_chr_pos  Gene         Consequence  \\\n",
       "2575         22187  EGB_00021  VIII       239187  ERG7    missense_variant   \n",
       "2576         22187  EGB_00021  VIII       239187  ERG7    missense_variant   \n",
       "2577         22187  EGB_00021  VIII       239187  ERG7    missense_variant   \n",
       "2578         22187  EGB_00021  VIII       239187  ERG7    missense_variant   \n",
       "2579         22187  EGB_00021  VIII       239187  ERG7    missense_variant   \n",
       "2580         22187  EGB_00021  VIII       239187  ERG7    missense_variant   \n",
       "2581         22198  EGB_00022  VIII       239187  ERG7  synonymous_variant   \n",
       "2582         22198  EGB_00022  VIII       239187  ERG7  synonymous_variant   \n",
       "2583         22198  EGB_00022  VIII       239187  ERG7  synonymous_variant   \n",
       "2584         22198  EGB_00022  VIII       239187  ERG7  synonymous_variant   \n",
       "2585         22198  EGB_00022  VIII       239187  ERG7  synonymous_variant   \n",
       "2586         22198  EGB_00022  VIII       239187  ERG7  synonymous_variant   \n",
       "\n",
       "     ALT CDS_position Protein_position Amino_acids  DISTANCE  Unnamed: 0_y  \\\n",
       "2575   G           90               30         S/R         0          1079   \n",
       "2576   G           90               30         S/R         0          1080   \n",
       "2577   G           90               30         S/R         0          1081   \n",
       "2578   G           90               30         S/R         0          1082   \n",
       "2579   G           90               30         S/R         0          1083   \n",
       "2580   G           90               30         S/R         0          1084   \n",
       "2581   T           90               30           S         0          1085   \n",
       "2582   T           90               30           S         0          1086   \n",
       "2583   T           90               30           S         0          1087   \n",
       "2584   T           90               30           S         0          1088   \n",
       "2585   T           90               30           S         0          1089   \n",
       "2586   T           90               30           S         0          1090   \n",
       "\n",
       "     condition      coef   std err          t          pval  Conf. Int. Low  \\\n",
       "2575      CAFF  0.001218  0.000800   1.522078  1.279929e-01       -0.000351   \n",
       "2576      COCL  0.035477  0.002511  14.130287  2.739628e-45        0.030556   \n",
       "2577       LOV  0.000653  0.001091   0.598598  5.494424e-01       -0.001485   \n",
       "2578      NACL  0.001682  0.000783   2.147460  3.175912e-02        0.000147   \n",
       "2579        SC -0.001976  0.000770  -2.564535  1.033290e-02       -0.003485   \n",
       "2580       TBF  0.000183  0.000588   0.311725  7.552502e-01       -0.000970   \n",
       "2581      CAFF  0.000766  0.000896   0.854975  3.925672e-01       -0.000990   \n",
       "2582      COCL  0.007269  0.003179   2.286125  2.224909e-02        0.001037   \n",
       "2583       LOV  0.001486  0.001135   1.309072  1.905130e-01       -0.000739   \n",
       "2584      NACL  0.000711  0.000895   0.794056  4.271646e-01       -0.001044   \n",
       "2585        SC -0.001180  0.000878  -1.343732  1.790381e-01       -0.002902   \n",
       "2586       TBF -0.000294  0.000671  -0.438439  6.610690e-01       -0.001610   \n",
       "\n",
       "      Conf. Int. Upp.          padj  \n",
       "2575         0.002787  4.777514e-01  \n",
       "2576         0.040398  1.906484e-43  \n",
       "2577         0.002791  8.454451e-01  \n",
       "2578         0.003218  2.373869e-01  \n",
       "2579        -0.000466  1.148001e-01  \n",
       "2580         0.001337  9.325007e-01  \n",
       "2581         0.002521  7.579107e-01  \n",
       "2582         0.013500  1.927896e-01  \n",
       "2583         0.003712  5.644341e-01  \n",
       "2584         0.002465  7.823484e-01  \n",
       "2585         0.000541  5.504194e-01  \n",
       "2586         0.001021  8.973865e-01  "
      ]
     },
     "execution_count": 8,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "#ERG7 missense variant with a significant fitness effect in cobalt chloride,\n",
    "#synonymous variant in same site has no significant effect\n",
    "\n",
    "chrom = 'VIII'\n",
    "pos = 239187\n",
    "\n",
    "annotated_fits[(annotated_fits['chrom']==chrom)&(annotated_fits['SNP_chr_pos']==pos)]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 9,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Unnamed: 0_x</th>\n",
       "      <th>var_id</th>\n",
       "      <th>chrom</th>\n",
       "      <th>SNP_chr_pos</th>\n",
       "      <th>Gene</th>\n",
       "      <th>Consequence</th>\n",
       "      <th>ALT</th>\n",
       "      <th>CDS_position</th>\n",
       "      <th>Protein_position</th>\n",
       "      <th>Amino_acids</th>\n",
       "      <th>DISTANCE</th>\n",
       "      <th>Unnamed: 0_y</th>\n",
       "      <th>condition</th>\n",
       "      <th>coef</th>\n",
       "      <th>std err</th>\n",
       "      <th>t</th>\n",
       "      <th>pval</th>\n",
       "      <th>Conf. Int. Low</th>\n",
       "      <th>Conf. Int. Upp.</th>\n",
       "      <th>padj</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>4808</th>\n",
       "      <td>31980</td>\n",
       "      <td>EGD_00085</td>\n",
       "      <td>XII</td>\n",
       "      <td>1034989</td>\n",
       "      <td>HMG2</td>\n",
       "      <td>missense_variant</td>\n",
       "      <td>T</td>\n",
       "      <td>2363</td>\n",
       "      <td>788</td>\n",
       "      <td>C/F</td>\n",
       "      <td>0</td>\n",
       "      <td>2006</td>\n",
       "      <td>CAFF</td>\n",
       "      <td>0.001182</td>\n",
       "      <td>0.000453</td>\n",
       "      <td>2.610618</td>\n",
       "      <td>9.039253e-03</td>\n",
       "      <td>0.000295</td>\n",
       "      <td>0.002069</td>\n",
       "      <td>1.060724e-01</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4809</th>\n",
       "      <td>31980</td>\n",
       "      <td>EGD_00085</td>\n",
       "      <td>XII</td>\n",
       "      <td>1034989</td>\n",
       "      <td>HMG2</td>\n",
       "      <td>missense_variant</td>\n",
       "      <td>T</td>\n",
       "      <td>2363</td>\n",
       "      <td>788</td>\n",
       "      <td>C/F</td>\n",
       "      <td>0</td>\n",
       "      <td>2007</td>\n",
       "      <td>COCL</td>\n",
       "      <td>0.030071</td>\n",
       "      <td>0.001353</td>\n",
       "      <td>22.224510</td>\n",
       "      <td>3.721192e-109</td>\n",
       "      <td>0.027419</td>\n",
       "      <td>0.032723</td>\n",
       "      <td>3.884318e-107</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4810</th>\n",
       "      <td>31980</td>\n",
       "      <td>EGD_00085</td>\n",
       "      <td>XII</td>\n",
       "      <td>1034989</td>\n",
       "      <td>HMG2</td>\n",
       "      <td>missense_variant</td>\n",
       "      <td>T</td>\n",
       "      <td>2363</td>\n",
       "      <td>788</td>\n",
       "      <td>C/F</td>\n",
       "      <td>0</td>\n",
       "      <td>2008</td>\n",
       "      <td>LOV</td>\n",
       "      <td>-0.024343</td>\n",
       "      <td>0.000696</td>\n",
       "      <td>-34.994723</td>\n",
       "      <td>1.231939e-266</td>\n",
       "      <td>-0.025706</td>\n",
       "      <td>-0.022979</td>\n",
       "      <td>2.457581e-264</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4811</th>\n",
       "      <td>31980</td>\n",
       "      <td>EGD_00085</td>\n",
       "      <td>XII</td>\n",
       "      <td>1034989</td>\n",
       "      <td>HMG2</td>\n",
       "      <td>missense_variant</td>\n",
       "      <td>T</td>\n",
       "      <td>2363</td>\n",
       "      <td>788</td>\n",
       "      <td>C/F</td>\n",
       "      <td>0</td>\n",
       "      <td>2009</td>\n",
       "      <td>NACL</td>\n",
       "      <td>-0.001828</td>\n",
       "      <td>0.000429</td>\n",
       "      <td>-4.259399</td>\n",
       "      <td>2.051685e-05</td>\n",
       "      <td>-0.002670</td>\n",
       "      <td>-0.000987</td>\n",
       "      <td>5.360108e-04</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4812</th>\n",
       "      <td>31980</td>\n",
       "      <td>EGD_00085</td>\n",
       "      <td>XII</td>\n",
       "      <td>1034989</td>\n",
       "      <td>HMG2</td>\n",
       "      <td>missense_variant</td>\n",
       "      <td>T</td>\n",
       "      <td>2363</td>\n",
       "      <td>788</td>\n",
       "      <td>C/F</td>\n",
       "      <td>0</td>\n",
       "      <td>2010</td>\n",
       "      <td>SC</td>\n",
       "      <td>0.001595</td>\n",
       "      <td>0.000427</td>\n",
       "      <td>3.732065</td>\n",
       "      <td>1.900243e-04</td>\n",
       "      <td>0.000758</td>\n",
       "      <td>0.002433</td>\n",
       "      <td>4.264621e-03</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4813</th>\n",
       "      <td>31980</td>\n",
       "      <td>EGD_00085</td>\n",
       "      <td>XII</td>\n",
       "      <td>1034989</td>\n",
       "      <td>HMG2</td>\n",
       "      <td>missense_variant</td>\n",
       "      <td>T</td>\n",
       "      <td>2363</td>\n",
       "      <td>788</td>\n",
       "      <td>C/F</td>\n",
       "      <td>0</td>\n",
       "      <td>2011</td>\n",
       "      <td>TBF</td>\n",
       "      <td>0.000238</td>\n",
       "      <td>0.000323</td>\n",
       "      <td>0.736919</td>\n",
       "      <td>4.611735e-01</td>\n",
       "      <td>-0.000396</td>\n",
       "      <td>0.000872</td>\n",
       "      <td>8.024723e-01</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4814</th>\n",
       "      <td>31984</td>\n",
       "      <td>EGD_00086</td>\n",
       "      <td>XII</td>\n",
       "      <td>1034989</td>\n",
       "      <td>HMG2</td>\n",
       "      <td>missense_variant</td>\n",
       "      <td>A</td>\n",
       "      <td>2363</td>\n",
       "      <td>788</td>\n",
       "      <td>C/Y</td>\n",
       "      <td>0</td>\n",
       "      <td>2012</td>\n",
       "      <td>CAFF</td>\n",
       "      <td>0.001369</td>\n",
       "      <td>0.000514</td>\n",
       "      <td>2.662458</td>\n",
       "      <td>7.758486e-03</td>\n",
       "      <td>0.000361</td>\n",
       "      <td>0.002377</td>\n",
       "      <td>9.488819e-02</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4815</th>\n",
       "      <td>31984</td>\n",
       "      <td>EGD_00086</td>\n",
       "      <td>XII</td>\n",
       "      <td>1034989</td>\n",
       "      <td>HMG2</td>\n",
       "      <td>missense_variant</td>\n",
       "      <td>A</td>\n",
       "      <td>2363</td>\n",
       "      <td>788</td>\n",
       "      <td>C/Y</td>\n",
       "      <td>0</td>\n",
       "      <td>2013</td>\n",
       "      <td>COCL</td>\n",
       "      <td>0.030950</td>\n",
       "      <td>0.001555</td>\n",
       "      <td>19.909303</td>\n",
       "      <td>5.060471e-88</td>\n",
       "      <td>0.027903</td>\n",
       "      <td>0.033997</td>\n",
       "      <td>4.683284e-86</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4816</th>\n",
       "      <td>31984</td>\n",
       "      <td>EGD_00086</td>\n",
       "      <td>XII</td>\n",
       "      <td>1034989</td>\n",
       "      <td>HMG2</td>\n",
       "      <td>missense_variant</td>\n",
       "      <td>A</td>\n",
       "      <td>2363</td>\n",
       "      <td>788</td>\n",
       "      <td>C/Y</td>\n",
       "      <td>0</td>\n",
       "      <td>2014</td>\n",
       "      <td>LOV</td>\n",
       "      <td>-0.044398</td>\n",
       "      <td>0.000955</td>\n",
       "      <td>-46.485171</td>\n",
       "      <td>0.000000e+00</td>\n",
       "      <td>-0.046270</td>\n",
       "      <td>-0.042526</td>\n",
       "      <td>0.000000e+00</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4817</th>\n",
       "      <td>31984</td>\n",
       "      <td>EGD_00086</td>\n",
       "      <td>XII</td>\n",
       "      <td>1034989</td>\n",
       "      <td>HMG2</td>\n",
       "      <td>missense_variant</td>\n",
       "      <td>A</td>\n",
       "      <td>2363</td>\n",
       "      <td>788</td>\n",
       "      <td>C/Y</td>\n",
       "      <td>0</td>\n",
       "      <td>2015</td>\n",
       "      <td>NACL</td>\n",
       "      <td>-0.003589</td>\n",
       "      <td>0.000494</td>\n",
       "      <td>-7.266212</td>\n",
       "      <td>3.724574e-13</td>\n",
       "      <td>-0.004558</td>\n",
       "      <td>-0.002621</td>\n",
       "      <td>1.519795e-11</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4818</th>\n",
       "      <td>31984</td>\n",
       "      <td>EGD_00086</td>\n",
       "      <td>XII</td>\n",
       "      <td>1034989</td>\n",
       "      <td>HMG2</td>\n",
       "      <td>missense_variant</td>\n",
       "      <td>A</td>\n",
       "      <td>2363</td>\n",
       "      <td>788</td>\n",
       "      <td>C/Y</td>\n",
       "      <td>0</td>\n",
       "      <td>2016</td>\n",
       "      <td>SC</td>\n",
       "      <td>0.000463</td>\n",
       "      <td>0.000483</td>\n",
       "      <td>0.958521</td>\n",
       "      <td>3.378023e-01</td>\n",
       "      <td>-0.000484</td>\n",
       "      <td>0.001411</td>\n",
       "      <td>7.146495e-01</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4819</th>\n",
       "      <td>31984</td>\n",
       "      <td>EGD_00086</td>\n",
       "      <td>XII</td>\n",
       "      <td>1034989</td>\n",
       "      <td>HMG2</td>\n",
       "      <td>missense_variant</td>\n",
       "      <td>A</td>\n",
       "      <td>2363</td>\n",
       "      <td>788</td>\n",
       "      <td>C/Y</td>\n",
       "      <td>0</td>\n",
       "      <td>2017</td>\n",
       "      <td>TBF</td>\n",
       "      <td>-0.000011</td>\n",
       "      <td>0.000362</td>\n",
       "      <td>-0.029928</td>\n",
       "      <td>9.761249e-01</td>\n",
       "      <td>-0.000720</td>\n",
       "      <td>0.000698</td>\n",
       "      <td>9.975721e-01</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "      Unnamed: 0_x     var_id chrom  SNP_chr_pos  Gene       Consequence ALT  \\\n",
       "4808         31980  EGD_00085   XII      1034989  HMG2  missense_variant   T   \n",
       "4809         31980  EGD_00085   XII      1034989  HMG2  missense_variant   T   \n",
       "4810         31980  EGD_00085   XII      1034989  HMG2  missense_variant   T   \n",
       "4811         31980  EGD_00085   XII      1034989  HMG2  missense_variant   T   \n",
       "4812         31980  EGD_00085   XII      1034989  HMG2  missense_variant   T   \n",
       "4813         31980  EGD_00085   XII      1034989  HMG2  missense_variant   T   \n",
       "4814         31984  EGD_00086   XII      1034989  HMG2  missense_variant   A   \n",
       "4815         31984  EGD_00086   XII      1034989  HMG2  missense_variant   A   \n",
       "4816         31984  EGD_00086   XII      1034989  HMG2  missense_variant   A   \n",
       "4817         31984  EGD_00086   XII      1034989  HMG2  missense_variant   A   \n",
       "4818         31984  EGD_00086   XII      1034989  HMG2  missense_variant   A   \n",
       "4819         31984  EGD_00086   XII      1034989  HMG2  missense_variant   A   \n",
       "\n",
       "     CDS_position Protein_position Amino_acids  DISTANCE  Unnamed: 0_y  \\\n",
       "4808         2363              788         C/F         0          2006   \n",
       "4809         2363              788         C/F         0          2007   \n",
       "4810         2363              788         C/F         0          2008   \n",
       "4811         2363              788         C/F         0          2009   \n",
       "4812         2363              788         C/F         0          2010   \n",
       "4813         2363              788         C/F         0          2011   \n",
       "4814         2363              788         C/Y         0          2012   \n",
       "4815         2363              788         C/Y         0          2013   \n",
       "4816         2363              788         C/Y         0          2014   \n",
       "4817         2363              788         C/Y         0          2015   \n",
       "4818         2363              788         C/Y         0          2016   \n",
       "4819         2363              788         C/Y         0          2017   \n",
       "\n",
       "     condition      coef   std err          t           pval  Conf. Int. Low  \\\n",
       "4808      CAFF  0.001182  0.000453   2.610618   9.039253e-03        0.000295   \n",
       "4809      COCL  0.030071  0.001353  22.224510  3.721192e-109        0.027419   \n",
       "4810       LOV -0.024343  0.000696 -34.994723  1.231939e-266       -0.025706   \n",
       "4811      NACL -0.001828  0.000429  -4.259399   2.051685e-05       -0.002670   \n",
       "4812        SC  0.001595  0.000427   3.732065   1.900243e-04        0.000758   \n",
       "4813       TBF  0.000238  0.000323   0.736919   4.611735e-01       -0.000396   \n",
       "4814      CAFF  0.001369  0.000514   2.662458   7.758486e-03        0.000361   \n",
       "4815      COCL  0.030950  0.001555  19.909303   5.060471e-88        0.027903   \n",
       "4816       LOV -0.044398  0.000955 -46.485171   0.000000e+00       -0.046270   \n",
       "4817      NACL -0.003589  0.000494  -7.266212   3.724574e-13       -0.004558   \n",
       "4818        SC  0.000463  0.000483   0.958521   3.378023e-01       -0.000484   \n",
       "4819       TBF -0.000011  0.000362  -0.029928   9.761249e-01       -0.000720   \n",
       "\n",
       "      Conf. Int. Upp.           padj  \n",
       "4808         0.002069   1.060724e-01  \n",
       "4809         0.032723  3.884318e-107  \n",
       "4810        -0.022979  2.457581e-264  \n",
       "4811        -0.000987   5.360108e-04  \n",
       "4812         0.002433   4.264621e-03  \n",
       "4813         0.000872   8.024723e-01  \n",
       "4814         0.002377   9.488819e-02  \n",
       "4815         0.033997   4.683284e-86  \n",
       "4816        -0.042526   0.000000e+00  \n",
       "4817        -0.002621   1.519795e-11  \n",
       "4818         0.001411   7.146495e-01  \n",
       "4819         0.000698   9.975721e-01  "
      ]
     },
     "execution_count": 9,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "#HMG2 site with two missense variants, both have significant positive effects in cobalt chloride, \n",
    "#significant negative effects in lovastatin, and significant negative effects in sodium chloride\n",
    "#the two variants are C>F and C>Y\n",
    "\n",
    "chrom = 'XII'\n",
    "pos = 1034989\n",
    "\n",
    "annotated_fits[(annotated_fits['chrom']==chrom)&(annotated_fits['SNP_chr_pos']==pos)]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 10,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Unnamed: 0_x</th>\n",
       "      <th>var_id</th>\n",
       "      <th>chrom</th>\n",
       "      <th>SNP_chr_pos</th>\n",
       "      <th>Gene</th>\n",
       "      <th>Consequence</th>\n",
       "      <th>ALT</th>\n",
       "      <th>CDS_position</th>\n",
       "      <th>Protein_position</th>\n",
       "      <th>Amino_acids</th>\n",
       "      <th>DISTANCE</th>\n",
       "      <th>Unnamed: 0_y</th>\n",
       "      <th>condition</th>\n",
       "      <th>coef</th>\n",
       "      <th>std err</th>\n",
       "      <th>t</th>\n",
       "      <th>pval</th>\n",
       "      <th>Conf. Int. Low</th>\n",
       "      <th>Conf. Int. Upp.</th>\n",
       "      <th>padj</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>5337</th>\n",
       "      <td>32625</td>\n",
       "      <td>ERG_01754</td>\n",
       "      <td>XIII</td>\n",
       "      <td>115802</td>\n",
       "      <td>HMG1</td>\n",
       "      <td>missense_variant</td>\n",
       "      <td>C</td>\n",
       "      <td>3097</td>\n",
       "      <td>1033</td>\n",
       "      <td>P/A</td>\n",
       "      <td>0</td>\n",
       "      <td>6501</td>\n",
       "      <td>CAFF</td>\n",
       "      <td>0.007017</td>\n",
       "      <td>0.001589</td>\n",
       "      <td>4.414814</td>\n",
       "      <td>1.012054e-05</td>\n",
       "      <td>0.003902</td>\n",
       "      <td>0.010132</td>\n",
       "      <td>2.786875e-04</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5338</th>\n",
       "      <td>32625</td>\n",
       "      <td>ERG_01754</td>\n",
       "      <td>XIII</td>\n",
       "      <td>115802</td>\n",
       "      <td>HMG1</td>\n",
       "      <td>missense_variant</td>\n",
       "      <td>C</td>\n",
       "      <td>3097</td>\n",
       "      <td>1033</td>\n",
       "      <td>P/A</td>\n",
       "      <td>0</td>\n",
       "      <td>6502</td>\n",
       "      <td>COCL</td>\n",
       "      <td>-0.001433</td>\n",
       "      <td>0.004438</td>\n",
       "      <td>-0.322893</td>\n",
       "      <td>7.467769e-01</td>\n",
       "      <td>-0.010131</td>\n",
       "      <td>0.007265</td>\n",
       "      <td>9.288132e-01</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5339</th>\n",
       "      <td>32625</td>\n",
       "      <td>ERG_01754</td>\n",
       "      <td>XIII</td>\n",
       "      <td>115802</td>\n",
       "      <td>HMG1</td>\n",
       "      <td>missense_variant</td>\n",
       "      <td>C</td>\n",
       "      <td>3097</td>\n",
       "      <td>1033</td>\n",
       "      <td>P/A</td>\n",
       "      <td>0</td>\n",
       "      <td>6503</td>\n",
       "      <td>LOV</td>\n",
       "      <td>0.020164</td>\n",
       "      <td>0.001627</td>\n",
       "      <td>12.396164</td>\n",
       "      <td>2.914492e-35</td>\n",
       "      <td>0.016976</td>\n",
       "      <td>0.023352</td>\n",
       "      <td>1.829608e-33</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5340</th>\n",
       "      <td>32625</td>\n",
       "      <td>ERG_01754</td>\n",
       "      <td>XIII</td>\n",
       "      <td>115802</td>\n",
       "      <td>HMG1</td>\n",
       "      <td>missense_variant</td>\n",
       "      <td>C</td>\n",
       "      <td>3097</td>\n",
       "      <td>1033</td>\n",
       "      <td>P/A</td>\n",
       "      <td>0</td>\n",
       "      <td>6504</td>\n",
       "      <td>NACL</td>\n",
       "      <td>0.000143</td>\n",
       "      <td>0.001602</td>\n",
       "      <td>0.089164</td>\n",
       "      <td>9.289519e-01</td>\n",
       "      <td>-0.002996</td>\n",
       "      <td>0.003282</td>\n",
       "      <td>9.873100e-01</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5341</th>\n",
       "      <td>32625</td>\n",
       "      <td>ERG_01754</td>\n",
       "      <td>XIII</td>\n",
       "      <td>115802</td>\n",
       "      <td>HMG1</td>\n",
       "      <td>missense_variant</td>\n",
       "      <td>C</td>\n",
       "      <td>3097</td>\n",
       "      <td>1033</td>\n",
       "      <td>P/A</td>\n",
       "      <td>0</td>\n",
       "      <td>6505</td>\n",
       "      <td>SC</td>\n",
       "      <td>-0.001622</td>\n",
       "      <td>0.001563</td>\n",
       "      <td>-1.037716</td>\n",
       "      <td>2.994048e-01</td>\n",
       "      <td>-0.004684</td>\n",
       "      <td>0.001441</td>\n",
       "      <td>6.807544e-01</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5342</th>\n",
       "      <td>32625</td>\n",
       "      <td>ERG_01754</td>\n",
       "      <td>XIII</td>\n",
       "      <td>115802</td>\n",
       "      <td>HMG1</td>\n",
       "      <td>missense_variant</td>\n",
       "      <td>C</td>\n",
       "      <td>3097</td>\n",
       "      <td>1033</td>\n",
       "      <td>P/A</td>\n",
       "      <td>0</td>\n",
       "      <td>6506</td>\n",
       "      <td>TBF</td>\n",
       "      <td>-0.001642</td>\n",
       "      <td>0.001217</td>\n",
       "      <td>-1.348774</td>\n",
       "      <td>1.774127e-01</td>\n",
       "      <td>-0.004028</td>\n",
       "      <td>0.000744</td>\n",
       "      <td>5.485773e-01</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5343</th>\n",
       "      <td>32631</td>\n",
       "      <td>ERG_01755</td>\n",
       "      <td>XIII</td>\n",
       "      <td>115802</td>\n",
       "      <td>HMG1</td>\n",
       "      <td>missense_variant</td>\n",
       "      <td>T</td>\n",
       "      <td>3097</td>\n",
       "      <td>1033</td>\n",
       "      <td>P/T</td>\n",
       "      <td>0</td>\n",
       "      <td>6507</td>\n",
       "      <td>CAFF</td>\n",
       "      <td>0.002833</td>\n",
       "      <td>0.002365</td>\n",
       "      <td>1.197716</td>\n",
       "      <td>2.310304e-01</td>\n",
       "      <td>-0.001803</td>\n",
       "      <td>0.007468</td>\n",
       "      <td>6.127100e-01</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5344</th>\n",
       "      <td>32631</td>\n",
       "      <td>ERG_01755</td>\n",
       "      <td>XIII</td>\n",
       "      <td>115802</td>\n",
       "      <td>HMG1</td>\n",
       "      <td>missense_variant</td>\n",
       "      <td>T</td>\n",
       "      <td>3097</td>\n",
       "      <td>1033</td>\n",
       "      <td>P/T</td>\n",
       "      <td>0</td>\n",
       "      <td>6508</td>\n",
       "      <td>COCL</td>\n",
       "      <td>-0.005717</td>\n",
       "      <td>0.006576</td>\n",
       "      <td>-0.869412</td>\n",
       "      <td>3.846242e-01</td>\n",
       "      <td>-0.018607</td>\n",
       "      <td>0.007172</td>\n",
       "      <td>7.534335e-01</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5345</th>\n",
       "      <td>32631</td>\n",
       "      <td>ERG_01755</td>\n",
       "      <td>XIII</td>\n",
       "      <td>115802</td>\n",
       "      <td>HMG1</td>\n",
       "      <td>missense_variant</td>\n",
       "      <td>T</td>\n",
       "      <td>3097</td>\n",
       "      <td>1033</td>\n",
       "      <td>P/T</td>\n",
       "      <td>0</td>\n",
       "      <td>6509</td>\n",
       "      <td>LOV</td>\n",
       "      <td>-0.013972</td>\n",
       "      <td>0.003414</td>\n",
       "      <td>-4.093143</td>\n",
       "      <td>4.259076e-05</td>\n",
       "      <td>-0.020662</td>\n",
       "      <td>-0.007282</td>\n",
       "      <td>1.059106e-03</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5346</th>\n",
       "      <td>32631</td>\n",
       "      <td>ERG_01755</td>\n",
       "      <td>XIII</td>\n",
       "      <td>115802</td>\n",
       "      <td>HMG1</td>\n",
       "      <td>missense_variant</td>\n",
       "      <td>T</td>\n",
       "      <td>3097</td>\n",
       "      <td>1033</td>\n",
       "      <td>P/T</td>\n",
       "      <td>0</td>\n",
       "      <td>6510</td>\n",
       "      <td>NACL</td>\n",
       "      <td>0.000968</td>\n",
       "      <td>0.002362</td>\n",
       "      <td>0.409776</td>\n",
       "      <td>6.819715e-01</td>\n",
       "      <td>-0.003662</td>\n",
       "      <td>0.005598</td>\n",
       "      <td>9.049737e-01</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5347</th>\n",
       "      <td>32631</td>\n",
       "      <td>ERG_01755</td>\n",
       "      <td>XIII</td>\n",
       "      <td>115802</td>\n",
       "      <td>HMG1</td>\n",
       "      <td>missense_variant</td>\n",
       "      <td>T</td>\n",
       "      <td>3097</td>\n",
       "      <td>1033</td>\n",
       "      <td>P/T</td>\n",
       "      <td>0</td>\n",
       "      <td>6511</td>\n",
       "      <td>SC</td>\n",
       "      <td>-0.001903</td>\n",
       "      <td>0.002390</td>\n",
       "      <td>-0.796265</td>\n",
       "      <td>4.258802e-01</td>\n",
       "      <td>-0.006588</td>\n",
       "      <td>0.002781</td>\n",
       "      <td>7.819853e-01</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5348</th>\n",
       "      <td>32631</td>\n",
       "      <td>ERG_01755</td>\n",
       "      <td>XIII</td>\n",
       "      <td>115802</td>\n",
       "      <td>HMG1</td>\n",
       "      <td>missense_variant</td>\n",
       "      <td>T</td>\n",
       "      <td>3097</td>\n",
       "      <td>1033</td>\n",
       "      <td>P/T</td>\n",
       "      <td>0</td>\n",
       "      <td>6512</td>\n",
       "      <td>TBF</td>\n",
       "      <td>-0.000377</td>\n",
       "      <td>0.002427</td>\n",
       "      <td>-0.155452</td>\n",
       "      <td>8.764653e-01</td>\n",
       "      <td>-0.005133</td>\n",
       "      <td>0.004379</td>\n",
       "      <td>9.750249e-01</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "      Unnamed: 0_x     var_id chrom  SNP_chr_pos  Gene       Consequence ALT  \\\n",
       "5337         32625  ERG_01754  XIII       115802  HMG1  missense_variant   C   \n",
       "5338         32625  ERG_01754  XIII       115802  HMG1  missense_variant   C   \n",
       "5339         32625  ERG_01754  XIII       115802  HMG1  missense_variant   C   \n",
       "5340         32625  ERG_01754  XIII       115802  HMG1  missense_variant   C   \n",
       "5341         32625  ERG_01754  XIII       115802  HMG1  missense_variant   C   \n",
       "5342         32625  ERG_01754  XIII       115802  HMG1  missense_variant   C   \n",
       "5343         32631  ERG_01755  XIII       115802  HMG1  missense_variant   T   \n",
       "5344         32631  ERG_01755  XIII       115802  HMG1  missense_variant   T   \n",
       "5345         32631  ERG_01755  XIII       115802  HMG1  missense_variant   T   \n",
       "5346         32631  ERG_01755  XIII       115802  HMG1  missense_variant   T   \n",
       "5347         32631  ERG_01755  XIII       115802  HMG1  missense_variant   T   \n",
       "5348         32631  ERG_01755  XIII       115802  HMG1  missense_variant   T   \n",
       "\n",
       "     CDS_position Protein_position Amino_acids  DISTANCE  Unnamed: 0_y  \\\n",
       "5337         3097             1033         P/A         0          6501   \n",
       "5338         3097             1033         P/A         0          6502   \n",
       "5339         3097             1033         P/A         0          6503   \n",
       "5340         3097             1033         P/A         0          6504   \n",
       "5341         3097             1033         P/A         0          6505   \n",
       "5342         3097             1033         P/A         0          6506   \n",
       "5343         3097             1033         P/T         0          6507   \n",
       "5344         3097             1033         P/T         0          6508   \n",
       "5345         3097             1033         P/T         0          6509   \n",
       "5346         3097             1033         P/T         0          6510   \n",
       "5347         3097             1033         P/T         0          6511   \n",
       "5348         3097             1033         P/T         0          6512   \n",
       "\n",
       "     condition      coef   std err          t          pval  Conf. Int. Low  \\\n",
       "5337      CAFF  0.007017  0.001589   4.414814  1.012054e-05        0.003902   \n",
       "5338      COCL -0.001433  0.004438  -0.322893  7.467769e-01       -0.010131   \n",
       "5339       LOV  0.020164  0.001627  12.396164  2.914492e-35        0.016976   \n",
       "5340      NACL  0.000143  0.001602   0.089164  9.289519e-01       -0.002996   \n",
       "5341        SC -0.001622  0.001563  -1.037716  2.994048e-01       -0.004684   \n",
       "5342       TBF -0.001642  0.001217  -1.348774  1.774127e-01       -0.004028   \n",
       "5343      CAFF  0.002833  0.002365   1.197716  2.310304e-01       -0.001803   \n",
       "5344      COCL -0.005717  0.006576  -0.869412  3.846242e-01       -0.018607   \n",
       "5345       LOV -0.013972  0.003414  -4.093143  4.259076e-05       -0.020662   \n",
       "5346      NACL  0.000968  0.002362   0.409776  6.819715e-01       -0.003662   \n",
       "5347        SC -0.001903  0.002390  -0.796265  4.258802e-01       -0.006588   \n",
       "5348       TBF -0.000377  0.002427  -0.155452  8.764653e-01       -0.005133   \n",
       "\n",
       "      Conf. Int. Upp.          padj  \n",
       "5337         0.010132  2.786875e-04  \n",
       "5338         0.007265  9.288132e-01  \n",
       "5339         0.023352  1.829608e-33  \n",
       "5340         0.003282  9.873100e-01  \n",
       "5341         0.001441  6.807544e-01  \n",
       "5342         0.000744  5.485773e-01  \n",
       "5343         0.007468  6.127100e-01  \n",
       "5344         0.007172  7.534335e-01  \n",
       "5345        -0.007282  1.059106e-03  \n",
       "5346         0.005598  9.049737e-01  \n",
       "5347         0.002781  7.819853e-01  \n",
       "5348         0.004379  9.750249e-01  "
      ]
     },
     "execution_count": 10,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "#HMG1 site with two missense variants, both have significant fitness effects in lovastatin,\n",
    "#however one has a positive effect and the other negative.\n",
    "\n",
    "chrom = 'XIII'\n",
    "\n",
    "pos = 115802\n",
    "\n",
    "annotated_fits[(annotated_fits['chrom']==chrom)&(annotated_fits['SNP_chr_pos']==pos)]"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### Synonymous Hits within sites w/ multiple edits"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 11,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Unnamed: 0_x</th>\n",
       "      <th>var_id</th>\n",
       "      <th>chrom</th>\n",
       "      <th>SNP_chr_pos</th>\n",
       "      <th>Gene</th>\n",
       "      <th>Consequence</th>\n",
       "      <th>ALT</th>\n",
       "      <th>CDS_position</th>\n",
       "      <th>Protein_position</th>\n",
       "      <th>Amino_acids</th>\n",
       "      <th>DISTANCE</th>\n",
       "      <th>Unnamed: 0_y</th>\n",
       "      <th>condition</th>\n",
       "      <th>coef</th>\n",
       "      <th>std err</th>\n",
       "      <th>t</th>\n",
       "      <th>pval</th>\n",
       "      <th>Conf. Int. Low</th>\n",
       "      <th>Conf. Int. Upp.</th>\n",
       "      <th>padj</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>2594</th>\n",
       "      <td>22220</td>\n",
       "      <td>EGA_00017</td>\n",
       "      <td>VIII</td>\n",
       "      <td>239202</td>\n",
       "      <td>ERG7</td>\n",
       "      <td>synonymous_variant</td>\n",
       "      <td>A</td>\n",
       "      <td>105</td>\n",
       "      <td>35</td>\n",
       "      <td>T</td>\n",
       "      <td>0</td>\n",
       "      <td>961</td>\n",
       "      <td>COCL</td>\n",
       "      <td>0.011834</td>\n",
       "      <td>0.001867</td>\n",
       "      <td>6.340083</td>\n",
       "      <td>2.306416e-10</td>\n",
       "      <td>0.008176</td>\n",
       "      <td>0.015492</td>\n",
       "      <td>8.591161e-09</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2798</th>\n",
       "      <td>22587</td>\n",
       "      <td>ERG_00825</td>\n",
       "      <td>VIII</td>\n",
       "      <td>239784</td>\n",
       "      <td>ERG7</td>\n",
       "      <td>synonymous_variant</td>\n",
       "      <td>T</td>\n",
       "      <td>687</td>\n",
       "      <td>229</td>\n",
       "      <td>G</td>\n",
       "      <td>0</td>\n",
       "      <td>4824</td>\n",
       "      <td>COCL</td>\n",
       "      <td>0.024358</td>\n",
       "      <td>0.003657</td>\n",
       "      <td>6.660545</td>\n",
       "      <td>2.742551e-11</td>\n",
       "      <td>0.017190</td>\n",
       "      <td>0.031526</td>\n",
       "      <td>1.070430e-09</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2974</th>\n",
       "      <td>22882</td>\n",
       "      <td>EGB_00027</td>\n",
       "      <td>VIII</td>\n",
       "      <td>240738</td>\n",
       "      <td>ERG7</td>\n",
       "      <td>synonymous_variant</td>\n",
       "      <td>C</td>\n",
       "      <td>1641</td>\n",
       "      <td>547</td>\n",
       "      <td>G</td>\n",
       "      <td>0</td>\n",
       "      <td>1110</td>\n",
       "      <td>COCL</td>\n",
       "      <td>0.077358</td>\n",
       "      <td>0.003184</td>\n",
       "      <td>24.293832</td>\n",
       "      <td>5.558787e-130</td>\n",
       "      <td>0.071117</td>\n",
       "      <td>0.083599</td>\n",
       "      <td>6.397594e-128</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3557</th>\n",
       "      <td>24260</td>\n",
       "      <td>EG8_00143</td>\n",
       "      <td>X</td>\n",
       "      <td>105664</td>\n",
       "      <td>ERG20</td>\n",
       "      <td>synonymous_variant</td>\n",
       "      <td>A</td>\n",
       "      <td>651</td>\n",
       "      <td>217</td>\n",
       "      <td>T</td>\n",
       "      <td>0</td>\n",
       "      <td>281</td>\n",
       "      <td>NACL</td>\n",
       "      <td>0.006764</td>\n",
       "      <td>0.000760</td>\n",
       "      <td>8.896273</td>\n",
       "      <td>5.871106e-19</td>\n",
       "      <td>0.005274</td>\n",
       "      <td>0.008254</td>\n",
       "      <td>2.648488e-17</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5186</th>\n",
       "      <td>32424</td>\n",
       "      <td>ERG_01662</td>\n",
       "      <td>XIII</td>\n",
       "      <td>19933</td>\n",
       "      <td>ERG13</td>\n",
       "      <td>synonymous_variant</td>\n",
       "      <td>T</td>\n",
       "      <td>603</td>\n",
       "      <td>201</td>\n",
       "      <td>T</td>\n",
       "      <td>0</td>\n",
       "      <td>6359</td>\n",
       "      <td>CAFF</td>\n",
       "      <td>0.001372</td>\n",
       "      <td>0.000328</td>\n",
       "      <td>4.183220</td>\n",
       "      <td>2.876603e-05</td>\n",
       "      <td>0.000729</td>\n",
       "      <td>0.002015</td>\n",
       "      <td>7.294707e-04</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5190</th>\n",
       "      <td>32424</td>\n",
       "      <td>ERG_01662</td>\n",
       "      <td>XIII</td>\n",
       "      <td>19933</td>\n",
       "      <td>ERG13</td>\n",
       "      <td>synonymous_variant</td>\n",
       "      <td>T</td>\n",
       "      <td>603</td>\n",
       "      <td>201</td>\n",
       "      <td>T</td>\n",
       "      <td>0</td>\n",
       "      <td>6363</td>\n",
       "      <td>SC</td>\n",
       "      <td>-0.001267</td>\n",
       "      <td>0.000305</td>\n",
       "      <td>-4.147841</td>\n",
       "      <td>3.359098e-05</td>\n",
       "      <td>-0.001865</td>\n",
       "      <td>-0.000668</td>\n",
       "      <td>8.446672e-04</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5194</th>\n",
       "      <td>32430</td>\n",
       "      <td>ERG_01663</td>\n",
       "      <td>XIII</td>\n",
       "      <td>19933</td>\n",
       "      <td>ERG13</td>\n",
       "      <td>synonymous_variant</td>\n",
       "      <td>A</td>\n",
       "      <td>603</td>\n",
       "      <td>201</td>\n",
       "      <td>T</td>\n",
       "      <td>0</td>\n",
       "      <td>6367</td>\n",
       "      <td>LOV</td>\n",
       "      <td>0.002975</td>\n",
       "      <td>0.000656</td>\n",
       "      <td>4.535114</td>\n",
       "      <td>5.764072e-06</td>\n",
       "      <td>0.001689</td>\n",
       "      <td>0.004260</td>\n",
       "      <td>1.606959e-04</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5196</th>\n",
       "      <td>32430</td>\n",
       "      <td>ERG_01663</td>\n",
       "      <td>XIII</td>\n",
       "      <td>19933</td>\n",
       "      <td>ERG13</td>\n",
       "      <td>synonymous_variant</td>\n",
       "      <td>A</td>\n",
       "      <td>603</td>\n",
       "      <td>201</td>\n",
       "      <td>T</td>\n",
       "      <td>0</td>\n",
       "      <td>6369</td>\n",
       "      <td>SC</td>\n",
       "      <td>-0.002254</td>\n",
       "      <td>0.000464</td>\n",
       "      <td>-4.858600</td>\n",
       "      <td>1.184014e-06</td>\n",
       "      <td>-0.003164</td>\n",
       "      <td>-0.001345</td>\n",
       "      <td>3.566744e-05</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5728</th>\n",
       "      <td>32913</td>\n",
       "      <td>ERG_01874</td>\n",
       "      <td>XIII</td>\n",
       "      <td>118671</td>\n",
       "      <td>HMG1</td>\n",
       "      <td>synonymous_variant</td>\n",
       "      <td>C</td>\n",
       "      <td>228</td>\n",
       "      <td>76</td>\n",
       "      <td>S</td>\n",
       "      <td>0</td>\n",
       "      <td>6833</td>\n",
       "      <td>LOV</td>\n",
       "      <td>-0.012673</td>\n",
       "      <td>0.002248</td>\n",
       "      <td>-5.637844</td>\n",
       "      <td>1.726664e-08</td>\n",
       "      <td>-0.017079</td>\n",
       "      <td>-0.008267</td>\n",
       "      <td>5.871312e-07</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "      Unnamed: 0_x     var_id chrom  SNP_chr_pos   Gene         Consequence  \\\n",
       "2594         22220  EGA_00017  VIII       239202   ERG7  synonymous_variant   \n",
       "2798         22587  ERG_00825  VIII       239784   ERG7  synonymous_variant   \n",
       "2974         22882  EGB_00027  VIII       240738   ERG7  synonymous_variant   \n",
       "3557         24260  EG8_00143     X       105664  ERG20  synonymous_variant   \n",
       "5186         32424  ERG_01662  XIII        19933  ERG13  synonymous_variant   \n",
       "5190         32424  ERG_01662  XIII        19933  ERG13  synonymous_variant   \n",
       "5194         32430  ERG_01663  XIII        19933  ERG13  synonymous_variant   \n",
       "5196         32430  ERG_01663  XIII        19933  ERG13  synonymous_variant   \n",
       "5728         32913  ERG_01874  XIII       118671   HMG1  synonymous_variant   \n",
       "\n",
       "     ALT CDS_position Protein_position Amino_acids  DISTANCE  Unnamed: 0_y  \\\n",
       "2594   A          105               35           T         0           961   \n",
       "2798   T          687              229           G         0          4824   \n",
       "2974   C         1641              547           G         0          1110   \n",
       "3557   A          651              217           T         0           281   \n",
       "5186   T          603              201           T         0          6359   \n",
       "5190   T          603              201           T         0          6363   \n",
       "5194   A          603              201           T         0          6367   \n",
       "5196   A          603              201           T         0          6369   \n",
       "5728   C          228               76           S         0          6833   \n",
       "\n",
       "     condition      coef   std err          t           pval  Conf. Int. Low  \\\n",
       "2594      COCL  0.011834  0.001867   6.340083   2.306416e-10        0.008176   \n",
       "2798      COCL  0.024358  0.003657   6.660545   2.742551e-11        0.017190   \n",
       "2974      COCL  0.077358  0.003184  24.293832  5.558787e-130        0.071117   \n",
       "3557      NACL  0.006764  0.000760   8.896273   5.871106e-19        0.005274   \n",
       "5186      CAFF  0.001372  0.000328   4.183220   2.876603e-05        0.000729   \n",
       "5190        SC -0.001267  0.000305  -4.147841   3.359098e-05       -0.001865   \n",
       "5194       LOV  0.002975  0.000656   4.535114   5.764072e-06        0.001689   \n",
       "5196        SC -0.002254  0.000464  -4.858600   1.184014e-06       -0.003164   \n",
       "5728       LOV -0.012673  0.002248  -5.637844   1.726664e-08       -0.017079   \n",
       "\n",
       "      Conf. Int. Upp.           padj  \n",
       "2594         0.015492   8.591161e-09  \n",
       "2798         0.031526   1.070430e-09  \n",
       "2974         0.083599  6.397594e-128  \n",
       "3557         0.008254   2.648488e-17  \n",
       "5186         0.002015   7.294707e-04  \n",
       "5190        -0.000668   8.446672e-04  \n",
       "5194         0.004260   1.606959e-04  \n",
       "5196        -0.001345   3.566744e-05  \n",
       "5728        -0.008267   5.871312e-07  "
      ]
     },
     "execution_count": 11,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "#displaying all synonymous variants which are hits and target a site with multiple edits \n",
    "annotated_fits[(annotated_fits['var_id'].isin(multi_edits))&(annotated_fits['padj']<.01)&(annotated_fits['Consequence']=='synonymous_variant')]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 12,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Unnamed: 0_x</th>\n",
       "      <th>var_id</th>\n",
       "      <th>chrom</th>\n",
       "      <th>SNP_chr_pos</th>\n",
       "      <th>Gene</th>\n",
       "      <th>Consequence</th>\n",
       "      <th>ALT</th>\n",
       "      <th>CDS_position</th>\n",
       "      <th>Protein_position</th>\n",
       "      <th>Amino_acids</th>\n",
       "      <th>DISTANCE</th>\n",
       "      <th>Unnamed: 0_y</th>\n",
       "      <th>condition</th>\n",
       "      <th>coef</th>\n",
       "      <th>std err</th>\n",
       "      <th>t</th>\n",
       "      <th>pval</th>\n",
       "      <th>Conf. Int. Low</th>\n",
       "      <th>Conf. Int. Upp.</th>\n",
       "      <th>padj</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>5726</th>\n",
       "      <td>32913</td>\n",
       "      <td>ERG_01874</td>\n",
       "      <td>XIII</td>\n",
       "      <td>118671</td>\n",
       "      <td>HMG1</td>\n",
       "      <td>synonymous_variant</td>\n",
       "      <td>C</td>\n",
       "      <td>228</td>\n",
       "      <td>76</td>\n",
       "      <td>S</td>\n",
       "      <td>0</td>\n",
       "      <td>6831</td>\n",
       "      <td>CAFF</td>\n",
       "      <td>-0.000479</td>\n",
       "      <td>0.001722</td>\n",
       "      <td>-0.278133</td>\n",
       "      <td>7.809105e-01</td>\n",
       "      <td>-0.003855</td>\n",
       "      <td>0.002897</td>\n",
       "      <td>9.444158e-01</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5727</th>\n",
       "      <td>32913</td>\n",
       "      <td>ERG_01874</td>\n",
       "      <td>XIII</td>\n",
       "      <td>118671</td>\n",
       "      <td>HMG1</td>\n",
       "      <td>synonymous_variant</td>\n",
       "      <td>C</td>\n",
       "      <td>228</td>\n",
       "      <td>76</td>\n",
       "      <td>S</td>\n",
       "      <td>0</td>\n",
       "      <td>6832</td>\n",
       "      <td>COCL</td>\n",
       "      <td>-0.013476</td>\n",
       "      <td>0.005012</td>\n",
       "      <td>-2.688700</td>\n",
       "      <td>7.174306e-03</td>\n",
       "      <td>-0.023299</td>\n",
       "      <td>-0.003652</td>\n",
       "      <td>8.877886e-02</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5728</th>\n",
       "      <td>32913</td>\n",
       "      <td>ERG_01874</td>\n",
       "      <td>XIII</td>\n",
       "      <td>118671</td>\n",
       "      <td>HMG1</td>\n",
       "      <td>synonymous_variant</td>\n",
       "      <td>C</td>\n",
       "      <td>228</td>\n",
       "      <td>76</td>\n",
       "      <td>S</td>\n",
       "      <td>0</td>\n",
       "      <td>6833</td>\n",
       "      <td>LOV</td>\n",
       "      <td>-0.012673</td>\n",
       "      <td>0.002248</td>\n",
       "      <td>-5.637844</td>\n",
       "      <td>1.726664e-08</td>\n",
       "      <td>-0.017079</td>\n",
       "      <td>-0.008267</td>\n",
       "      <td>5.871312e-07</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5729</th>\n",
       "      <td>32913</td>\n",
       "      <td>ERG_01874</td>\n",
       "      <td>XIII</td>\n",
       "      <td>118671</td>\n",
       "      <td>HMG1</td>\n",
       "      <td>synonymous_variant</td>\n",
       "      <td>C</td>\n",
       "      <td>228</td>\n",
       "      <td>76</td>\n",
       "      <td>S</td>\n",
       "      <td>0</td>\n",
       "      <td>6834</td>\n",
       "      <td>NACL</td>\n",
       "      <td>-0.002862</td>\n",
       "      <td>0.001878</td>\n",
       "      <td>-1.524042</td>\n",
       "      <td>1.275014e-01</td>\n",
       "      <td>-0.006543</td>\n",
       "      <td>0.000819</td>\n",
       "      <td>4.771071e-01</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5730</th>\n",
       "      <td>32913</td>\n",
       "      <td>ERG_01874</td>\n",
       "      <td>XIII</td>\n",
       "      <td>118671</td>\n",
       "      <td>HMG1</td>\n",
       "      <td>synonymous_variant</td>\n",
       "      <td>C</td>\n",
       "      <td>228</td>\n",
       "      <td>76</td>\n",
       "      <td>S</td>\n",
       "      <td>0</td>\n",
       "      <td>6835</td>\n",
       "      <td>SC</td>\n",
       "      <td>-0.001400</td>\n",
       "      <td>0.001755</td>\n",
       "      <td>-0.797375</td>\n",
       "      <td>4.252353e-01</td>\n",
       "      <td>-0.004840</td>\n",
       "      <td>0.002041</td>\n",
       "      <td>7.819853e-01</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5731</th>\n",
       "      <td>32913</td>\n",
       "      <td>ERG_01874</td>\n",
       "      <td>XIII</td>\n",
       "      <td>118671</td>\n",
       "      <td>HMG1</td>\n",
       "      <td>synonymous_variant</td>\n",
       "      <td>C</td>\n",
       "      <td>228</td>\n",
       "      <td>76</td>\n",
       "      <td>S</td>\n",
       "      <td>0</td>\n",
       "      <td>6836</td>\n",
       "      <td>TBF</td>\n",
       "      <td>-0.001890</td>\n",
       "      <td>0.001406</td>\n",
       "      <td>-1.343894</td>\n",
       "      <td>1.789859e-01</td>\n",
       "      <td>-0.004646</td>\n",
       "      <td>0.000866</td>\n",
       "      <td>5.504194e-01</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5732</th>\n",
       "      <td>32917</td>\n",
       "      <td>ERG_01875</td>\n",
       "      <td>XIII</td>\n",
       "      <td>118671</td>\n",
       "      <td>HMG1</td>\n",
       "      <td>synonymous_variant</td>\n",
       "      <td>A</td>\n",
       "      <td>228</td>\n",
       "      <td>76</td>\n",
       "      <td>S</td>\n",
       "      <td>0</td>\n",
       "      <td>6837</td>\n",
       "      <td>CAFF</td>\n",
       "      <td>0.003000</td>\n",
       "      <td>0.001834</td>\n",
       "      <td>1.636114</td>\n",
       "      <td>1.018190e-01</td>\n",
       "      <td>-0.000594</td>\n",
       "      <td>0.006594</td>\n",
       "      <td>4.277162e-01</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5733</th>\n",
       "      <td>32917</td>\n",
       "      <td>ERG_01875</td>\n",
       "      <td>XIII</td>\n",
       "      <td>118671</td>\n",
       "      <td>HMG1</td>\n",
       "      <td>synonymous_variant</td>\n",
       "      <td>A</td>\n",
       "      <td>228</td>\n",
       "      <td>76</td>\n",
       "      <td>S</td>\n",
       "      <td>0</td>\n",
       "      <td>6838</td>\n",
       "      <td>COCL</td>\n",
       "      <td>-0.005776</td>\n",
       "      <td>0.005274</td>\n",
       "      <td>-1.095163</td>\n",
       "      <td>2.734481e-01</td>\n",
       "      <td>-0.016112</td>\n",
       "      <td>0.004561</td>\n",
       "      <td>6.609433e-01</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5734</th>\n",
       "      <td>32917</td>\n",
       "      <td>ERG_01875</td>\n",
       "      <td>XIII</td>\n",
       "      <td>118671</td>\n",
       "      <td>HMG1</td>\n",
       "      <td>synonymous_variant</td>\n",
       "      <td>A</td>\n",
       "      <td>228</td>\n",
       "      <td>76</td>\n",
       "      <td>S</td>\n",
       "      <td>0</td>\n",
       "      <td>6839</td>\n",
       "      <td>LOV</td>\n",
       "      <td>-0.003032</td>\n",
       "      <td>0.002158</td>\n",
       "      <td>-1.405052</td>\n",
       "      <td>1.600091e-01</td>\n",
       "      <td>-0.007261</td>\n",
       "      <td>0.001197</td>\n",
       "      <td>5.248311e-01</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5735</th>\n",
       "      <td>32917</td>\n",
       "      <td>ERG_01875</td>\n",
       "      <td>XIII</td>\n",
       "      <td>118671</td>\n",
       "      <td>HMG1</td>\n",
       "      <td>synonymous_variant</td>\n",
       "      <td>A</td>\n",
       "      <td>228</td>\n",
       "      <td>76</td>\n",
       "      <td>S</td>\n",
       "      <td>0</td>\n",
       "      <td>6840</td>\n",
       "      <td>NACL</td>\n",
       "      <td>0.002442</td>\n",
       "      <td>0.001884</td>\n",
       "      <td>1.295833</td>\n",
       "      <td>1.950360e-01</td>\n",
       "      <td>-0.001252</td>\n",
       "      <td>0.006135</td>\n",
       "      <td>5.706773e-01</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5736</th>\n",
       "      <td>32917</td>\n",
       "      <td>ERG_01875</td>\n",
       "      <td>XIII</td>\n",
       "      <td>118671</td>\n",
       "      <td>HMG1</td>\n",
       "      <td>synonymous_variant</td>\n",
       "      <td>A</td>\n",
       "      <td>228</td>\n",
       "      <td>76</td>\n",
       "      <td>S</td>\n",
       "      <td>0</td>\n",
       "      <td>6841</td>\n",
       "      <td>SC</td>\n",
       "      <td>-0.001553</td>\n",
       "      <td>0.001832</td>\n",
       "      <td>-0.847668</td>\n",
       "      <td>3.966252e-01</td>\n",
       "      <td>-0.005145</td>\n",
       "      <td>0.002038</td>\n",
       "      <td>7.607750e-01</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5737</th>\n",
       "      <td>32917</td>\n",
       "      <td>ERG_01875</td>\n",
       "      <td>XIII</td>\n",
       "      <td>118671</td>\n",
       "      <td>HMG1</td>\n",
       "      <td>synonymous_variant</td>\n",
       "      <td>A</td>\n",
       "      <td>228</td>\n",
       "      <td>76</td>\n",
       "      <td>S</td>\n",
       "      <td>0</td>\n",
       "      <td>6842</td>\n",
       "      <td>TBF</td>\n",
       "      <td>0.000747</td>\n",
       "      <td>0.001443</td>\n",
       "      <td>0.517854</td>\n",
       "      <td>6.045615e-01</td>\n",
       "      <td>-0.002082</td>\n",
       "      <td>0.003577</td>\n",
       "      <td>8.714112e-01</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "      Unnamed: 0_x     var_id chrom  SNP_chr_pos  Gene         Consequence  \\\n",
       "5726         32913  ERG_01874  XIII       118671  HMG1  synonymous_variant   \n",
       "5727         32913  ERG_01874  XIII       118671  HMG1  synonymous_variant   \n",
       "5728         32913  ERG_01874  XIII       118671  HMG1  synonymous_variant   \n",
       "5729         32913  ERG_01874  XIII       118671  HMG1  synonymous_variant   \n",
       "5730         32913  ERG_01874  XIII       118671  HMG1  synonymous_variant   \n",
       "5731         32913  ERG_01874  XIII       118671  HMG1  synonymous_variant   \n",
       "5732         32917  ERG_01875  XIII       118671  HMG1  synonymous_variant   \n",
       "5733         32917  ERG_01875  XIII       118671  HMG1  synonymous_variant   \n",
       "5734         32917  ERG_01875  XIII       118671  HMG1  synonymous_variant   \n",
       "5735         32917  ERG_01875  XIII       118671  HMG1  synonymous_variant   \n",
       "5736         32917  ERG_01875  XIII       118671  HMG1  synonymous_variant   \n",
       "5737         32917  ERG_01875  XIII       118671  HMG1  synonymous_variant   \n",
       "\n",
       "     ALT CDS_position Protein_position Amino_acids  DISTANCE  Unnamed: 0_y  \\\n",
       "5726   C          228               76           S         0          6831   \n",
       "5727   C          228               76           S         0          6832   \n",
       "5728   C          228               76           S         0          6833   \n",
       "5729   C          228               76           S         0          6834   \n",
       "5730   C          228               76           S         0          6835   \n",
       "5731   C          228               76           S         0          6836   \n",
       "5732   A          228               76           S         0          6837   \n",
       "5733   A          228               76           S         0          6838   \n",
       "5734   A          228               76           S         0          6839   \n",
       "5735   A          228               76           S         0          6840   \n",
       "5736   A          228               76           S         0          6841   \n",
       "5737   A          228               76           S         0          6842   \n",
       "\n",
       "     condition      coef   std err         t          pval  Conf. Int. Low  \\\n",
       "5726      CAFF -0.000479  0.001722 -0.278133  7.809105e-01       -0.003855   \n",
       "5727      COCL -0.013476  0.005012 -2.688700  7.174306e-03       -0.023299   \n",
       "5728       LOV -0.012673  0.002248 -5.637844  1.726664e-08       -0.017079   \n",
       "5729      NACL -0.002862  0.001878 -1.524042  1.275014e-01       -0.006543   \n",
       "5730        SC -0.001400  0.001755 -0.797375  4.252353e-01       -0.004840   \n",
       "5731       TBF -0.001890  0.001406 -1.343894  1.789859e-01       -0.004646   \n",
       "5732      CAFF  0.003000  0.001834  1.636114  1.018190e-01       -0.000594   \n",
       "5733      COCL -0.005776  0.005274 -1.095163  2.734481e-01       -0.016112   \n",
       "5734       LOV -0.003032  0.002158 -1.405052  1.600091e-01       -0.007261   \n",
       "5735      NACL  0.002442  0.001884  1.295833  1.950360e-01       -0.001252   \n",
       "5736        SC -0.001553  0.001832 -0.847668  3.966252e-01       -0.005145   \n",
       "5737       TBF  0.000747  0.001443  0.517854  6.045615e-01       -0.002082   \n",
       "\n",
       "      Conf. Int. Upp.          padj  \n",
       "5726         0.002897  9.444158e-01  \n",
       "5727        -0.003652  8.877886e-02  \n",
       "5728        -0.008267  5.871312e-07  \n",
       "5729         0.000819  4.771071e-01  \n",
       "5730         0.002041  7.819853e-01  \n",
       "5731         0.000866  5.504194e-01  \n",
       "5732         0.006594  4.277162e-01  \n",
       "5733         0.004561  6.609433e-01  \n",
       "5734         0.001197  5.248311e-01  \n",
       "5735         0.006135  5.706773e-01  \n",
       "5736         0.002038  7.607750e-01  \n",
       "5737         0.003577  8.714112e-01  "
      ]
     },
     "execution_count": 12,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "#HMG1 site with two synonymous variants, one with a significant effect in lovastatin and one without\n",
    "\n",
    "\n",
    "chrom = 'XIII'\n",
    "pos = 118671\n",
    "\n",
    "annotated_fits[(annotated_fits['chrom']==chrom)&(annotated_fits['SNP_chr_pos']==pos)]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 13,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Unnamed: 0_x</th>\n",
       "      <th>var_id</th>\n",
       "      <th>chrom</th>\n",
       "      <th>SNP_chr_pos</th>\n",
       "      <th>Gene</th>\n",
       "      <th>Consequence</th>\n",
       "      <th>ALT</th>\n",
       "      <th>CDS_position</th>\n",
       "      <th>Protein_position</th>\n",
       "      <th>Amino_acids</th>\n",
       "      <th>DISTANCE</th>\n",
       "      <th>Unnamed: 0_y</th>\n",
       "      <th>condition</th>\n",
       "      <th>coef</th>\n",
       "      <th>std err</th>\n",
       "      <th>t</th>\n",
       "      <th>pval</th>\n",
       "      <th>Conf. Int. Low</th>\n",
       "      <th>Conf. Int. Upp.</th>\n",
       "      <th>padj</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>5186</th>\n",
       "      <td>32424</td>\n",
       "      <td>ERG_01662</td>\n",
       "      <td>XIII</td>\n",
       "      <td>19933</td>\n",
       "      <td>ERG13</td>\n",
       "      <td>synonymous_variant</td>\n",
       "      <td>T</td>\n",
       "      <td>603</td>\n",
       "      <td>201</td>\n",
       "      <td>T</td>\n",
       "      <td>0</td>\n",
       "      <td>6359</td>\n",
       "      <td>CAFF</td>\n",
       "      <td>0.001372</td>\n",
       "      <td>0.000328</td>\n",
       "      <td>4.183220</td>\n",
       "      <td>0.000029</td>\n",
       "      <td>0.000729</td>\n",
       "      <td>0.002015</td>\n",
       "      <td>0.000729</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5187</th>\n",
       "      <td>32424</td>\n",
       "      <td>ERG_01662</td>\n",
       "      <td>XIII</td>\n",
       "      <td>19933</td>\n",
       "      <td>ERG13</td>\n",
       "      <td>synonymous_variant</td>\n",
       "      <td>T</td>\n",
       "      <td>603</td>\n",
       "      <td>201</td>\n",
       "      <td>T</td>\n",
       "      <td>0</td>\n",
       "      <td>6360</td>\n",
       "      <td>COCL</td>\n",
       "      <td>-0.000541</td>\n",
       "      <td>0.001221</td>\n",
       "      <td>-0.442895</td>\n",
       "      <td>0.657843</td>\n",
       "      <td>-0.002935</td>\n",
       "      <td>0.001853</td>\n",
       "      <td>0.896217</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5188</th>\n",
       "      <td>32424</td>\n",
       "      <td>ERG_01662</td>\n",
       "      <td>XIII</td>\n",
       "      <td>19933</td>\n",
       "      <td>ERG13</td>\n",
       "      <td>synonymous_variant</td>\n",
       "      <td>T</td>\n",
       "      <td>603</td>\n",
       "      <td>201</td>\n",
       "      <td>T</td>\n",
       "      <td>0</td>\n",
       "      <td>6361</td>\n",
       "      <td>LOV</td>\n",
       "      <td>-0.001460</td>\n",
       "      <td>0.000474</td>\n",
       "      <td>-3.083182</td>\n",
       "      <td>0.002049</td>\n",
       "      <td>-0.002388</td>\n",
       "      <td>-0.000532</td>\n",
       "      <td>0.033558</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5189</th>\n",
       "      <td>32424</td>\n",
       "      <td>ERG_01662</td>\n",
       "      <td>XIII</td>\n",
       "      <td>19933</td>\n",
       "      <td>ERG13</td>\n",
       "      <td>synonymous_variant</td>\n",
       "      <td>T</td>\n",
       "      <td>603</td>\n",
       "      <td>201</td>\n",
       "      <td>T</td>\n",
       "      <td>0</td>\n",
       "      <td>6362</td>\n",
       "      <td>NACL</td>\n",
       "      <td>-0.000636</td>\n",
       "      <td>0.000312</td>\n",
       "      <td>-2.039360</td>\n",
       "      <td>0.041417</td>\n",
       "      <td>-0.001248</td>\n",
       "      <td>-0.000025</td>\n",
       "      <td>0.272380</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5190</th>\n",
       "      <td>32424</td>\n",
       "      <td>ERG_01662</td>\n",
       "      <td>XIII</td>\n",
       "      <td>19933</td>\n",
       "      <td>ERG13</td>\n",
       "      <td>synonymous_variant</td>\n",
       "      <td>T</td>\n",
       "      <td>603</td>\n",
       "      <td>201</td>\n",
       "      <td>T</td>\n",
       "      <td>0</td>\n",
       "      <td>6363</td>\n",
       "      <td>SC</td>\n",
       "      <td>-0.001267</td>\n",
       "      <td>0.000305</td>\n",
       "      <td>-4.147841</td>\n",
       "      <td>0.000034</td>\n",
       "      <td>-0.001865</td>\n",
       "      <td>-0.000668</td>\n",
       "      <td>0.000845</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5191</th>\n",
       "      <td>32424</td>\n",
       "      <td>ERG_01662</td>\n",
       "      <td>XIII</td>\n",
       "      <td>19933</td>\n",
       "      <td>ERG13</td>\n",
       "      <td>synonymous_variant</td>\n",
       "      <td>T</td>\n",
       "      <td>603</td>\n",
       "      <td>201</td>\n",
       "      <td>T</td>\n",
       "      <td>0</td>\n",
       "      <td>6364</td>\n",
       "      <td>TBF</td>\n",
       "      <td>0.000009</td>\n",
       "      <td>0.000230</td>\n",
       "      <td>0.037198</td>\n",
       "      <td>0.970327</td>\n",
       "      <td>-0.000443</td>\n",
       "      <td>0.000460</td>\n",
       "      <td>0.996191</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5192</th>\n",
       "      <td>32430</td>\n",
       "      <td>ERG_01663</td>\n",
       "      <td>XIII</td>\n",
       "      <td>19933</td>\n",
       "      <td>ERG13</td>\n",
       "      <td>synonymous_variant</td>\n",
       "      <td>A</td>\n",
       "      <td>603</td>\n",
       "      <td>201</td>\n",
       "      <td>T</td>\n",
       "      <td>0</td>\n",
       "      <td>6365</td>\n",
       "      <td>CAFF</td>\n",
       "      <td>0.000772</td>\n",
       "      <td>0.000523</td>\n",
       "      <td>1.475704</td>\n",
       "      <td>0.140027</td>\n",
       "      <td>-0.000253</td>\n",
       "      <td>0.001798</td>\n",
       "      <td>0.499015</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5193</th>\n",
       "      <td>32430</td>\n",
       "      <td>ERG_01663</td>\n",
       "      <td>XIII</td>\n",
       "      <td>19933</td>\n",
       "      <td>ERG13</td>\n",
       "      <td>synonymous_variant</td>\n",
       "      <td>A</td>\n",
       "      <td>603</td>\n",
       "      <td>201</td>\n",
       "      <td>T</td>\n",
       "      <td>0</td>\n",
       "      <td>6366</td>\n",
       "      <td>COCL</td>\n",
       "      <td>-0.003465</td>\n",
       "      <td>0.001799</td>\n",
       "      <td>-1.925417</td>\n",
       "      <td>0.054180</td>\n",
       "      <td>-0.006992</td>\n",
       "      <td>0.000062</td>\n",
       "      <td>0.311792</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5194</th>\n",
       "      <td>32430</td>\n",
       "      <td>ERG_01663</td>\n",
       "      <td>XIII</td>\n",
       "      <td>19933</td>\n",
       "      <td>ERG13</td>\n",
       "      <td>synonymous_variant</td>\n",
       "      <td>A</td>\n",
       "      <td>603</td>\n",
       "      <td>201</td>\n",
       "      <td>T</td>\n",
       "      <td>0</td>\n",
       "      <td>6367</td>\n",
       "      <td>LOV</td>\n",
       "      <td>0.002975</td>\n",
       "      <td>0.000656</td>\n",
       "      <td>4.535114</td>\n",
       "      <td>0.000006</td>\n",
       "      <td>0.001689</td>\n",
       "      <td>0.004260</td>\n",
       "      <td>0.000161</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5195</th>\n",
       "      <td>32430</td>\n",
       "      <td>ERG_01663</td>\n",
       "      <td>XIII</td>\n",
       "      <td>19933</td>\n",
       "      <td>ERG13</td>\n",
       "      <td>synonymous_variant</td>\n",
       "      <td>A</td>\n",
       "      <td>603</td>\n",
       "      <td>201</td>\n",
       "      <td>T</td>\n",
       "      <td>0</td>\n",
       "      <td>6368</td>\n",
       "      <td>NACL</td>\n",
       "      <td>0.000663</td>\n",
       "      <td>0.000523</td>\n",
       "      <td>1.267137</td>\n",
       "      <td>0.205109</td>\n",
       "      <td>-0.000363</td>\n",
       "      <td>0.001689</td>\n",
       "      <td>0.584708</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5196</th>\n",
       "      <td>32430</td>\n",
       "      <td>ERG_01663</td>\n",
       "      <td>XIII</td>\n",
       "      <td>19933</td>\n",
       "      <td>ERG13</td>\n",
       "      <td>synonymous_variant</td>\n",
       "      <td>A</td>\n",
       "      <td>603</td>\n",
       "      <td>201</td>\n",
       "      <td>T</td>\n",
       "      <td>0</td>\n",
       "      <td>6369</td>\n",
       "      <td>SC</td>\n",
       "      <td>-0.002254</td>\n",
       "      <td>0.000464</td>\n",
       "      <td>-4.858600</td>\n",
       "      <td>0.000001</td>\n",
       "      <td>-0.003164</td>\n",
       "      <td>-0.001345</td>\n",
       "      <td>0.000036</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5197</th>\n",
       "      <td>32430</td>\n",
       "      <td>ERG_01663</td>\n",
       "      <td>XIII</td>\n",
       "      <td>19933</td>\n",
       "      <td>ERG13</td>\n",
       "      <td>synonymous_variant</td>\n",
       "      <td>A</td>\n",
       "      <td>603</td>\n",
       "      <td>201</td>\n",
       "      <td>T</td>\n",
       "      <td>0</td>\n",
       "      <td>6370</td>\n",
       "      <td>TBF</td>\n",
       "      <td>0.000253</td>\n",
       "      <td>0.000320</td>\n",
       "      <td>0.791347</td>\n",
       "      <td>0.428743</td>\n",
       "      <td>-0.000374</td>\n",
       "      <td>0.000881</td>\n",
       "      <td>0.783238</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "      Unnamed: 0_x     var_id chrom  SNP_chr_pos   Gene         Consequence  \\\n",
       "5186         32424  ERG_01662  XIII        19933  ERG13  synonymous_variant   \n",
       "5187         32424  ERG_01662  XIII        19933  ERG13  synonymous_variant   \n",
       "5188         32424  ERG_01662  XIII        19933  ERG13  synonymous_variant   \n",
       "5189         32424  ERG_01662  XIII        19933  ERG13  synonymous_variant   \n",
       "5190         32424  ERG_01662  XIII        19933  ERG13  synonymous_variant   \n",
       "5191         32424  ERG_01662  XIII        19933  ERG13  synonymous_variant   \n",
       "5192         32430  ERG_01663  XIII        19933  ERG13  synonymous_variant   \n",
       "5193         32430  ERG_01663  XIII        19933  ERG13  synonymous_variant   \n",
       "5194         32430  ERG_01663  XIII        19933  ERG13  synonymous_variant   \n",
       "5195         32430  ERG_01663  XIII        19933  ERG13  synonymous_variant   \n",
       "5196         32430  ERG_01663  XIII        19933  ERG13  synonymous_variant   \n",
       "5197         32430  ERG_01663  XIII        19933  ERG13  synonymous_variant   \n",
       "\n",
       "     ALT CDS_position Protein_position Amino_acids  DISTANCE  Unnamed: 0_y  \\\n",
       "5186   T          603              201           T         0          6359   \n",
       "5187   T          603              201           T         0          6360   \n",
       "5188   T          603              201           T         0          6361   \n",
       "5189   T          603              201           T         0          6362   \n",
       "5190   T          603              201           T         0          6363   \n",
       "5191   T          603              201           T         0          6364   \n",
       "5192   A          603              201           T         0          6365   \n",
       "5193   A          603              201           T         0          6366   \n",
       "5194   A          603              201           T         0          6367   \n",
       "5195   A          603              201           T         0          6368   \n",
       "5196   A          603              201           T         0          6369   \n",
       "5197   A          603              201           T         0          6370   \n",
       "\n",
       "     condition      coef   std err         t      pval  Conf. Int. Low  \\\n",
       "5186      CAFF  0.001372  0.000328  4.183220  0.000029        0.000729   \n",
       "5187      COCL -0.000541  0.001221 -0.442895  0.657843       -0.002935   \n",
       "5188       LOV -0.001460  0.000474 -3.083182  0.002049       -0.002388   \n",
       "5189      NACL -0.000636  0.000312 -2.039360  0.041417       -0.001248   \n",
       "5190        SC -0.001267  0.000305 -4.147841  0.000034       -0.001865   \n",
       "5191       TBF  0.000009  0.000230  0.037198  0.970327       -0.000443   \n",
       "5192      CAFF  0.000772  0.000523  1.475704  0.140027       -0.000253   \n",
       "5193      COCL -0.003465  0.001799 -1.925417  0.054180       -0.006992   \n",
       "5194       LOV  0.002975  0.000656  4.535114  0.000006        0.001689   \n",
       "5195      NACL  0.000663  0.000523  1.267137  0.205109       -0.000363   \n",
       "5196        SC -0.002254  0.000464 -4.858600  0.000001       -0.003164   \n",
       "5197       TBF  0.000253  0.000320  0.791347  0.428743       -0.000374   \n",
       "\n",
       "      Conf. Int. Upp.      padj  \n",
       "5186         0.002015  0.000729  \n",
       "5187         0.001853  0.896217  \n",
       "5188        -0.000532  0.033558  \n",
       "5189        -0.000025  0.272380  \n",
       "5190        -0.000668  0.000845  \n",
       "5191         0.000460  0.996191  \n",
       "5192         0.001798  0.499015  \n",
       "5193         0.000062  0.311792  \n",
       "5194         0.004260  0.000161  \n",
       "5195         0.001689  0.584708  \n",
       "5196        -0.001345  0.000036  \n",
       "5197         0.000881  0.783238  "
      ]
     },
     "execution_count": 13,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "#ERG13 site with two synonymous variants, both with significant fitness effects in the same direction in SC\n",
    "\n",
    "chrom = 'XIII'\n",
    "pos = 19933\n",
    "\n",
    "annotated_fits[(annotated_fits['chrom']==chrom)&(annotated_fits['SNP_chr_pos']==pos)]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 14,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Unnamed: 0_x</th>\n",
       "      <th>var_id</th>\n",
       "      <th>chrom</th>\n",
       "      <th>SNP_chr_pos</th>\n",
       "      <th>Gene</th>\n",
       "      <th>Consequence</th>\n",
       "      <th>ALT</th>\n",
       "      <th>CDS_position</th>\n",
       "      <th>Protein_position</th>\n",
       "      <th>Amino_acids</th>\n",
       "      <th>DISTANCE</th>\n",
       "      <th>Unnamed: 0_y</th>\n",
       "      <th>condition</th>\n",
       "      <th>coef</th>\n",
       "      <th>std err</th>\n",
       "      <th>t</th>\n",
       "      <th>pval</th>\n",
       "      <th>Conf. Int. Low</th>\n",
       "      <th>Conf. Int. Upp.</th>\n",
       "      <th>padj</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>3554</th>\n",
       "      <td>24260</td>\n",
       "      <td>EG8_00143</td>\n",
       "      <td>X</td>\n",
       "      <td>105664</td>\n",
       "      <td>ERG20</td>\n",
       "      <td>synonymous_variant</td>\n",
       "      <td>A</td>\n",
       "      <td>651</td>\n",
       "      <td>217</td>\n",
       "      <td>T</td>\n",
       "      <td>0</td>\n",
       "      <td>278</td>\n",
       "      <td>CAFF</td>\n",
       "      <td>0.000613</td>\n",
       "      <td>0.000758</td>\n",
       "      <td>0.807910</td>\n",
       "      <td>4.191444e-01</td>\n",
       "      <td>-0.000874</td>\n",
       "      <td>0.002099</td>\n",
       "      <td>7.766067e-01</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3555</th>\n",
       "      <td>24260</td>\n",
       "      <td>EG8_00143</td>\n",
       "      <td>X</td>\n",
       "      <td>105664</td>\n",
       "      <td>ERG20</td>\n",
       "      <td>synonymous_variant</td>\n",
       "      <td>A</td>\n",
       "      <td>651</td>\n",
       "      <td>217</td>\n",
       "      <td>T</td>\n",
       "      <td>0</td>\n",
       "      <td>279</td>\n",
       "      <td>COCL</td>\n",
       "      <td>-0.002869</td>\n",
       "      <td>0.002611</td>\n",
       "      <td>-1.098752</td>\n",
       "      <td>2.718789e-01</td>\n",
       "      <td>-0.007988</td>\n",
       "      <td>0.002249</td>\n",
       "      <td>6.589246e-01</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3556</th>\n",
       "      <td>24260</td>\n",
       "      <td>EG8_00143</td>\n",
       "      <td>X</td>\n",
       "      <td>105664</td>\n",
       "      <td>ERG20</td>\n",
       "      <td>synonymous_variant</td>\n",
       "      <td>A</td>\n",
       "      <td>651</td>\n",
       "      <td>217</td>\n",
       "      <td>T</td>\n",
       "      <td>0</td>\n",
       "      <td>280</td>\n",
       "      <td>LOV</td>\n",
       "      <td>0.003201</td>\n",
       "      <td>0.000980</td>\n",
       "      <td>3.267927</td>\n",
       "      <td>1.083757e-03</td>\n",
       "      <td>0.001281</td>\n",
       "      <td>0.005121</td>\n",
       "      <td>1.977416e-02</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3557</th>\n",
       "      <td>24260</td>\n",
       "      <td>EG8_00143</td>\n",
       "      <td>X</td>\n",
       "      <td>105664</td>\n",
       "      <td>ERG20</td>\n",
       "      <td>synonymous_variant</td>\n",
       "      <td>A</td>\n",
       "      <td>651</td>\n",
       "      <td>217</td>\n",
       "      <td>T</td>\n",
       "      <td>0</td>\n",
       "      <td>281</td>\n",
       "      <td>NACL</td>\n",
       "      <td>0.006764</td>\n",
       "      <td>0.000760</td>\n",
       "      <td>8.896273</td>\n",
       "      <td>5.871106e-19</td>\n",
       "      <td>0.005274</td>\n",
       "      <td>0.008254</td>\n",
       "      <td>2.648488e-17</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3558</th>\n",
       "      <td>24260</td>\n",
       "      <td>EG8_00143</td>\n",
       "      <td>X</td>\n",
       "      <td>105664</td>\n",
       "      <td>ERG20</td>\n",
       "      <td>synonymous_variant</td>\n",
       "      <td>A</td>\n",
       "      <td>651</td>\n",
       "      <td>217</td>\n",
       "      <td>T</td>\n",
       "      <td>0</td>\n",
       "      <td>282</td>\n",
       "      <td>SC</td>\n",
       "      <td>-0.000556</td>\n",
       "      <td>0.000790</td>\n",
       "      <td>-0.703047</td>\n",
       "      <td>4.820280e-01</td>\n",
       "      <td>-0.002105</td>\n",
       "      <td>0.000993</td>\n",
       "      <td>8.164463e-01</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3559</th>\n",
       "      <td>24260</td>\n",
       "      <td>EG8_00143</td>\n",
       "      <td>X</td>\n",
       "      <td>105664</td>\n",
       "      <td>ERG20</td>\n",
       "      <td>synonymous_variant</td>\n",
       "      <td>A</td>\n",
       "      <td>651</td>\n",
       "      <td>217</td>\n",
       "      <td>T</td>\n",
       "      <td>0</td>\n",
       "      <td>283</td>\n",
       "      <td>TBF</td>\n",
       "      <td>-0.000717</td>\n",
       "      <td>0.000654</td>\n",
       "      <td>-1.096428</td>\n",
       "      <td>2.728942e-01</td>\n",
       "      <td>-0.001998</td>\n",
       "      <td>0.000564</td>\n",
       "      <td>6.599600e-01</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3560</th>\n",
       "      <td>24268</td>\n",
       "      <td>EG8_00144</td>\n",
       "      <td>X</td>\n",
       "      <td>105664</td>\n",
       "      <td>ERG20</td>\n",
       "      <td>synonymous_variant</td>\n",
       "      <td>C</td>\n",
       "      <td>651</td>\n",
       "      <td>217</td>\n",
       "      <td>T</td>\n",
       "      <td>0</td>\n",
       "      <td>284</td>\n",
       "      <td>CAFF</td>\n",
       "      <td>0.000355</td>\n",
       "      <td>0.000819</td>\n",
       "      <td>0.433923</td>\n",
       "      <td>6.643452e-01</td>\n",
       "      <td>-0.001250</td>\n",
       "      <td>0.001960</td>\n",
       "      <td>8.983156e-01</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3561</th>\n",
       "      <td>24268</td>\n",
       "      <td>EG8_00144</td>\n",
       "      <td>X</td>\n",
       "      <td>105664</td>\n",
       "      <td>ERG20</td>\n",
       "      <td>synonymous_variant</td>\n",
       "      <td>C</td>\n",
       "      <td>651</td>\n",
       "      <td>217</td>\n",
       "      <td>T</td>\n",
       "      <td>0</td>\n",
       "      <td>285</td>\n",
       "      <td>COCL</td>\n",
       "      <td>0.001231</td>\n",
       "      <td>0.002720</td>\n",
       "      <td>0.452366</td>\n",
       "      <td>6.510062e-01</td>\n",
       "      <td>-0.004101</td>\n",
       "      <td>0.006562</td>\n",
       "      <td>8.928889e-01</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3562</th>\n",
       "      <td>24268</td>\n",
       "      <td>EG8_00144</td>\n",
       "      <td>X</td>\n",
       "      <td>105664</td>\n",
       "      <td>ERG20</td>\n",
       "      <td>synonymous_variant</td>\n",
       "      <td>C</td>\n",
       "      <td>651</td>\n",
       "      <td>217</td>\n",
       "      <td>T</td>\n",
       "      <td>0</td>\n",
       "      <td>286</td>\n",
       "      <td>LOV</td>\n",
       "      <td>0.001006</td>\n",
       "      <td>0.000993</td>\n",
       "      <td>1.012299</td>\n",
       "      <td>3.113976e-01</td>\n",
       "      <td>-0.000941</td>\n",
       "      <td>0.002953</td>\n",
       "      <td>6.915922e-01</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3563</th>\n",
       "      <td>24268</td>\n",
       "      <td>EG8_00144</td>\n",
       "      <td>X</td>\n",
       "      <td>105664</td>\n",
       "      <td>ERG20</td>\n",
       "      <td>synonymous_variant</td>\n",
       "      <td>C</td>\n",
       "      <td>651</td>\n",
       "      <td>217</td>\n",
       "      <td>T</td>\n",
       "      <td>0</td>\n",
       "      <td>287</td>\n",
       "      <td>NACL</td>\n",
       "      <td>-0.000781</td>\n",
       "      <td>0.000822</td>\n",
       "      <td>-0.950841</td>\n",
       "      <td>3.416877e-01</td>\n",
       "      <td>-0.002392</td>\n",
       "      <td>0.000829</td>\n",
       "      <td>7.174010e-01</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3564</th>\n",
       "      <td>24268</td>\n",
       "      <td>EG8_00144</td>\n",
       "      <td>X</td>\n",
       "      <td>105664</td>\n",
       "      <td>ERG20</td>\n",
       "      <td>synonymous_variant</td>\n",
       "      <td>C</td>\n",
       "      <td>651</td>\n",
       "      <td>217</td>\n",
       "      <td>T</td>\n",
       "      <td>0</td>\n",
       "      <td>288</td>\n",
       "      <td>SC</td>\n",
       "      <td>0.000527</td>\n",
       "      <td>0.000754</td>\n",
       "      <td>0.698707</td>\n",
       "      <td>4.847369e-01</td>\n",
       "      <td>-0.000951</td>\n",
       "      <td>0.002004</td>\n",
       "      <td>8.168860e-01</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3565</th>\n",
       "      <td>24268</td>\n",
       "      <td>EG8_00144</td>\n",
       "      <td>X</td>\n",
       "      <td>105664</td>\n",
       "      <td>ERG20</td>\n",
       "      <td>synonymous_variant</td>\n",
       "      <td>C</td>\n",
       "      <td>651</td>\n",
       "      <td>217</td>\n",
       "      <td>T</td>\n",
       "      <td>0</td>\n",
       "      <td>289</td>\n",
       "      <td>TBF</td>\n",
       "      <td>-0.000639</td>\n",
       "      <td>0.000574</td>\n",
       "      <td>-1.112977</td>\n",
       "      <td>2.657211e-01</td>\n",
       "      <td>-0.001764</td>\n",
       "      <td>0.000486</td>\n",
       "      <td>6.529972e-01</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "      Unnamed: 0_x     var_id chrom  SNP_chr_pos   Gene         Consequence  \\\n",
       "3554         24260  EG8_00143     X       105664  ERG20  synonymous_variant   \n",
       "3555         24260  EG8_00143     X       105664  ERG20  synonymous_variant   \n",
       "3556         24260  EG8_00143     X       105664  ERG20  synonymous_variant   \n",
       "3557         24260  EG8_00143     X       105664  ERG20  synonymous_variant   \n",
       "3558         24260  EG8_00143     X       105664  ERG20  synonymous_variant   \n",
       "3559         24260  EG8_00143     X       105664  ERG20  synonymous_variant   \n",
       "3560         24268  EG8_00144     X       105664  ERG20  synonymous_variant   \n",
       "3561         24268  EG8_00144     X       105664  ERG20  synonymous_variant   \n",
       "3562         24268  EG8_00144     X       105664  ERG20  synonymous_variant   \n",
       "3563         24268  EG8_00144     X       105664  ERG20  synonymous_variant   \n",
       "3564         24268  EG8_00144     X       105664  ERG20  synonymous_variant   \n",
       "3565         24268  EG8_00144     X       105664  ERG20  synonymous_variant   \n",
       "\n",
       "     ALT CDS_position Protein_position Amino_acids  DISTANCE  Unnamed: 0_y  \\\n",
       "3554   A          651              217           T         0           278   \n",
       "3555   A          651              217           T         0           279   \n",
       "3556   A          651              217           T         0           280   \n",
       "3557   A          651              217           T         0           281   \n",
       "3558   A          651              217           T         0           282   \n",
       "3559   A          651              217           T         0           283   \n",
       "3560   C          651              217           T         0           284   \n",
       "3561   C          651              217           T         0           285   \n",
       "3562   C          651              217           T         0           286   \n",
       "3563   C          651              217           T         0           287   \n",
       "3564   C          651              217           T         0           288   \n",
       "3565   C          651              217           T         0           289   \n",
       "\n",
       "     condition      coef   std err         t          pval  Conf. Int. Low  \\\n",
       "3554      CAFF  0.000613  0.000758  0.807910  4.191444e-01       -0.000874   \n",
       "3555      COCL -0.002869  0.002611 -1.098752  2.718789e-01       -0.007988   \n",
       "3556       LOV  0.003201  0.000980  3.267927  1.083757e-03        0.001281   \n",
       "3557      NACL  0.006764  0.000760  8.896273  5.871106e-19        0.005274   \n",
       "3558        SC -0.000556  0.000790 -0.703047  4.820280e-01       -0.002105   \n",
       "3559       TBF -0.000717  0.000654 -1.096428  2.728942e-01       -0.001998   \n",
       "3560      CAFF  0.000355  0.000819  0.433923  6.643452e-01       -0.001250   \n",
       "3561      COCL  0.001231  0.002720  0.452366  6.510062e-01       -0.004101   \n",
       "3562       LOV  0.001006  0.000993  1.012299  3.113976e-01       -0.000941   \n",
       "3563      NACL -0.000781  0.000822 -0.950841  3.416877e-01       -0.002392   \n",
       "3564        SC  0.000527  0.000754  0.698707  4.847369e-01       -0.000951   \n",
       "3565       TBF -0.000639  0.000574 -1.112977  2.657211e-01       -0.001764   \n",
       "\n",
       "      Conf. Int. Upp.          padj  \n",
       "3554         0.002099  7.766067e-01  \n",
       "3555         0.002249  6.589246e-01  \n",
       "3556         0.005121  1.977416e-02  \n",
       "3557         0.008254  2.648488e-17  \n",
       "3558         0.000993  8.164463e-01  \n",
       "3559         0.000564  6.599600e-01  \n",
       "3560         0.001960  8.983156e-01  \n",
       "3561         0.006562  8.928889e-01  \n",
       "3562         0.002953  6.915922e-01  \n",
       "3563         0.000829  7.174010e-01  \n",
       "3564         0.002004  8.168860e-01  \n",
       "3565         0.000486  6.529972e-01  "
      ]
     },
     "execution_count": 14,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "#ERG20 site with two synonymous variants, one with a significant effect in sodium chloride and one without\n",
    "\n",
    "chrom = 'X'\n",
    "pos = 105664\n",
    "\n",
    "annotated_fits[(annotated_fits['chrom']==chrom)&(annotated_fits['SNP_chr_pos']==pos)]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 15,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Unnamed: 0_x</th>\n",
       "      <th>var_id</th>\n",
       "      <th>chrom</th>\n",
       "      <th>SNP_chr_pos</th>\n",
       "      <th>Gene</th>\n",
       "      <th>Consequence</th>\n",
       "      <th>ALT</th>\n",
       "      <th>CDS_position</th>\n",
       "      <th>Protein_position</th>\n",
       "      <th>Amino_acids</th>\n",
       "      <th>DISTANCE</th>\n",
       "      <th>Unnamed: 0_y</th>\n",
       "      <th>condition</th>\n",
       "      <th>coef</th>\n",
       "      <th>std err</th>\n",
       "      <th>t</th>\n",
       "      <th>pval</th>\n",
       "      <th>Conf. Int. Low</th>\n",
       "      <th>Conf. Int. Upp.</th>\n",
       "      <th>padj</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>2967</th>\n",
       "      <td>22872</td>\n",
       "      <td>EGB_00026</td>\n",
       "      <td>VIII</td>\n",
       "      <td>240738</td>\n",
       "      <td>ERG7</td>\n",
       "      <td>synonymous_variant</td>\n",
       "      <td>T</td>\n",
       "      <td>1641</td>\n",
       "      <td>547</td>\n",
       "      <td>G</td>\n",
       "      <td>0</td>\n",
       "      <td>1103</td>\n",
       "      <td>CAFF</td>\n",
       "      <td>0.000308</td>\n",
       "      <td>0.001230</td>\n",
       "      <td>0.250274</td>\n",
       "      <td>8.023761e-01</td>\n",
       "      <td>-0.002102</td>\n",
       "      <td>0.002718</td>\n",
       "      <td>9.525466e-01</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2968</th>\n",
       "      <td>22872</td>\n",
       "      <td>EGB_00026</td>\n",
       "      <td>VIII</td>\n",
       "      <td>240738</td>\n",
       "      <td>ERG7</td>\n",
       "      <td>synonymous_variant</td>\n",
       "      <td>T</td>\n",
       "      <td>1641</td>\n",
       "      <td>547</td>\n",
       "      <td>G</td>\n",
       "      <td>0</td>\n",
       "      <td>1104</td>\n",
       "      <td>COCL</td>\n",
       "      <td>-0.017969</td>\n",
       "      <td>0.011281</td>\n",
       "      <td>-1.592894</td>\n",
       "      <td>1.111872e-01</td>\n",
       "      <td>-0.040079</td>\n",
       "      <td>0.004141</td>\n",
       "      <td>4.481939e-01</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2969</th>\n",
       "      <td>22872</td>\n",
       "      <td>EGB_00026</td>\n",
       "      <td>VIII</td>\n",
       "      <td>240738</td>\n",
       "      <td>ERG7</td>\n",
       "      <td>synonymous_variant</td>\n",
       "      <td>T</td>\n",
       "      <td>1641</td>\n",
       "      <td>547</td>\n",
       "      <td>G</td>\n",
       "      <td>0</td>\n",
       "      <td>1105</td>\n",
       "      <td>LOV</td>\n",
       "      <td>-0.001969</td>\n",
       "      <td>0.001845</td>\n",
       "      <td>-1.067331</td>\n",
       "      <td>2.858251e-01</td>\n",
       "      <td>-0.005585</td>\n",
       "      <td>0.001647</td>\n",
       "      <td>6.706642e-01</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2970</th>\n",
       "      <td>22872</td>\n",
       "      <td>EGB_00026</td>\n",
       "      <td>VIII</td>\n",
       "      <td>240738</td>\n",
       "      <td>ERG7</td>\n",
       "      <td>synonymous_variant</td>\n",
       "      <td>T</td>\n",
       "      <td>1641</td>\n",
       "      <td>547</td>\n",
       "      <td>G</td>\n",
       "      <td>0</td>\n",
       "      <td>1106</td>\n",
       "      <td>NACL</td>\n",
       "      <td>0.001724</td>\n",
       "      <td>0.001374</td>\n",
       "      <td>1.255040</td>\n",
       "      <td>2.094672e-01</td>\n",
       "      <td>-0.000968</td>\n",
       "      <td>0.004416</td>\n",
       "      <td>5.893554e-01</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2971</th>\n",
       "      <td>22872</td>\n",
       "      <td>EGB_00026</td>\n",
       "      <td>VIII</td>\n",
       "      <td>240738</td>\n",
       "      <td>ERG7</td>\n",
       "      <td>synonymous_variant</td>\n",
       "      <td>T</td>\n",
       "      <td>1641</td>\n",
       "      <td>547</td>\n",
       "      <td>G</td>\n",
       "      <td>0</td>\n",
       "      <td>1107</td>\n",
       "      <td>SC</td>\n",
       "      <td>-0.000612</td>\n",
       "      <td>0.001205</td>\n",
       "      <td>-0.508176</td>\n",
       "      <td>6.113309e-01</td>\n",
       "      <td>-0.002973</td>\n",
       "      <td>0.001749</td>\n",
       "      <td>8.749904e-01</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2972</th>\n",
       "      <td>22872</td>\n",
       "      <td>EGB_00026</td>\n",
       "      <td>VIII</td>\n",
       "      <td>240738</td>\n",
       "      <td>ERG7</td>\n",
       "      <td>synonymous_variant</td>\n",
       "      <td>T</td>\n",
       "      <td>1641</td>\n",
       "      <td>547</td>\n",
       "      <td>G</td>\n",
       "      <td>0</td>\n",
       "      <td>1108</td>\n",
       "      <td>TBF</td>\n",
       "      <td>-0.000923</td>\n",
       "      <td>0.000930</td>\n",
       "      <td>-0.992934</td>\n",
       "      <td>3.207446e-01</td>\n",
       "      <td>-0.002745</td>\n",
       "      <td>0.000899</td>\n",
       "      <td>7.002247e-01</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2973</th>\n",
       "      <td>22882</td>\n",
       "      <td>EGB_00027</td>\n",
       "      <td>VIII</td>\n",
       "      <td>240738</td>\n",
       "      <td>ERG7</td>\n",
       "      <td>synonymous_variant</td>\n",
       "      <td>C</td>\n",
       "      <td>1641</td>\n",
       "      <td>547</td>\n",
       "      <td>G</td>\n",
       "      <td>0</td>\n",
       "      <td>1109</td>\n",
       "      <td>CAFF</td>\n",
       "      <td>0.000483</td>\n",
       "      <td>0.000940</td>\n",
       "      <td>0.514277</td>\n",
       "      <td>6.070592e-01</td>\n",
       "      <td>-0.001358</td>\n",
       "      <td>0.002325</td>\n",
       "      <td>8.721453e-01</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2974</th>\n",
       "      <td>22882</td>\n",
       "      <td>EGB_00027</td>\n",
       "      <td>VIII</td>\n",
       "      <td>240738</td>\n",
       "      <td>ERG7</td>\n",
       "      <td>synonymous_variant</td>\n",
       "      <td>C</td>\n",
       "      <td>1641</td>\n",
       "      <td>547</td>\n",
       "      <td>G</td>\n",
       "      <td>0</td>\n",
       "      <td>1110</td>\n",
       "      <td>COCL</td>\n",
       "      <td>0.077358</td>\n",
       "      <td>0.003184</td>\n",
       "      <td>24.293832</td>\n",
       "      <td>5.558787e-130</td>\n",
       "      <td>0.071117</td>\n",
       "      <td>0.083599</td>\n",
       "      <td>6.397594e-128</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2975</th>\n",
       "      <td>22882</td>\n",
       "      <td>EGB_00027</td>\n",
       "      <td>VIII</td>\n",
       "      <td>240738</td>\n",
       "      <td>ERG7</td>\n",
       "      <td>synonymous_variant</td>\n",
       "      <td>C</td>\n",
       "      <td>1641</td>\n",
       "      <td>547</td>\n",
       "      <td>G</td>\n",
       "      <td>0</td>\n",
       "      <td>1111</td>\n",
       "      <td>LOV</td>\n",
       "      <td>-0.002131</td>\n",
       "      <td>0.001269</td>\n",
       "      <td>-1.679057</td>\n",
       "      <td>9.314405e-02</td>\n",
       "      <td>-0.004619</td>\n",
       "      <td>0.000357</td>\n",
       "      <td>4.096807e-01</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2976</th>\n",
       "      <td>22882</td>\n",
       "      <td>EGB_00027</td>\n",
       "      <td>VIII</td>\n",
       "      <td>240738</td>\n",
       "      <td>ERG7</td>\n",
       "      <td>synonymous_variant</td>\n",
       "      <td>C</td>\n",
       "      <td>1641</td>\n",
       "      <td>547</td>\n",
       "      <td>G</td>\n",
       "      <td>0</td>\n",
       "      <td>1112</td>\n",
       "      <td>NACL</td>\n",
       "      <td>0.000592</td>\n",
       "      <td>0.000937</td>\n",
       "      <td>0.631675</td>\n",
       "      <td>5.276006e-01</td>\n",
       "      <td>-0.001245</td>\n",
       "      <td>0.002429</td>\n",
       "      <td>8.389248e-01</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2977</th>\n",
       "      <td>22882</td>\n",
       "      <td>EGB_00027</td>\n",
       "      <td>VIII</td>\n",
       "      <td>240738</td>\n",
       "      <td>ERG7</td>\n",
       "      <td>synonymous_variant</td>\n",
       "      <td>C</td>\n",
       "      <td>1641</td>\n",
       "      <td>547</td>\n",
       "      <td>G</td>\n",
       "      <td>0</td>\n",
       "      <td>1113</td>\n",
       "      <td>SC</td>\n",
       "      <td>0.002595</td>\n",
       "      <td>0.000925</td>\n",
       "      <td>2.803975</td>\n",
       "      <td>5.048678e-03</td>\n",
       "      <td>0.000781</td>\n",
       "      <td>0.004408</td>\n",
       "      <td>6.856578e-02</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2978</th>\n",
       "      <td>22882</td>\n",
       "      <td>EGB_00027</td>\n",
       "      <td>VIII</td>\n",
       "      <td>240738</td>\n",
       "      <td>ERG7</td>\n",
       "      <td>synonymous_variant</td>\n",
       "      <td>C</td>\n",
       "      <td>1641</td>\n",
       "      <td>547</td>\n",
       "      <td>G</td>\n",
       "      <td>0</td>\n",
       "      <td>1114</td>\n",
       "      <td>TBF</td>\n",
       "      <td>0.000800</td>\n",
       "      <td>0.000705</td>\n",
       "      <td>1.135169</td>\n",
       "      <td>2.563071e-01</td>\n",
       "      <td>-0.000582</td>\n",
       "      <td>0.002182</td>\n",
       "      <td>6.433892e-01</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "      Unnamed: 0_x     var_id chrom  SNP_chr_pos  Gene         Consequence  \\\n",
       "2967         22872  EGB_00026  VIII       240738  ERG7  synonymous_variant   \n",
       "2968         22872  EGB_00026  VIII       240738  ERG7  synonymous_variant   \n",
       "2969         22872  EGB_00026  VIII       240738  ERG7  synonymous_variant   \n",
       "2970         22872  EGB_00026  VIII       240738  ERG7  synonymous_variant   \n",
       "2971         22872  EGB_00026  VIII       240738  ERG7  synonymous_variant   \n",
       "2972         22872  EGB_00026  VIII       240738  ERG7  synonymous_variant   \n",
       "2973         22882  EGB_00027  VIII       240738  ERG7  synonymous_variant   \n",
       "2974         22882  EGB_00027  VIII       240738  ERG7  synonymous_variant   \n",
       "2975         22882  EGB_00027  VIII       240738  ERG7  synonymous_variant   \n",
       "2976         22882  EGB_00027  VIII       240738  ERG7  synonymous_variant   \n",
       "2977         22882  EGB_00027  VIII       240738  ERG7  synonymous_variant   \n",
       "2978         22882  EGB_00027  VIII       240738  ERG7  synonymous_variant   \n",
       "\n",
       "     ALT CDS_position Protein_position Amino_acids  DISTANCE  Unnamed: 0_y  \\\n",
       "2967   T         1641              547           G         0          1103   \n",
       "2968   T         1641              547           G         0          1104   \n",
       "2969   T         1641              547           G         0          1105   \n",
       "2970   T         1641              547           G         0          1106   \n",
       "2971   T         1641              547           G         0          1107   \n",
       "2972   T         1641              547           G         0          1108   \n",
       "2973   C         1641              547           G         0          1109   \n",
       "2974   C         1641              547           G         0          1110   \n",
       "2975   C         1641              547           G         0          1111   \n",
       "2976   C         1641              547           G         0          1112   \n",
       "2977   C         1641              547           G         0          1113   \n",
       "2978   C         1641              547           G         0          1114   \n",
       "\n",
       "     condition      coef   std err          t           pval  Conf. Int. Low  \\\n",
       "2967      CAFF  0.000308  0.001230   0.250274   8.023761e-01       -0.002102   \n",
       "2968      COCL -0.017969  0.011281  -1.592894   1.111872e-01       -0.040079   \n",
       "2969       LOV -0.001969  0.001845  -1.067331   2.858251e-01       -0.005585   \n",
       "2970      NACL  0.001724  0.001374   1.255040   2.094672e-01       -0.000968   \n",
       "2971        SC -0.000612  0.001205  -0.508176   6.113309e-01       -0.002973   \n",
       "2972       TBF -0.000923  0.000930  -0.992934   3.207446e-01       -0.002745   \n",
       "2973      CAFF  0.000483  0.000940   0.514277   6.070592e-01       -0.001358   \n",
       "2974      COCL  0.077358  0.003184  24.293832  5.558787e-130        0.071117   \n",
       "2975       LOV -0.002131  0.001269  -1.679057   9.314405e-02       -0.004619   \n",
       "2976      NACL  0.000592  0.000937   0.631675   5.276006e-01       -0.001245   \n",
       "2977        SC  0.002595  0.000925   2.803975   5.048678e-03        0.000781   \n",
       "2978       TBF  0.000800  0.000705   1.135169   2.563071e-01       -0.000582   \n",
       "\n",
       "      Conf. Int. Upp.           padj  \n",
       "2967         0.002718   9.525466e-01  \n",
       "2968         0.004141   4.481939e-01  \n",
       "2969         0.001647   6.706642e-01  \n",
       "2970         0.004416   5.893554e-01  \n",
       "2971         0.001749   8.749904e-01  \n",
       "2972         0.000899   7.002247e-01  \n",
       "2973         0.002325   8.721453e-01  \n",
       "2974         0.083599  6.397594e-128  \n",
       "2975         0.000357   4.096807e-01  \n",
       "2976         0.002429   8.389248e-01  \n",
       "2977         0.004408   6.856578e-02  \n",
       "2978         0.002182   6.433892e-01  "
      ]
     },
     "execution_count": 15,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "#ERG20 site with two synonymous variants, one with a significant effect in cobalt chloride and one without\n",
    "\n",
    "chrom = 'VIII'\n",
    "pos = 240738\n",
    "\n",
    "annotated_fits[(annotated_fits['chrom']==chrom)&(annotated_fits['SNP_chr_pos']==pos)]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 16,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Unnamed: 0_x</th>\n",
       "      <th>var_id</th>\n",
       "      <th>chrom</th>\n",
       "      <th>SNP_chr_pos</th>\n",
       "      <th>Gene</th>\n",
       "      <th>Consequence</th>\n",
       "      <th>ALT</th>\n",
       "      <th>CDS_position</th>\n",
       "      <th>Protein_position</th>\n",
       "      <th>Amino_acids</th>\n",
       "      <th>DISTANCE</th>\n",
       "      <th>Unnamed: 0_y</th>\n",
       "      <th>condition</th>\n",
       "      <th>coef</th>\n",
       "      <th>std err</th>\n",
       "      <th>t</th>\n",
       "      <th>pval</th>\n",
       "      <th>Conf. Int. Low</th>\n",
       "      <th>Conf. Int. Upp.</th>\n",
       "      <th>padj</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>2797</th>\n",
       "      <td>22587</td>\n",
       "      <td>ERG_00825</td>\n",
       "      <td>VIII</td>\n",
       "      <td>239784</td>\n",
       "      <td>ERG7</td>\n",
       "      <td>synonymous_variant</td>\n",
       "      <td>T</td>\n",
       "      <td>687</td>\n",
       "      <td>229</td>\n",
       "      <td>G</td>\n",
       "      <td>0</td>\n",
       "      <td>4823</td>\n",
       "      <td>CAFF</td>\n",
       "      <td>0.000380</td>\n",
       "      <td>0.001153</td>\n",
       "      <td>0.329555</td>\n",
       "      <td>7.417367e-01</td>\n",
       "      <td>-0.001880</td>\n",
       "      <td>0.002640</td>\n",
       "      <td>9.280483e-01</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2798</th>\n",
       "      <td>22587</td>\n",
       "      <td>ERG_00825</td>\n",
       "      <td>VIII</td>\n",
       "      <td>239784</td>\n",
       "      <td>ERG7</td>\n",
       "      <td>synonymous_variant</td>\n",
       "      <td>T</td>\n",
       "      <td>687</td>\n",
       "      <td>229</td>\n",
       "      <td>G</td>\n",
       "      <td>0</td>\n",
       "      <td>4824</td>\n",
       "      <td>COCL</td>\n",
       "      <td>0.024358</td>\n",
       "      <td>0.003657</td>\n",
       "      <td>6.660545</td>\n",
       "      <td>2.742551e-11</td>\n",
       "      <td>0.017190</td>\n",
       "      <td>0.031526</td>\n",
       "      <td>1.070430e-09</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2799</th>\n",
       "      <td>22587</td>\n",
       "      <td>ERG_00825</td>\n",
       "      <td>VIII</td>\n",
       "      <td>239784</td>\n",
       "      <td>ERG7</td>\n",
       "      <td>synonymous_variant</td>\n",
       "      <td>T</td>\n",
       "      <td>687</td>\n",
       "      <td>229</td>\n",
       "      <td>G</td>\n",
       "      <td>0</td>\n",
       "      <td>4825</td>\n",
       "      <td>LOV</td>\n",
       "      <td>0.001968</td>\n",
       "      <td>0.001365</td>\n",
       "      <td>1.442124</td>\n",
       "      <td>1.492707e-01</td>\n",
       "      <td>-0.000707</td>\n",
       "      <td>0.004643</td>\n",
       "      <td>5.137419e-01</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2800</th>\n",
       "      <td>22587</td>\n",
       "      <td>ERG_00825</td>\n",
       "      <td>VIII</td>\n",
       "      <td>239784</td>\n",
       "      <td>ERG7</td>\n",
       "      <td>synonymous_variant</td>\n",
       "      <td>T</td>\n",
       "      <td>687</td>\n",
       "      <td>229</td>\n",
       "      <td>G</td>\n",
       "      <td>0</td>\n",
       "      <td>4826</td>\n",
       "      <td>NACL</td>\n",
       "      <td>0.000148</td>\n",
       "      <td>0.001218</td>\n",
       "      <td>0.121817</td>\n",
       "      <td>9.030441e-01</td>\n",
       "      <td>-0.002239</td>\n",
       "      <td>0.002536</td>\n",
       "      <td>9.809568e-01</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2801</th>\n",
       "      <td>22587</td>\n",
       "      <td>ERG_00825</td>\n",
       "      <td>VIII</td>\n",
       "      <td>239784</td>\n",
       "      <td>ERG7</td>\n",
       "      <td>synonymous_variant</td>\n",
       "      <td>T</td>\n",
       "      <td>687</td>\n",
       "      <td>229</td>\n",
       "      <td>G</td>\n",
       "      <td>0</td>\n",
       "      <td>4827</td>\n",
       "      <td>SC</td>\n",
       "      <td>-0.000506</td>\n",
       "      <td>0.001144</td>\n",
       "      <td>-0.442110</td>\n",
       "      <td>6.584108e-01</td>\n",
       "      <td>-0.002747</td>\n",
       "      <td>0.001736</td>\n",
       "      <td>8.962175e-01</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2802</th>\n",
       "      <td>22587</td>\n",
       "      <td>ERG_00825</td>\n",
       "      <td>VIII</td>\n",
       "      <td>239784</td>\n",
       "      <td>ERG7</td>\n",
       "      <td>synonymous_variant</td>\n",
       "      <td>T</td>\n",
       "      <td>687</td>\n",
       "      <td>229</td>\n",
       "      <td>G</td>\n",
       "      <td>0</td>\n",
       "      <td>4828</td>\n",
       "      <td>TBF</td>\n",
       "      <td>-0.000172</td>\n",
       "      <td>0.000890</td>\n",
       "      <td>-0.193503</td>\n",
       "      <td>8.465654e-01</td>\n",
       "      <td>-0.001916</td>\n",
       "      <td>0.001572</td>\n",
       "      <td>9.674885e-01</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2803</th>\n",
       "      <td>22597</td>\n",
       "      <td>ERG_00826</td>\n",
       "      <td>VIII</td>\n",
       "      <td>239784</td>\n",
       "      <td>ERG7</td>\n",
       "      <td>synonymous_variant</td>\n",
       "      <td>C</td>\n",
       "      <td>687</td>\n",
       "      <td>229</td>\n",
       "      <td>G</td>\n",
       "      <td>0</td>\n",
       "      <td>4829</td>\n",
       "      <td>CAFF</td>\n",
       "      <td>0.000555</td>\n",
       "      <td>0.001289</td>\n",
       "      <td>0.430702</td>\n",
       "      <td>6.666863e-01</td>\n",
       "      <td>-0.001972</td>\n",
       "      <td>0.003082</td>\n",
       "      <td>8.992584e-01</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2804</th>\n",
       "      <td>22597</td>\n",
       "      <td>ERG_00826</td>\n",
       "      <td>VIII</td>\n",
       "      <td>239784</td>\n",
       "      <td>ERG7</td>\n",
       "      <td>synonymous_variant</td>\n",
       "      <td>C</td>\n",
       "      <td>687</td>\n",
       "      <td>229</td>\n",
       "      <td>G</td>\n",
       "      <td>0</td>\n",
       "      <td>4830</td>\n",
       "      <td>COCL</td>\n",
       "      <td>0.010895</td>\n",
       "      <td>0.005338</td>\n",
       "      <td>2.041133</td>\n",
       "      <td>4.124030e-02</td>\n",
       "      <td>0.000433</td>\n",
       "      <td>0.021358</td>\n",
       "      <td>2.720163e-01</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2805</th>\n",
       "      <td>22597</td>\n",
       "      <td>ERG_00826</td>\n",
       "      <td>VIII</td>\n",
       "      <td>239784</td>\n",
       "      <td>ERG7</td>\n",
       "      <td>synonymous_variant</td>\n",
       "      <td>C</td>\n",
       "      <td>687</td>\n",
       "      <td>229</td>\n",
       "      <td>G</td>\n",
       "      <td>0</td>\n",
       "      <td>4831</td>\n",
       "      <td>LOV</td>\n",
       "      <td>-0.000919</td>\n",
       "      <td>0.001570</td>\n",
       "      <td>-0.585207</td>\n",
       "      <td>5.584102e-01</td>\n",
       "      <td>-0.003995</td>\n",
       "      <td>0.002158</td>\n",
       "      <td>8.500328e-01</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2806</th>\n",
       "      <td>22597</td>\n",
       "      <td>ERG_00826</td>\n",
       "      <td>VIII</td>\n",
       "      <td>239784</td>\n",
       "      <td>ERG7</td>\n",
       "      <td>synonymous_variant</td>\n",
       "      <td>C</td>\n",
       "      <td>687</td>\n",
       "      <td>229</td>\n",
       "      <td>G</td>\n",
       "      <td>0</td>\n",
       "      <td>4832</td>\n",
       "      <td>NACL</td>\n",
       "      <td>-0.000744</td>\n",
       "      <td>0.001268</td>\n",
       "      <td>-0.586711</td>\n",
       "      <td>5.573989e-01</td>\n",
       "      <td>-0.003229</td>\n",
       "      <td>0.001741</td>\n",
       "      <td>8.495364e-01</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2807</th>\n",
       "      <td>22597</td>\n",
       "      <td>ERG_00826</td>\n",
       "      <td>VIII</td>\n",
       "      <td>239784</td>\n",
       "      <td>ERG7</td>\n",
       "      <td>synonymous_variant</td>\n",
       "      <td>C</td>\n",
       "      <td>687</td>\n",
       "      <td>229</td>\n",
       "      <td>G</td>\n",
       "      <td>0</td>\n",
       "      <td>4833</td>\n",
       "      <td>SC</td>\n",
       "      <td>0.000379</td>\n",
       "      <td>0.001243</td>\n",
       "      <td>0.304539</td>\n",
       "      <td>7.607181e-01</td>\n",
       "      <td>-0.002058</td>\n",
       "      <td>0.002815</td>\n",
       "      <td>9.354108e-01</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2808</th>\n",
       "      <td>22597</td>\n",
       "      <td>ERG_00826</td>\n",
       "      <td>VIII</td>\n",
       "      <td>239784</td>\n",
       "      <td>ERG7</td>\n",
       "      <td>synonymous_variant</td>\n",
       "      <td>C</td>\n",
       "      <td>687</td>\n",
       "      <td>229</td>\n",
       "      <td>G</td>\n",
       "      <td>0</td>\n",
       "      <td>4834</td>\n",
       "      <td>TBF</td>\n",
       "      <td>-0.000829</td>\n",
       "      <td>0.000984</td>\n",
       "      <td>-0.842535</td>\n",
       "      <td>3.994904e-01</td>\n",
       "      <td>-0.002757</td>\n",
       "      <td>0.001099</td>\n",
       "      <td>7.622672e-01</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "      Unnamed: 0_x     var_id chrom  SNP_chr_pos  Gene         Consequence  \\\n",
       "2797         22587  ERG_00825  VIII       239784  ERG7  synonymous_variant   \n",
       "2798         22587  ERG_00825  VIII       239784  ERG7  synonymous_variant   \n",
       "2799         22587  ERG_00825  VIII       239784  ERG7  synonymous_variant   \n",
       "2800         22587  ERG_00825  VIII       239784  ERG7  synonymous_variant   \n",
       "2801         22587  ERG_00825  VIII       239784  ERG7  synonymous_variant   \n",
       "2802         22587  ERG_00825  VIII       239784  ERG7  synonymous_variant   \n",
       "2803         22597  ERG_00826  VIII       239784  ERG7  synonymous_variant   \n",
       "2804         22597  ERG_00826  VIII       239784  ERG7  synonymous_variant   \n",
       "2805         22597  ERG_00826  VIII       239784  ERG7  synonymous_variant   \n",
       "2806         22597  ERG_00826  VIII       239784  ERG7  synonymous_variant   \n",
       "2807         22597  ERG_00826  VIII       239784  ERG7  synonymous_variant   \n",
       "2808         22597  ERG_00826  VIII       239784  ERG7  synonymous_variant   \n",
       "\n",
       "     ALT CDS_position Protein_position Amino_acids  DISTANCE  Unnamed: 0_y  \\\n",
       "2797   T          687              229           G         0          4823   \n",
       "2798   T          687              229           G         0          4824   \n",
       "2799   T          687              229           G         0          4825   \n",
       "2800   T          687              229           G         0          4826   \n",
       "2801   T          687              229           G         0          4827   \n",
       "2802   T          687              229           G         0          4828   \n",
       "2803   C          687              229           G         0          4829   \n",
       "2804   C          687              229           G         0          4830   \n",
       "2805   C          687              229           G         0          4831   \n",
       "2806   C          687              229           G         0          4832   \n",
       "2807   C          687              229           G         0          4833   \n",
       "2808   C          687              229           G         0          4834   \n",
       "\n",
       "     condition      coef   std err         t          pval  Conf. Int. Low  \\\n",
       "2797      CAFF  0.000380  0.001153  0.329555  7.417367e-01       -0.001880   \n",
       "2798      COCL  0.024358  0.003657  6.660545  2.742551e-11        0.017190   \n",
       "2799       LOV  0.001968  0.001365  1.442124  1.492707e-01       -0.000707   \n",
       "2800      NACL  0.000148  0.001218  0.121817  9.030441e-01       -0.002239   \n",
       "2801        SC -0.000506  0.001144 -0.442110  6.584108e-01       -0.002747   \n",
       "2802       TBF -0.000172  0.000890 -0.193503  8.465654e-01       -0.001916   \n",
       "2803      CAFF  0.000555  0.001289  0.430702  6.666863e-01       -0.001972   \n",
       "2804      COCL  0.010895  0.005338  2.041133  4.124030e-02        0.000433   \n",
       "2805       LOV -0.000919  0.001570 -0.585207  5.584102e-01       -0.003995   \n",
       "2806      NACL -0.000744  0.001268 -0.586711  5.573989e-01       -0.003229   \n",
       "2807        SC  0.000379  0.001243  0.304539  7.607181e-01       -0.002058   \n",
       "2808       TBF -0.000829  0.000984 -0.842535  3.994904e-01       -0.002757   \n",
       "\n",
       "      Conf. Int. Upp.          padj  \n",
       "2797         0.002640  9.280483e-01  \n",
       "2798         0.031526  1.070430e-09  \n",
       "2799         0.004643  5.137419e-01  \n",
       "2800         0.002536  9.809568e-01  \n",
       "2801         0.001736  8.962175e-01  \n",
       "2802         0.001572  9.674885e-01  \n",
       "2803         0.003082  8.992584e-01  \n",
       "2804         0.021358  2.720163e-01  \n",
       "2805         0.002158  8.500328e-01  \n",
       "2806         0.001741  8.495364e-01  \n",
       "2807         0.002815  9.354108e-01  \n",
       "2808         0.001099  7.622672e-01  "
      ]
     },
     "execution_count": 16,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "#ERG7 site with two synonymous variants, one with a significant effect in cobalt chloride and one without\n",
    "\n",
    "chrom = 'VIII'\n",
    "pos = 239784\n",
    "\n",
    "annotated_fits[(annotated_fits['chrom']==chrom)&(annotated_fits['SNP_chr_pos']==pos)]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 17,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Unnamed: 0_x</th>\n",
       "      <th>var_id</th>\n",
       "      <th>chrom</th>\n",
       "      <th>SNP_chr_pos</th>\n",
       "      <th>Gene</th>\n",
       "      <th>Consequence</th>\n",
       "      <th>ALT</th>\n",
       "      <th>CDS_position</th>\n",
       "      <th>Protein_position</th>\n",
       "      <th>Amino_acids</th>\n",
       "      <th>DISTANCE</th>\n",
       "      <th>Unnamed: 0_y</th>\n",
       "      <th>condition</th>\n",
       "      <th>coef</th>\n",
       "      <th>std err</th>\n",
       "      <th>t</th>\n",
       "      <th>pval</th>\n",
       "      <th>Conf. Int. Low</th>\n",
       "      <th>Conf. Int. Upp.</th>\n",
       "      <th>padj</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>2587</th>\n",
       "      <td>22209</td>\n",
       "      <td>EGA_00016</td>\n",
       "      <td>VIII</td>\n",
       "      <td>239202</td>\n",
       "      <td>ERG7</td>\n",
       "      <td>synonymous_variant</td>\n",
       "      <td>G</td>\n",
       "      <td>105</td>\n",
       "      <td>35</td>\n",
       "      <td>T</td>\n",
       "      <td>0</td>\n",
       "      <td>954</td>\n",
       "      <td>CAFF</td>\n",
       "      <td>0.000447</td>\n",
       "      <td>0.000685</td>\n",
       "      <td>0.652787</td>\n",
       "      <td>5.138950e-01</td>\n",
       "      <td>-0.000896</td>\n",
       "      <td>0.001791</td>\n",
       "      <td>8.317516e-01</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2588</th>\n",
       "      <td>22209</td>\n",
       "      <td>EGA_00016</td>\n",
       "      <td>VIII</td>\n",
       "      <td>239202</td>\n",
       "      <td>ERG7</td>\n",
       "      <td>synonymous_variant</td>\n",
       "      <td>G</td>\n",
       "      <td>105</td>\n",
       "      <td>35</td>\n",
       "      <td>T</td>\n",
       "      <td>0</td>\n",
       "      <td>955</td>\n",
       "      <td>COCL</td>\n",
       "      <td>0.001721</td>\n",
       "      <td>0.002127</td>\n",
       "      <td>0.809075</td>\n",
       "      <td>4.184742e-01</td>\n",
       "      <td>-0.002448</td>\n",
       "      <td>0.005889</td>\n",
       "      <td>7.759867e-01</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2589</th>\n",
       "      <td>22209</td>\n",
       "      <td>EGA_00016</td>\n",
       "      <td>VIII</td>\n",
       "      <td>239202</td>\n",
       "      <td>ERG7</td>\n",
       "      <td>synonymous_variant</td>\n",
       "      <td>G</td>\n",
       "      <td>105</td>\n",
       "      <td>35</td>\n",
       "      <td>T</td>\n",
       "      <td>0</td>\n",
       "      <td>956</td>\n",
       "      <td>LOV</td>\n",
       "      <td>-0.000548</td>\n",
       "      <td>0.000817</td>\n",
       "      <td>-0.671067</td>\n",
       "      <td>5.021796e-01</td>\n",
       "      <td>-0.002149</td>\n",
       "      <td>0.001053</td>\n",
       "      <td>8.258803e-01</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2590</th>\n",
       "      <td>22209</td>\n",
       "      <td>EGA_00016</td>\n",
       "      <td>VIII</td>\n",
       "      <td>239202</td>\n",
       "      <td>ERG7</td>\n",
       "      <td>synonymous_variant</td>\n",
       "      <td>G</td>\n",
       "      <td>105</td>\n",
       "      <td>35</td>\n",
       "      <td>T</td>\n",
       "      <td>0</td>\n",
       "      <td>957</td>\n",
       "      <td>NACL</td>\n",
       "      <td>-0.000074</td>\n",
       "      <td>0.000590</td>\n",
       "      <td>-0.125946</td>\n",
       "      <td>8.997749e-01</td>\n",
       "      <td>-0.001231</td>\n",
       "      <td>0.001083</td>\n",
       "      <td>9.799139e-01</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2591</th>\n",
       "      <td>22209</td>\n",
       "      <td>EGA_00016</td>\n",
       "      <td>VIII</td>\n",
       "      <td>239202</td>\n",
       "      <td>ERG7</td>\n",
       "      <td>synonymous_variant</td>\n",
       "      <td>G</td>\n",
       "      <td>105</td>\n",
       "      <td>35</td>\n",
       "      <td>T</td>\n",
       "      <td>0</td>\n",
       "      <td>958</td>\n",
       "      <td>SC</td>\n",
       "      <td>0.000454</td>\n",
       "      <td>0.000579</td>\n",
       "      <td>0.784966</td>\n",
       "      <td>4.324754e-01</td>\n",
       "      <td>-0.000680</td>\n",
       "      <td>0.001588</td>\n",
       "      <td>7.853715e-01</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2592</th>\n",
       "      <td>22209</td>\n",
       "      <td>EGA_00016</td>\n",
       "      <td>VIII</td>\n",
       "      <td>239202</td>\n",
       "      <td>ERG7</td>\n",
       "      <td>synonymous_variant</td>\n",
       "      <td>G</td>\n",
       "      <td>105</td>\n",
       "      <td>35</td>\n",
       "      <td>T</td>\n",
       "      <td>0</td>\n",
       "      <td>959</td>\n",
       "      <td>TBF</td>\n",
       "      <td>0.000431</td>\n",
       "      <td>0.000562</td>\n",
       "      <td>0.767981</td>\n",
       "      <td>4.425002e-01</td>\n",
       "      <td>-0.000669</td>\n",
       "      <td>0.001532</td>\n",
       "      <td>7.908976e-01</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2593</th>\n",
       "      <td>22220</td>\n",
       "      <td>EGA_00017</td>\n",
       "      <td>VIII</td>\n",
       "      <td>239202</td>\n",
       "      <td>ERG7</td>\n",
       "      <td>synonymous_variant</td>\n",
       "      <td>A</td>\n",
       "      <td>105</td>\n",
       "      <td>35</td>\n",
       "      <td>T</td>\n",
       "      <td>0</td>\n",
       "      <td>960</td>\n",
       "      <td>CAFF</td>\n",
       "      <td>-0.000950</td>\n",
       "      <td>0.000558</td>\n",
       "      <td>-1.701549</td>\n",
       "      <td>8.884315e-02</td>\n",
       "      <td>-0.002044</td>\n",
       "      <td>0.000144</td>\n",
       "      <td>4.015836e-01</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2594</th>\n",
       "      <td>22220</td>\n",
       "      <td>EGA_00017</td>\n",
       "      <td>VIII</td>\n",
       "      <td>239202</td>\n",
       "      <td>ERG7</td>\n",
       "      <td>synonymous_variant</td>\n",
       "      <td>A</td>\n",
       "      <td>105</td>\n",
       "      <td>35</td>\n",
       "      <td>T</td>\n",
       "      <td>0</td>\n",
       "      <td>961</td>\n",
       "      <td>COCL</td>\n",
       "      <td>0.011834</td>\n",
       "      <td>0.001867</td>\n",
       "      <td>6.340083</td>\n",
       "      <td>2.306416e-10</td>\n",
       "      <td>0.008176</td>\n",
       "      <td>0.015492</td>\n",
       "      <td>8.591161e-09</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2595</th>\n",
       "      <td>22220</td>\n",
       "      <td>EGA_00017</td>\n",
       "      <td>VIII</td>\n",
       "      <td>239202</td>\n",
       "      <td>ERG7</td>\n",
       "      <td>synonymous_variant</td>\n",
       "      <td>A</td>\n",
       "      <td>105</td>\n",
       "      <td>35</td>\n",
       "      <td>T</td>\n",
       "      <td>0</td>\n",
       "      <td>962</td>\n",
       "      <td>LOV</td>\n",
       "      <td>-0.002570</td>\n",
       "      <td>0.000762</td>\n",
       "      <td>-3.371934</td>\n",
       "      <td>7.467128e-04</td>\n",
       "      <td>-0.004064</td>\n",
       "      <td>-0.001076</td>\n",
       "      <td>1.417176e-02</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2596</th>\n",
       "      <td>22220</td>\n",
       "      <td>EGA_00017</td>\n",
       "      <td>VIII</td>\n",
       "      <td>239202</td>\n",
       "      <td>ERG7</td>\n",
       "      <td>synonymous_variant</td>\n",
       "      <td>A</td>\n",
       "      <td>105</td>\n",
       "      <td>35</td>\n",
       "      <td>T</td>\n",
       "      <td>0</td>\n",
       "      <td>963</td>\n",
       "      <td>NACL</td>\n",
       "      <td>-0.001395</td>\n",
       "      <td>0.000542</td>\n",
       "      <td>-2.574788</td>\n",
       "      <td>1.003160e-02</td>\n",
       "      <td>-0.002458</td>\n",
       "      <td>-0.000333</td>\n",
       "      <td>1.132646e-01</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2597</th>\n",
       "      <td>22220</td>\n",
       "      <td>EGA_00017</td>\n",
       "      <td>VIII</td>\n",
       "      <td>239202</td>\n",
       "      <td>ERG7</td>\n",
       "      <td>synonymous_variant</td>\n",
       "      <td>A</td>\n",
       "      <td>105</td>\n",
       "      <td>35</td>\n",
       "      <td>T</td>\n",
       "      <td>0</td>\n",
       "      <td>964</td>\n",
       "      <td>SC</td>\n",
       "      <td>0.001544</td>\n",
       "      <td>0.000544</td>\n",
       "      <td>2.836970</td>\n",
       "      <td>4.555324e-03</td>\n",
       "      <td>0.000477</td>\n",
       "      <td>0.002611</td>\n",
       "      <td>6.340023e-02</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2598</th>\n",
       "      <td>22220</td>\n",
       "      <td>EGA_00017</td>\n",
       "      <td>VIII</td>\n",
       "      <td>239202</td>\n",
       "      <td>ERG7</td>\n",
       "      <td>synonymous_variant</td>\n",
       "      <td>A</td>\n",
       "      <td>105</td>\n",
       "      <td>35</td>\n",
       "      <td>T</td>\n",
       "      <td>0</td>\n",
       "      <td>965</td>\n",
       "      <td>TBF</td>\n",
       "      <td>-0.000031</td>\n",
       "      <td>0.000412</td>\n",
       "      <td>-0.074551</td>\n",
       "      <td>9.405723e-01</td>\n",
       "      <td>-0.000839</td>\n",
       "      <td>0.000777</td>\n",
       "      <td>9.892226e-01</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "      Unnamed: 0_x     var_id chrom  SNP_chr_pos  Gene         Consequence  \\\n",
       "2587         22209  EGA_00016  VIII       239202  ERG7  synonymous_variant   \n",
       "2588         22209  EGA_00016  VIII       239202  ERG7  synonymous_variant   \n",
       "2589         22209  EGA_00016  VIII       239202  ERG7  synonymous_variant   \n",
       "2590         22209  EGA_00016  VIII       239202  ERG7  synonymous_variant   \n",
       "2591         22209  EGA_00016  VIII       239202  ERG7  synonymous_variant   \n",
       "2592         22209  EGA_00016  VIII       239202  ERG7  synonymous_variant   \n",
       "2593         22220  EGA_00017  VIII       239202  ERG7  synonymous_variant   \n",
       "2594         22220  EGA_00017  VIII       239202  ERG7  synonymous_variant   \n",
       "2595         22220  EGA_00017  VIII       239202  ERG7  synonymous_variant   \n",
       "2596         22220  EGA_00017  VIII       239202  ERG7  synonymous_variant   \n",
       "2597         22220  EGA_00017  VIII       239202  ERG7  synonymous_variant   \n",
       "2598         22220  EGA_00017  VIII       239202  ERG7  synonymous_variant   \n",
       "\n",
       "     ALT CDS_position Protein_position Amino_acids  DISTANCE  Unnamed: 0_y  \\\n",
       "2587   G          105               35           T         0           954   \n",
       "2588   G          105               35           T         0           955   \n",
       "2589   G          105               35           T         0           956   \n",
       "2590   G          105               35           T         0           957   \n",
       "2591   G          105               35           T         0           958   \n",
       "2592   G          105               35           T         0           959   \n",
       "2593   A          105               35           T         0           960   \n",
       "2594   A          105               35           T         0           961   \n",
       "2595   A          105               35           T         0           962   \n",
       "2596   A          105               35           T         0           963   \n",
       "2597   A          105               35           T         0           964   \n",
       "2598   A          105               35           T         0           965   \n",
       "\n",
       "     condition      coef   std err         t          pval  Conf. Int. Low  \\\n",
       "2587      CAFF  0.000447  0.000685  0.652787  5.138950e-01       -0.000896   \n",
       "2588      COCL  0.001721  0.002127  0.809075  4.184742e-01       -0.002448   \n",
       "2589       LOV -0.000548  0.000817 -0.671067  5.021796e-01       -0.002149   \n",
       "2590      NACL -0.000074  0.000590 -0.125946  8.997749e-01       -0.001231   \n",
       "2591        SC  0.000454  0.000579  0.784966  4.324754e-01       -0.000680   \n",
       "2592       TBF  0.000431  0.000562  0.767981  4.425002e-01       -0.000669   \n",
       "2593      CAFF -0.000950  0.000558 -1.701549  8.884315e-02       -0.002044   \n",
       "2594      COCL  0.011834  0.001867  6.340083  2.306416e-10        0.008176   \n",
       "2595       LOV -0.002570  0.000762 -3.371934  7.467128e-04       -0.004064   \n",
       "2596      NACL -0.001395  0.000542 -2.574788  1.003160e-02       -0.002458   \n",
       "2597        SC  0.001544  0.000544  2.836970  4.555324e-03        0.000477   \n",
       "2598       TBF -0.000031  0.000412 -0.074551  9.405723e-01       -0.000839   \n",
       "\n",
       "      Conf. Int. Upp.          padj  \n",
       "2587         0.001791  8.317516e-01  \n",
       "2588         0.005889  7.759867e-01  \n",
       "2589         0.001053  8.258803e-01  \n",
       "2590         0.001083  9.799139e-01  \n",
       "2591         0.001588  7.853715e-01  \n",
       "2592         0.001532  7.908976e-01  \n",
       "2593         0.000144  4.015836e-01  \n",
       "2594         0.015492  8.591161e-09  \n",
       "2595        -0.001076  1.417176e-02  \n",
       "2596        -0.000333  1.132646e-01  \n",
       "2597         0.002611  6.340023e-02  \n",
       "2598         0.000777  9.892226e-01  "
      ]
     },
     "execution_count": 17,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "#ERG7 site with two synonymous variants, one with a significant effect in cobalt chloride and one without\n",
    "\n",
    "chrom = 'VIII'\n",
    "pos = 239202\n",
    "\n",
    "annotated_fits[(annotated_fits['chrom']==chrom)&(annotated_fits['SNP_chr_pos']==pos)]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 18,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Unnamed: 0_x</th>\n",
       "      <th>var_id</th>\n",
       "      <th>chrom</th>\n",
       "      <th>SNP_chr_pos</th>\n",
       "      <th>Gene</th>\n",
       "      <th>Consequence</th>\n",
       "      <th>ALT</th>\n",
       "      <th>CDS_position</th>\n",
       "      <th>Protein_position</th>\n",
       "      <th>Amino_acids</th>\n",
       "      <th>DISTANCE</th>\n",
       "      <th>Unnamed: 0_y</th>\n",
       "      <th>condition</th>\n",
       "      <th>coef</th>\n",
       "      <th>std err</th>\n",
       "      <th>t</th>\n",
       "      <th>pval</th>\n",
       "      <th>Conf. Int. Low</th>\n",
       "      <th>Conf. Int. Upp.</th>\n",
       "      <th>padj</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>2967</th>\n",
       "      <td>22872</td>\n",
       "      <td>EGB_00026</td>\n",
       "      <td>VIII</td>\n",
       "      <td>240738</td>\n",
       "      <td>ERG7</td>\n",
       "      <td>synonymous_variant</td>\n",
       "      <td>T</td>\n",
       "      <td>1641</td>\n",
       "      <td>547</td>\n",
       "      <td>G</td>\n",
       "      <td>0</td>\n",
       "      <td>1103</td>\n",
       "      <td>CAFF</td>\n",
       "      <td>0.000308</td>\n",
       "      <td>0.001230</td>\n",
       "      <td>0.250274</td>\n",
       "      <td>8.023761e-01</td>\n",
       "      <td>-0.002102</td>\n",
       "      <td>0.002718</td>\n",
       "      <td>9.525466e-01</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2968</th>\n",
       "      <td>22872</td>\n",
       "      <td>EGB_00026</td>\n",
       "      <td>VIII</td>\n",
       "      <td>240738</td>\n",
       "      <td>ERG7</td>\n",
       "      <td>synonymous_variant</td>\n",
       "      <td>T</td>\n",
       "      <td>1641</td>\n",
       "      <td>547</td>\n",
       "      <td>G</td>\n",
       "      <td>0</td>\n",
       "      <td>1104</td>\n",
       "      <td>COCL</td>\n",
       "      <td>-0.017969</td>\n",
       "      <td>0.011281</td>\n",
       "      <td>-1.592894</td>\n",
       "      <td>1.111872e-01</td>\n",
       "      <td>-0.040079</td>\n",
       "      <td>0.004141</td>\n",
       "      <td>4.481939e-01</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2969</th>\n",
       "      <td>22872</td>\n",
       "      <td>EGB_00026</td>\n",
       "      <td>VIII</td>\n",
       "      <td>240738</td>\n",
       "      <td>ERG7</td>\n",
       "      <td>synonymous_variant</td>\n",
       "      <td>T</td>\n",
       "      <td>1641</td>\n",
       "      <td>547</td>\n",
       "      <td>G</td>\n",
       "      <td>0</td>\n",
       "      <td>1105</td>\n",
       "      <td>LOV</td>\n",
       "      <td>-0.001969</td>\n",
       "      <td>0.001845</td>\n",
       "      <td>-1.067331</td>\n",
       "      <td>2.858251e-01</td>\n",
       "      <td>-0.005585</td>\n",
       "      <td>0.001647</td>\n",
       "      <td>6.706642e-01</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2970</th>\n",
       "      <td>22872</td>\n",
       "      <td>EGB_00026</td>\n",
       "      <td>VIII</td>\n",
       "      <td>240738</td>\n",
       "      <td>ERG7</td>\n",
       "      <td>synonymous_variant</td>\n",
       "      <td>T</td>\n",
       "      <td>1641</td>\n",
       "      <td>547</td>\n",
       "      <td>G</td>\n",
       "      <td>0</td>\n",
       "      <td>1106</td>\n",
       "      <td>NACL</td>\n",
       "      <td>0.001724</td>\n",
       "      <td>0.001374</td>\n",
       "      <td>1.255040</td>\n",
       "      <td>2.094672e-01</td>\n",
       "      <td>-0.000968</td>\n",
       "      <td>0.004416</td>\n",
       "      <td>5.893554e-01</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2971</th>\n",
       "      <td>22872</td>\n",
       "      <td>EGB_00026</td>\n",
       "      <td>VIII</td>\n",
       "      <td>240738</td>\n",
       "      <td>ERG7</td>\n",
       "      <td>synonymous_variant</td>\n",
       "      <td>T</td>\n",
       "      <td>1641</td>\n",
       "      <td>547</td>\n",
       "      <td>G</td>\n",
       "      <td>0</td>\n",
       "      <td>1107</td>\n",
       "      <td>SC</td>\n",
       "      <td>-0.000612</td>\n",
       "      <td>0.001205</td>\n",
       "      <td>-0.508176</td>\n",
       "      <td>6.113309e-01</td>\n",
       "      <td>-0.002973</td>\n",
       "      <td>0.001749</td>\n",
       "      <td>8.749904e-01</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2972</th>\n",
       "      <td>22872</td>\n",
       "      <td>EGB_00026</td>\n",
       "      <td>VIII</td>\n",
       "      <td>240738</td>\n",
       "      <td>ERG7</td>\n",
       "      <td>synonymous_variant</td>\n",
       "      <td>T</td>\n",
       "      <td>1641</td>\n",
       "      <td>547</td>\n",
       "      <td>G</td>\n",
       "      <td>0</td>\n",
       "      <td>1108</td>\n",
       "      <td>TBF</td>\n",
       "      <td>-0.000923</td>\n",
       "      <td>0.000930</td>\n",
       "      <td>-0.992934</td>\n",
       "      <td>3.207446e-01</td>\n",
       "      <td>-0.002745</td>\n",
       "      <td>0.000899</td>\n",
       "      <td>7.002247e-01</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2973</th>\n",
       "      <td>22882</td>\n",
       "      <td>EGB_00027</td>\n",
       "      <td>VIII</td>\n",
       "      <td>240738</td>\n",
       "      <td>ERG7</td>\n",
       "      <td>synonymous_variant</td>\n",
       "      <td>C</td>\n",
       "      <td>1641</td>\n",
       "      <td>547</td>\n",
       "      <td>G</td>\n",
       "      <td>0</td>\n",
       "      <td>1109</td>\n",
       "      <td>CAFF</td>\n",
       "      <td>0.000483</td>\n",
       "      <td>0.000940</td>\n",
       "      <td>0.514277</td>\n",
       "      <td>6.070592e-01</td>\n",
       "      <td>-0.001358</td>\n",
       "      <td>0.002325</td>\n",
       "      <td>8.721453e-01</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2974</th>\n",
       "      <td>22882</td>\n",
       "      <td>EGB_00027</td>\n",
       "      <td>VIII</td>\n",
       "      <td>240738</td>\n",
       "      <td>ERG7</td>\n",
       "      <td>synonymous_variant</td>\n",
       "      <td>C</td>\n",
       "      <td>1641</td>\n",
       "      <td>547</td>\n",
       "      <td>G</td>\n",
       "      <td>0</td>\n",
       "      <td>1110</td>\n",
       "      <td>COCL</td>\n",
       "      <td>0.077358</td>\n",
       "      <td>0.003184</td>\n",
       "      <td>24.293832</td>\n",
       "      <td>5.558787e-130</td>\n",
       "      <td>0.071117</td>\n",
       "      <td>0.083599</td>\n",
       "      <td>6.397594e-128</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2975</th>\n",
       "      <td>22882</td>\n",
       "      <td>EGB_00027</td>\n",
       "      <td>VIII</td>\n",
       "      <td>240738</td>\n",
       "      <td>ERG7</td>\n",
       "      <td>synonymous_variant</td>\n",
       "      <td>C</td>\n",
       "      <td>1641</td>\n",
       "      <td>547</td>\n",
       "      <td>G</td>\n",
       "      <td>0</td>\n",
       "      <td>1111</td>\n",
       "      <td>LOV</td>\n",
       "      <td>-0.002131</td>\n",
       "      <td>0.001269</td>\n",
       "      <td>-1.679057</td>\n",
       "      <td>9.314405e-02</td>\n",
       "      <td>-0.004619</td>\n",
       "      <td>0.000357</td>\n",
       "      <td>4.096807e-01</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2976</th>\n",
       "      <td>22882</td>\n",
       "      <td>EGB_00027</td>\n",
       "      <td>VIII</td>\n",
       "      <td>240738</td>\n",
       "      <td>ERG7</td>\n",
       "      <td>synonymous_variant</td>\n",
       "      <td>C</td>\n",
       "      <td>1641</td>\n",
       "      <td>547</td>\n",
       "      <td>G</td>\n",
       "      <td>0</td>\n",
       "      <td>1112</td>\n",
       "      <td>NACL</td>\n",
       "      <td>0.000592</td>\n",
       "      <td>0.000937</td>\n",
       "      <td>0.631675</td>\n",
       "      <td>5.276006e-01</td>\n",
       "      <td>-0.001245</td>\n",
       "      <td>0.002429</td>\n",
       "      <td>8.389248e-01</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2977</th>\n",
       "      <td>22882</td>\n",
       "      <td>EGB_00027</td>\n",
       "      <td>VIII</td>\n",
       "      <td>240738</td>\n",
       "      <td>ERG7</td>\n",
       "      <td>synonymous_variant</td>\n",
       "      <td>C</td>\n",
       "      <td>1641</td>\n",
       "      <td>547</td>\n",
       "      <td>G</td>\n",
       "      <td>0</td>\n",
       "      <td>1113</td>\n",
       "      <td>SC</td>\n",
       "      <td>0.002595</td>\n",
       "      <td>0.000925</td>\n",
       "      <td>2.803975</td>\n",
       "      <td>5.048678e-03</td>\n",
       "      <td>0.000781</td>\n",
       "      <td>0.004408</td>\n",
       "      <td>6.856578e-02</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2978</th>\n",
       "      <td>22882</td>\n",
       "      <td>EGB_00027</td>\n",
       "      <td>VIII</td>\n",
       "      <td>240738</td>\n",
       "      <td>ERG7</td>\n",
       "      <td>synonymous_variant</td>\n",
       "      <td>C</td>\n",
       "      <td>1641</td>\n",
       "      <td>547</td>\n",
       "      <td>G</td>\n",
       "      <td>0</td>\n",
       "      <td>1114</td>\n",
       "      <td>TBF</td>\n",
       "      <td>0.000800</td>\n",
       "      <td>0.000705</td>\n",
       "      <td>1.135169</td>\n",
       "      <td>2.563071e-01</td>\n",
       "      <td>-0.000582</td>\n",
       "      <td>0.002182</td>\n",
       "      <td>6.433892e-01</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "      Unnamed: 0_x     var_id chrom  SNP_chr_pos  Gene         Consequence  \\\n",
       "2967         22872  EGB_00026  VIII       240738  ERG7  synonymous_variant   \n",
       "2968         22872  EGB_00026  VIII       240738  ERG7  synonymous_variant   \n",
       "2969         22872  EGB_00026  VIII       240738  ERG7  synonymous_variant   \n",
       "2970         22872  EGB_00026  VIII       240738  ERG7  synonymous_variant   \n",
       "2971         22872  EGB_00026  VIII       240738  ERG7  synonymous_variant   \n",
       "2972         22872  EGB_00026  VIII       240738  ERG7  synonymous_variant   \n",
       "2973         22882  EGB_00027  VIII       240738  ERG7  synonymous_variant   \n",
       "2974         22882  EGB_00027  VIII       240738  ERG7  synonymous_variant   \n",
       "2975         22882  EGB_00027  VIII       240738  ERG7  synonymous_variant   \n",
       "2976         22882  EGB_00027  VIII       240738  ERG7  synonymous_variant   \n",
       "2977         22882  EGB_00027  VIII       240738  ERG7  synonymous_variant   \n",
       "2978         22882  EGB_00027  VIII       240738  ERG7  synonymous_variant   \n",
       "\n",
       "     ALT CDS_position Protein_position Amino_acids  DISTANCE  Unnamed: 0_y  \\\n",
       "2967   T         1641              547           G         0          1103   \n",
       "2968   T         1641              547           G         0          1104   \n",
       "2969   T         1641              547           G         0          1105   \n",
       "2970   T         1641              547           G         0          1106   \n",
       "2971   T         1641              547           G         0          1107   \n",
       "2972   T         1641              547           G         0          1108   \n",
       "2973   C         1641              547           G         0          1109   \n",
       "2974   C         1641              547           G         0          1110   \n",
       "2975   C         1641              547           G         0          1111   \n",
       "2976   C         1641              547           G         0          1112   \n",
       "2977   C         1641              547           G         0          1113   \n",
       "2978   C         1641              547           G         0          1114   \n",
       "\n",
       "     condition      coef   std err          t           pval  Conf. Int. Low  \\\n",
       "2967      CAFF  0.000308  0.001230   0.250274   8.023761e-01       -0.002102   \n",
       "2968      COCL -0.017969  0.011281  -1.592894   1.111872e-01       -0.040079   \n",
       "2969       LOV -0.001969  0.001845  -1.067331   2.858251e-01       -0.005585   \n",
       "2970      NACL  0.001724  0.001374   1.255040   2.094672e-01       -0.000968   \n",
       "2971        SC -0.000612  0.001205  -0.508176   6.113309e-01       -0.002973   \n",
       "2972       TBF -0.000923  0.000930  -0.992934   3.207446e-01       -0.002745   \n",
       "2973      CAFF  0.000483  0.000940   0.514277   6.070592e-01       -0.001358   \n",
       "2974      COCL  0.077358  0.003184  24.293832  5.558787e-130        0.071117   \n",
       "2975       LOV -0.002131  0.001269  -1.679057   9.314405e-02       -0.004619   \n",
       "2976      NACL  0.000592  0.000937   0.631675   5.276006e-01       -0.001245   \n",
       "2977        SC  0.002595  0.000925   2.803975   5.048678e-03        0.000781   \n",
       "2978       TBF  0.000800  0.000705   1.135169   2.563071e-01       -0.000582   \n",
       "\n",
       "      Conf. Int. Upp.           padj  \n",
       "2967         0.002718   9.525466e-01  \n",
       "2968         0.004141   4.481939e-01  \n",
       "2969         0.001647   6.706642e-01  \n",
       "2970         0.004416   5.893554e-01  \n",
       "2971         0.001749   8.749904e-01  \n",
       "2972         0.000899   7.002247e-01  \n",
       "2973         0.002325   8.721453e-01  \n",
       "2974         0.083599  6.397594e-128  \n",
       "2975         0.000357   4.096807e-01  \n",
       "2976         0.002429   8.389248e-01  \n",
       "2977         0.004408   6.856578e-02  \n",
       "2978         0.002182   6.433892e-01  "
      ]
     },
     "execution_count": 18,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "#ERG7 site with two synonymous variants, one with a significant effect in cobalt chloride and one without\n",
    "\n",
    "chrom = 'VIII'\n",
    "pos = 240738\n",
    "\n",
    "annotated_fits[(annotated_fits['chrom']==chrom)&(annotated_fits['SNP_chr_pos']==pos)]"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### Upstream Hits within sites w/ multiple edits"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 19,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Unnamed: 0_x</th>\n",
       "      <th>var_id</th>\n",
       "      <th>chrom</th>\n",
       "      <th>SNP_chr_pos</th>\n",
       "      <th>Gene</th>\n",
       "      <th>Consequence</th>\n",
       "      <th>ALT</th>\n",
       "      <th>CDS_position</th>\n",
       "      <th>Protein_position</th>\n",
       "      <th>Amino_acids</th>\n",
       "      <th>DISTANCE</th>\n",
       "      <th>Unnamed: 0_y</th>\n",
       "      <th>condition</th>\n",
       "      <th>coef</th>\n",
       "      <th>std err</th>\n",
       "      <th>t</th>\n",
       "      <th>pval</th>\n",
       "      <th>Conf. Int. Low</th>\n",
       "      <th>Conf. Int. Upp.</th>\n",
       "      <th>padj</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>359</th>\n",
       "      <td>14037</td>\n",
       "      <td>EG8_00029</td>\n",
       "      <td>VII</td>\n",
       "      <td>472609</td>\n",
       "      <td>ERG4</td>\n",
       "      <td>upstream_gene_variant</td>\n",
       "      <td>C</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>246</td>\n",
       "      <td>67</td>\n",
       "      <td>COCL</td>\n",
       "      <td>-0.012122</td>\n",
       "      <td>0.003450</td>\n",
       "      <td>-3.513432</td>\n",
       "      <td>4.425583e-04</td>\n",
       "      <td>-0.018884</td>\n",
       "      <td>-0.005359</td>\n",
       "      <td>8.968049e-03</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>362</th>\n",
       "      <td>14037</td>\n",
       "      <td>EG8_00029</td>\n",
       "      <td>VII</td>\n",
       "      <td>472609</td>\n",
       "      <td>ERG4</td>\n",
       "      <td>upstream_gene_variant</td>\n",
       "      <td>C</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>246</td>\n",
       "      <td>70</td>\n",
       "      <td>SC</td>\n",
       "      <td>-0.003537</td>\n",
       "      <td>0.001004</td>\n",
       "      <td>-3.523292</td>\n",
       "      <td>4.264156e-04</td>\n",
       "      <td>-0.005505</td>\n",
       "      <td>-0.001570</td>\n",
       "      <td>8.680765e-03</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>372</th>\n",
       "      <td>14049</td>\n",
       "      <td>EG8_00031</td>\n",
       "      <td>VII</td>\n",
       "      <td>472612</td>\n",
       "      <td>ERG4</td>\n",
       "      <td>upstream_gene_variant</td>\n",
       "      <td>T</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>243</td>\n",
       "      <td>80</td>\n",
       "      <td>LOV</td>\n",
       "      <td>0.007146</td>\n",
       "      <td>0.001517</td>\n",
       "      <td>4.712358</td>\n",
       "      <td>2.452032e-06</td>\n",
       "      <td>0.004174</td>\n",
       "      <td>0.010119</td>\n",
       "      <td>7.123589e-05</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>947</th>\n",
       "      <td>14747</td>\n",
       "      <td>ERG_00285</td>\n",
       "      <td>VII</td>\n",
       "      <td>610071</td>\n",
       "      <td>ERG25</td>\n",
       "      <td>upstream_gene_variant</td>\n",
       "      <td>A</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>493</td>\n",
       "      <td>3668</td>\n",
       "      <td>COCL</td>\n",
       "      <td>-0.020771</td>\n",
       "      <td>0.004743</td>\n",
       "      <td>-4.379045</td>\n",
       "      <td>1.193245e-05</td>\n",
       "      <td>-0.030068</td>\n",
       "      <td>-0.011474</td>\n",
       "      <td>3.245989e-04</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1327</th>\n",
       "      <td>16747</td>\n",
       "      <td>ERG_00444</td>\n",
       "      <td>VII</td>\n",
       "      <td>848783</td>\n",
       "      <td>ERG1</td>\n",
       "      <td>upstream_gene_variant</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>361</td>\n",
       "      <td>3981</td>\n",
       "      <td>COCL</td>\n",
       "      <td>0.062549</td>\n",
       "      <td>0.002172</td>\n",
       "      <td>28.801341</td>\n",
       "      <td>1.196889e-181</td>\n",
       "      <td>0.058293</td>\n",
       "      <td>0.066806</td>\n",
       "      <td>1.705472e-179</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1328</th>\n",
       "      <td>16747</td>\n",
       "      <td>ERG_00444</td>\n",
       "      <td>VII</td>\n",
       "      <td>848783</td>\n",
       "      <td>ERG1</td>\n",
       "      <td>upstream_gene_variant</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>361</td>\n",
       "      <td>3982</td>\n",
       "      <td>LOV</td>\n",
       "      <td>0.003645</td>\n",
       "      <td>0.000894</td>\n",
       "      <td>4.075507</td>\n",
       "      <td>4.595049e-05</td>\n",
       "      <td>0.001892</td>\n",
       "      <td>0.005398</td>\n",
       "      <td>1.139496e-03</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1331</th>\n",
       "      <td>16747</td>\n",
       "      <td>ERG_00444</td>\n",
       "      <td>VII</td>\n",
       "      <td>848783</td>\n",
       "      <td>ERG1</td>\n",
       "      <td>upstream_gene_variant</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>361</td>\n",
       "      <td>3985</td>\n",
       "      <td>TBF</td>\n",
       "      <td>-0.100394</td>\n",
       "      <td>0.001044</td>\n",
       "      <td>-96.128673</td>\n",
       "      <td>0.000000e+00</td>\n",
       "      <td>-0.102441</td>\n",
       "      <td>-0.098347</td>\n",
       "      <td>0.000000e+00</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1333</th>\n",
       "      <td>16756</td>\n",
       "      <td>ERG_00445</td>\n",
       "      <td>VII</td>\n",
       "      <td>848783</td>\n",
       "      <td>ERG1</td>\n",
       "      <td>upstream_gene_variant</td>\n",
       "      <td>GC</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>360</td>\n",
       "      <td>3987</td>\n",
       "      <td>COCL</td>\n",
       "      <td>0.044502</td>\n",
       "      <td>0.003226</td>\n",
       "      <td>13.795166</td>\n",
       "      <td>2.992865e-43</td>\n",
       "      <td>0.038179</td>\n",
       "      <td>0.050824</td>\n",
       "      <td>2.035375e-41</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1337</th>\n",
       "      <td>16756</td>\n",
       "      <td>ERG_00445</td>\n",
       "      <td>VII</td>\n",
       "      <td>848783</td>\n",
       "      <td>ERG1</td>\n",
       "      <td>upstream_gene_variant</td>\n",
       "      <td>GC</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>360</td>\n",
       "      <td>3991</td>\n",
       "      <td>TBF</td>\n",
       "      <td>-0.078456</td>\n",
       "      <td>0.001455</td>\n",
       "      <td>-53.915900</td>\n",
       "      <td>0.000000e+00</td>\n",
       "      <td>-0.081309</td>\n",
       "      <td>-0.075604</td>\n",
       "      <td>0.000000e+00</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1801</th>\n",
       "      <td>21069</td>\n",
       "      <td>ERG_00576</td>\n",
       "      <td>VIII</td>\n",
       "      <td>190196</td>\n",
       "      <td>NCP1</td>\n",
       "      <td>upstream_gene_variant</td>\n",
       "      <td>TGTG</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>346</td>\n",
       "      <td>4165</td>\n",
       "      <td>TBF</td>\n",
       "      <td>-0.003378</td>\n",
       "      <td>0.000494</td>\n",
       "      <td>-6.837636</td>\n",
       "      <td>8.098222e-12</td>\n",
       "      <td>-0.004346</td>\n",
       "      <td>-0.002410</td>\n",
       "      <td>3.231011e-10</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5273</th>\n",
       "      <td>32532</td>\n",
       "      <td>ERG_01689</td>\n",
       "      <td>XIII</td>\n",
       "      <td>20611</td>\n",
       "      <td>ERG13</td>\n",
       "      <td>upstream_gene_variant</td>\n",
       "      <td>A</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>76</td>\n",
       "      <td>6437</td>\n",
       "      <td>LOV</td>\n",
       "      <td>0.010003</td>\n",
       "      <td>0.001267</td>\n",
       "      <td>7.895830</td>\n",
       "      <td>2.913668e-15</td>\n",
       "      <td>0.007520</td>\n",
       "      <td>0.012486</td>\n",
       "      <td>1.245524e-13</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5279</th>\n",
       "      <td>32538</td>\n",
       "      <td>ERG_01690</td>\n",
       "      <td>XIII</td>\n",
       "      <td>20611</td>\n",
       "      <td>ERG13</td>\n",
       "      <td>upstream_gene_variant</td>\n",
       "      <td>C</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>76</td>\n",
       "      <td>6443</td>\n",
       "      <td>LOV</td>\n",
       "      <td>0.007932</td>\n",
       "      <td>0.001272</td>\n",
       "      <td>6.235127</td>\n",
       "      <td>4.532523e-10</td>\n",
       "      <td>0.005439</td>\n",
       "      <td>0.010425</td>\n",
       "      <td>1.674422e-08</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5752</th>\n",
       "      <td>32929</td>\n",
       "      <td>ERG_01887</td>\n",
       "      <td>XIII</td>\n",
       "      <td>118967</td>\n",
       "      <td>HMG1</td>\n",
       "      <td>upstream_gene_variant</td>\n",
       "      <td>A</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>69</td>\n",
       "      <td>6857</td>\n",
       "      <td>LOV</td>\n",
       "      <td>0.020362</td>\n",
       "      <td>0.003185</td>\n",
       "      <td>6.393992</td>\n",
       "      <td>1.623372e-10</td>\n",
       "      <td>0.014120</td>\n",
       "      <td>0.026604</td>\n",
       "      <td>6.097495e-09</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>8071</th>\n",
       "      <td>51186</td>\n",
       "      <td>ERG_02672</td>\n",
       "      <td>XVI</td>\n",
       "      <td>497884</td>\n",
       "      <td>ERG10</td>\n",
       "      <td>upstream_gene_variant</td>\n",
       "      <td>A</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>212</td>\n",
       "      <td>8207</td>\n",
       "      <td>CAFF</td>\n",
       "      <td>0.004485</td>\n",
       "      <td>0.000500</td>\n",
       "      <td>8.968342</td>\n",
       "      <td>3.061659e-19</td>\n",
       "      <td>0.003505</td>\n",
       "      <td>0.005465</td>\n",
       "      <td>1.395153e-17</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>8073</th>\n",
       "      <td>51186</td>\n",
       "      <td>ERG_02672</td>\n",
       "      <td>XVI</td>\n",
       "      <td>497884</td>\n",
       "      <td>ERG10</td>\n",
       "      <td>upstream_gene_variant</td>\n",
       "      <td>A</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>212</td>\n",
       "      <td>8209</td>\n",
       "      <td>LOV</td>\n",
       "      <td>-0.003719</td>\n",
       "      <td>0.000683</td>\n",
       "      <td>-5.442711</td>\n",
       "      <td>5.260164e-08</td>\n",
       "      <td>-0.005058</td>\n",
       "      <td>-0.002380</td>\n",
       "      <td>1.742454e-06</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>8074</th>\n",
       "      <td>51186</td>\n",
       "      <td>ERG_02672</td>\n",
       "      <td>XVI</td>\n",
       "      <td>497884</td>\n",
       "      <td>ERG10</td>\n",
       "      <td>upstream_gene_variant</td>\n",
       "      <td>A</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>212</td>\n",
       "      <td>8210</td>\n",
       "      <td>NACL</td>\n",
       "      <td>-0.002156</td>\n",
       "      <td>0.000471</td>\n",
       "      <td>-4.577842</td>\n",
       "      <td>4.703769e-06</td>\n",
       "      <td>-0.003080</td>\n",
       "      <td>-0.001233</td>\n",
       "      <td>1.323691e-04</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "      Unnamed: 0_x     var_id chrom  SNP_chr_pos   Gene  \\\n",
       "359          14037  EG8_00029   VII       472609   ERG4   \n",
       "362          14037  EG8_00029   VII       472609   ERG4   \n",
       "372          14049  EG8_00031   VII       472612   ERG4   \n",
       "947          14747  ERG_00285   VII       610071  ERG25   \n",
       "1327         16747  ERG_00444   VII       848783   ERG1   \n",
       "1328         16747  ERG_00444   VII       848783   ERG1   \n",
       "1331         16747  ERG_00444   VII       848783   ERG1   \n",
       "1333         16756  ERG_00445   VII       848783   ERG1   \n",
       "1337         16756  ERG_00445   VII       848783   ERG1   \n",
       "1801         21069  ERG_00576  VIII       190196   NCP1   \n",
       "5273         32532  ERG_01689  XIII        20611  ERG13   \n",
       "5279         32538  ERG_01690  XIII        20611  ERG13   \n",
       "5752         32929  ERG_01887  XIII       118967   HMG1   \n",
       "8071         51186  ERG_02672   XVI       497884  ERG10   \n",
       "8073         51186  ERG_02672   XVI       497884  ERG10   \n",
       "8074         51186  ERG_02672   XVI       497884  ERG10   \n",
       "\n",
       "                Consequence   ALT CDS_position Protein_position Amino_acids  \\\n",
       "359   upstream_gene_variant     C            -                -           -   \n",
       "362   upstream_gene_variant     C            -                -           -   \n",
       "372   upstream_gene_variant     T            -                -           -   \n",
       "947   upstream_gene_variant     A            -                -           -   \n",
       "1327  upstream_gene_variant     -            -                -           -   \n",
       "1328  upstream_gene_variant     -            -                -           -   \n",
       "1331  upstream_gene_variant     -            -                -           -   \n",
       "1333  upstream_gene_variant    GC            -                -           -   \n",
       "1337  upstream_gene_variant    GC            -                -           -   \n",
       "1801  upstream_gene_variant  TGTG            -                -           -   \n",
       "5273  upstream_gene_variant     A            -                -           -   \n",
       "5279  upstream_gene_variant     C            -                -           -   \n",
       "5752  upstream_gene_variant     A            -                -           -   \n",
       "8071  upstream_gene_variant     A            -                -           -   \n",
       "8073  upstream_gene_variant     A            -                -           -   \n",
       "8074  upstream_gene_variant     A            -                -           -   \n",
       "\n",
       "      DISTANCE  Unnamed: 0_y condition      coef   std err          t  \\\n",
       "359        246            67      COCL -0.012122  0.003450  -3.513432   \n",
       "362        246            70        SC -0.003537  0.001004  -3.523292   \n",
       "372        243            80       LOV  0.007146  0.001517   4.712358   \n",
       "947        493          3668      COCL -0.020771  0.004743  -4.379045   \n",
       "1327       361          3981      COCL  0.062549  0.002172  28.801341   \n",
       "1328       361          3982       LOV  0.003645  0.000894   4.075507   \n",
       "1331       361          3985       TBF -0.100394  0.001044 -96.128673   \n",
       "1333       360          3987      COCL  0.044502  0.003226  13.795166   \n",
       "1337       360          3991       TBF -0.078456  0.001455 -53.915900   \n",
       "1801       346          4165       TBF -0.003378  0.000494  -6.837636   \n",
       "5273        76          6437       LOV  0.010003  0.001267   7.895830   \n",
       "5279        76          6443       LOV  0.007932  0.001272   6.235127   \n",
       "5752        69          6857       LOV  0.020362  0.003185   6.393992   \n",
       "8071       212          8207      CAFF  0.004485  0.000500   8.968342   \n",
       "8073       212          8209       LOV -0.003719  0.000683  -5.442711   \n",
       "8074       212          8210      NACL -0.002156  0.000471  -4.577842   \n",
       "\n",
       "               pval  Conf. Int. Low  Conf. Int. Upp.           padj  \n",
       "359    4.425583e-04       -0.018884        -0.005359   8.968049e-03  \n",
       "362    4.264156e-04       -0.005505        -0.001570   8.680765e-03  \n",
       "372    2.452032e-06        0.004174         0.010119   7.123589e-05  \n",
       "947    1.193245e-05       -0.030068        -0.011474   3.245989e-04  \n",
       "1327  1.196889e-181        0.058293         0.066806  1.705472e-179  \n",
       "1328   4.595049e-05        0.001892         0.005398   1.139496e-03  \n",
       "1331   0.000000e+00       -0.102441        -0.098347   0.000000e+00  \n",
       "1333   2.992865e-43        0.038179         0.050824   2.035375e-41  \n",
       "1337   0.000000e+00       -0.081309        -0.075604   0.000000e+00  \n",
       "1801   8.098222e-12       -0.004346        -0.002410   3.231011e-10  \n",
       "5273   2.913668e-15        0.007520         0.012486   1.245524e-13  \n",
       "5279   4.532523e-10        0.005439         0.010425   1.674422e-08  \n",
       "5752   1.623372e-10        0.014120         0.026604   6.097495e-09  \n",
       "8071   3.061659e-19        0.003505         0.005465   1.395153e-17  \n",
       "8073   5.260164e-08       -0.005058        -0.002380   1.742454e-06  \n",
       "8074   4.703769e-06       -0.003080        -0.001233   1.323691e-04  "
      ]
     },
     "execution_count": 19,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "#displaying all synonymous variants which are hits and target a site with multiple edits \n",
    "annotated_fits[(annotated_fits['var_id'].isin(multi_edits))&(annotated_fits['padj']<.01)&(annotated_fits['Consequence']=='upstream_gene_variant')]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 20,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Unnamed: 0_x</th>\n",
       "      <th>var_id</th>\n",
       "      <th>chrom</th>\n",
       "      <th>SNP_chr_pos</th>\n",
       "      <th>Gene</th>\n",
       "      <th>Consequence</th>\n",
       "      <th>ALT</th>\n",
       "      <th>CDS_position</th>\n",
       "      <th>Protein_position</th>\n",
       "      <th>Amino_acids</th>\n",
       "      <th>DISTANCE</th>\n",
       "      <th>Unnamed: 0_y</th>\n",
       "      <th>condition</th>\n",
       "      <th>coef</th>\n",
       "      <th>std err</th>\n",
       "      <th>t</th>\n",
       "      <th>pval</th>\n",
       "      <th>Conf. Int. Low</th>\n",
       "      <th>Conf. Int. Upp.</th>\n",
       "      <th>padj</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>352</th>\n",
       "      <td>14031</td>\n",
       "      <td>EG8_00028</td>\n",
       "      <td>VII</td>\n",
       "      <td>472609</td>\n",
       "      <td>ERG4</td>\n",
       "      <td>upstream_gene_variant</td>\n",
       "      <td>G</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>246</td>\n",
       "      <td>60</td>\n",
       "      <td>CAFF</td>\n",
       "      <td>0.000746</td>\n",
       "      <td>0.001032</td>\n",
       "      <td>0.722323</td>\n",
       "      <td>0.470098</td>\n",
       "      <td>-0.001277</td>\n",
       "      <td>0.002769</td>\n",
       "      <td>0.808442</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>353</th>\n",
       "      <td>14031</td>\n",
       "      <td>EG8_00028</td>\n",
       "      <td>VII</td>\n",
       "      <td>472609</td>\n",
       "      <td>ERG4</td>\n",
       "      <td>upstream_gene_variant</td>\n",
       "      <td>G</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>246</td>\n",
       "      <td>61</td>\n",
       "      <td>COCL</td>\n",
       "      <td>-0.005886</td>\n",
       "      <td>0.003174</td>\n",
       "      <td>-1.854097</td>\n",
       "      <td>0.063728</td>\n",
       "      <td>-0.012107</td>\n",
       "      <td>0.000336</td>\n",
       "      <td>0.340935</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>354</th>\n",
       "      <td>14031</td>\n",
       "      <td>EG8_00028</td>\n",
       "      <td>VII</td>\n",
       "      <td>472609</td>\n",
       "      <td>ERG4</td>\n",
       "      <td>upstream_gene_variant</td>\n",
       "      <td>G</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>246</td>\n",
       "      <td>62</td>\n",
       "      <td>LOV</td>\n",
       "      <td>-0.001118</td>\n",
       "      <td>0.001256</td>\n",
       "      <td>-0.890344</td>\n",
       "      <td>0.373283</td>\n",
       "      <td>-0.003579</td>\n",
       "      <td>0.001343</td>\n",
       "      <td>0.745248</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>355</th>\n",
       "      <td>14031</td>\n",
       "      <td>EG8_00028</td>\n",
       "      <td>VII</td>\n",
       "      <td>472609</td>\n",
       "      <td>ERG4</td>\n",
       "      <td>upstream_gene_variant</td>\n",
       "      <td>G</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>246</td>\n",
       "      <td>63</td>\n",
       "      <td>NACL</td>\n",
       "      <td>-0.000098</td>\n",
       "      <td>0.000987</td>\n",
       "      <td>-0.099631</td>\n",
       "      <td>0.920637</td>\n",
       "      <td>-0.002032</td>\n",
       "      <td>0.001836</td>\n",
       "      <td>0.986353</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>356</th>\n",
       "      <td>14031</td>\n",
       "      <td>EG8_00028</td>\n",
       "      <td>VII</td>\n",
       "      <td>472609</td>\n",
       "      <td>ERG4</td>\n",
       "      <td>upstream_gene_variant</td>\n",
       "      <td>G</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>246</td>\n",
       "      <td>64</td>\n",
       "      <td>SC</td>\n",
       "      <td>-0.001689</td>\n",
       "      <td>0.000958</td>\n",
       "      <td>-1.762787</td>\n",
       "      <td>0.077940</td>\n",
       "      <td>-0.003566</td>\n",
       "      <td>0.000189</td>\n",
       "      <td>0.374727</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>357</th>\n",
       "      <td>14031</td>\n",
       "      <td>EG8_00028</td>\n",
       "      <td>VII</td>\n",
       "      <td>472609</td>\n",
       "      <td>ERG4</td>\n",
       "      <td>upstream_gene_variant</td>\n",
       "      <td>G</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>246</td>\n",
       "      <td>65</td>\n",
       "      <td>TBF</td>\n",
       "      <td>-0.000108</td>\n",
       "      <td>0.000726</td>\n",
       "      <td>-0.148306</td>\n",
       "      <td>0.882102</td>\n",
       "      <td>-0.001531</td>\n",
       "      <td>0.001316</td>\n",
       "      <td>0.976042</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>358</th>\n",
       "      <td>14037</td>\n",
       "      <td>EG8_00029</td>\n",
       "      <td>VII</td>\n",
       "      <td>472609</td>\n",
       "      <td>ERG4</td>\n",
       "      <td>upstream_gene_variant</td>\n",
       "      <td>C</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>246</td>\n",
       "      <td>66</td>\n",
       "      <td>CAFF</td>\n",
       "      <td>0.000165</td>\n",
       "      <td>0.001095</td>\n",
       "      <td>0.150785</td>\n",
       "      <td>0.880146</td>\n",
       "      <td>-0.001981</td>\n",
       "      <td>0.002311</td>\n",
       "      <td>0.975423</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>359</th>\n",
       "      <td>14037</td>\n",
       "      <td>EG8_00029</td>\n",
       "      <td>VII</td>\n",
       "      <td>472609</td>\n",
       "      <td>ERG4</td>\n",
       "      <td>upstream_gene_variant</td>\n",
       "      <td>C</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>246</td>\n",
       "      <td>67</td>\n",
       "      <td>COCL</td>\n",
       "      <td>-0.012122</td>\n",
       "      <td>0.003450</td>\n",
       "      <td>-3.513432</td>\n",
       "      <td>0.000443</td>\n",
       "      <td>-0.018884</td>\n",
       "      <td>-0.005359</td>\n",
       "      <td>0.008968</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>360</th>\n",
       "      <td>14037</td>\n",
       "      <td>EG8_00029</td>\n",
       "      <td>VII</td>\n",
       "      <td>472609</td>\n",
       "      <td>ERG4</td>\n",
       "      <td>upstream_gene_variant</td>\n",
       "      <td>C</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>246</td>\n",
       "      <td>68</td>\n",
       "      <td>LOV</td>\n",
       "      <td>0.000648</td>\n",
       "      <td>0.001347</td>\n",
       "      <td>0.480978</td>\n",
       "      <td>0.630533</td>\n",
       "      <td>-0.001992</td>\n",
       "      <td>0.003288</td>\n",
       "      <td>0.884254</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>361</th>\n",
       "      <td>14037</td>\n",
       "      <td>EG8_00029</td>\n",
       "      <td>VII</td>\n",
       "      <td>472609</td>\n",
       "      <td>ERG4</td>\n",
       "      <td>upstream_gene_variant</td>\n",
       "      <td>C</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>246</td>\n",
       "      <td>69</td>\n",
       "      <td>NACL</td>\n",
       "      <td>0.001119</td>\n",
       "      <td>0.001069</td>\n",
       "      <td>1.046762</td>\n",
       "      <td>0.295212</td>\n",
       "      <td>-0.000976</td>\n",
       "      <td>0.003213</td>\n",
       "      <td>0.677260</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>362</th>\n",
       "      <td>14037</td>\n",
       "      <td>EG8_00029</td>\n",
       "      <td>VII</td>\n",
       "      <td>472609</td>\n",
       "      <td>ERG4</td>\n",
       "      <td>upstream_gene_variant</td>\n",
       "      <td>C</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>246</td>\n",
       "      <td>70</td>\n",
       "      <td>SC</td>\n",
       "      <td>-0.003537</td>\n",
       "      <td>0.001004</td>\n",
       "      <td>-3.523292</td>\n",
       "      <td>0.000426</td>\n",
       "      <td>-0.005505</td>\n",
       "      <td>-0.001570</td>\n",
       "      <td>0.008681</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>363</th>\n",
       "      <td>14037</td>\n",
       "      <td>EG8_00029</td>\n",
       "      <td>VII</td>\n",
       "      <td>472609</td>\n",
       "      <td>ERG4</td>\n",
       "      <td>upstream_gene_variant</td>\n",
       "      <td>C</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>246</td>\n",
       "      <td>71</td>\n",
       "      <td>TBF</td>\n",
       "      <td>-0.000262</td>\n",
       "      <td>0.000767</td>\n",
       "      <td>-0.341092</td>\n",
       "      <td>0.733035</td>\n",
       "      <td>-0.001766</td>\n",
       "      <td>0.001242</td>\n",
       "      <td>0.923962</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "     Unnamed: 0_x     var_id chrom  SNP_chr_pos  Gene            Consequence  \\\n",
       "352         14031  EG8_00028   VII       472609  ERG4  upstream_gene_variant   \n",
       "353         14031  EG8_00028   VII       472609  ERG4  upstream_gene_variant   \n",
       "354         14031  EG8_00028   VII       472609  ERG4  upstream_gene_variant   \n",
       "355         14031  EG8_00028   VII       472609  ERG4  upstream_gene_variant   \n",
       "356         14031  EG8_00028   VII       472609  ERG4  upstream_gene_variant   \n",
       "357         14031  EG8_00028   VII       472609  ERG4  upstream_gene_variant   \n",
       "358         14037  EG8_00029   VII       472609  ERG4  upstream_gene_variant   \n",
       "359         14037  EG8_00029   VII       472609  ERG4  upstream_gene_variant   \n",
       "360         14037  EG8_00029   VII       472609  ERG4  upstream_gene_variant   \n",
       "361         14037  EG8_00029   VII       472609  ERG4  upstream_gene_variant   \n",
       "362         14037  EG8_00029   VII       472609  ERG4  upstream_gene_variant   \n",
       "363         14037  EG8_00029   VII       472609  ERG4  upstream_gene_variant   \n",
       "\n",
       "    ALT CDS_position Protein_position Amino_acids  DISTANCE  Unnamed: 0_y  \\\n",
       "352   G            -                -           -       246            60   \n",
       "353   G            -                -           -       246            61   \n",
       "354   G            -                -           -       246            62   \n",
       "355   G            -                -           -       246            63   \n",
       "356   G            -                -           -       246            64   \n",
       "357   G            -                -           -       246            65   \n",
       "358   C            -                -           -       246            66   \n",
       "359   C            -                -           -       246            67   \n",
       "360   C            -                -           -       246            68   \n",
       "361   C            -                -           -       246            69   \n",
       "362   C            -                -           -       246            70   \n",
       "363   C            -                -           -       246            71   \n",
       "\n",
       "    condition      coef   std err         t      pval  Conf. Int. Low  \\\n",
       "352      CAFF  0.000746  0.001032  0.722323  0.470098       -0.001277   \n",
       "353      COCL -0.005886  0.003174 -1.854097  0.063728       -0.012107   \n",
       "354       LOV -0.001118  0.001256 -0.890344  0.373283       -0.003579   \n",
       "355      NACL -0.000098  0.000987 -0.099631  0.920637       -0.002032   \n",
       "356        SC -0.001689  0.000958 -1.762787  0.077940       -0.003566   \n",
       "357       TBF -0.000108  0.000726 -0.148306  0.882102       -0.001531   \n",
       "358      CAFF  0.000165  0.001095  0.150785  0.880146       -0.001981   \n",
       "359      COCL -0.012122  0.003450 -3.513432  0.000443       -0.018884   \n",
       "360       LOV  0.000648  0.001347  0.480978  0.630533       -0.001992   \n",
       "361      NACL  0.001119  0.001069  1.046762  0.295212       -0.000976   \n",
       "362        SC -0.003537  0.001004 -3.523292  0.000426       -0.005505   \n",
       "363       TBF -0.000262  0.000767 -0.341092  0.733035       -0.001766   \n",
       "\n",
       "     Conf. Int. Upp.      padj  \n",
       "352         0.002769  0.808442  \n",
       "353         0.000336  0.340935  \n",
       "354         0.001343  0.745248  \n",
       "355         0.001836  0.986353  \n",
       "356         0.000189  0.374727  \n",
       "357         0.001316  0.976042  \n",
       "358         0.002311  0.975423  \n",
       "359        -0.005359  0.008968  \n",
       "360         0.003288  0.884254  \n",
       "361         0.003213  0.677260  \n",
       "362        -0.001570  0.008681  \n",
       "363         0.001242  0.923962  "
      ]
     },
     "execution_count": 20,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "#ERG4 site with two upstream variants, one with a significant effect in cobalt chloride and SC and one without\n",
    "\n",
    "chrom = 'VII'\n",
    "pos = 472609\n",
    "\n",
    "annotated_fits[(annotated_fits['chrom']==chrom)&(annotated_fits['SNP_chr_pos']==pos)]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 21,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Unnamed: 0_x</th>\n",
       "      <th>var_id</th>\n",
       "      <th>chrom</th>\n",
       "      <th>SNP_chr_pos</th>\n",
       "      <th>Gene</th>\n",
       "      <th>Consequence</th>\n",
       "      <th>ALT</th>\n",
       "      <th>CDS_position</th>\n",
       "      <th>Protein_position</th>\n",
       "      <th>Amino_acids</th>\n",
       "      <th>DISTANCE</th>\n",
       "      <th>Unnamed: 0_y</th>\n",
       "      <th>condition</th>\n",
       "      <th>coef</th>\n",
       "      <th>std err</th>\n",
       "      <th>t</th>\n",
       "      <th>pval</th>\n",
       "      <th>Conf. Int. Low</th>\n",
       "      <th>Conf. Int. Upp.</th>\n",
       "      <th>padj</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>364</th>\n",
       "      <td>14043</td>\n",
       "      <td>EG8_00030</td>\n",
       "      <td>VII</td>\n",
       "      <td>472612</td>\n",
       "      <td>ERG4</td>\n",
       "      <td>upstream_gene_variant</td>\n",
       "      <td>A</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>243</td>\n",
       "      <td>72</td>\n",
       "      <td>CAFF</td>\n",
       "      <td>-0.001099</td>\n",
       "      <td>0.001076</td>\n",
       "      <td>-1.021815</td>\n",
       "      <td>0.306871</td>\n",
       "      <td>-0.003207</td>\n",
       "      <td>0.001009</td>\n",
       "      <td>0.686635</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>365</th>\n",
       "      <td>14043</td>\n",
       "      <td>EG8_00030</td>\n",
       "      <td>VII</td>\n",
       "      <td>472612</td>\n",
       "      <td>ERG4</td>\n",
       "      <td>upstream_gene_variant</td>\n",
       "      <td>A</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>243</td>\n",
       "      <td>73</td>\n",
       "      <td>COCL</td>\n",
       "      <td>-0.003337</td>\n",
       "      <td>0.003359</td>\n",
       "      <td>-0.993492</td>\n",
       "      <td>0.320473</td>\n",
       "      <td>-0.009920</td>\n",
       "      <td>0.003246</td>\n",
       "      <td>0.699971</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>366</th>\n",
       "      <td>14043</td>\n",
       "      <td>EG8_00030</td>\n",
       "      <td>VII</td>\n",
       "      <td>472612</td>\n",
       "      <td>ERG4</td>\n",
       "      <td>upstream_gene_variant</td>\n",
       "      <td>A</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>243</td>\n",
       "      <td>74</td>\n",
       "      <td>LOV</td>\n",
       "      <td>0.002025</td>\n",
       "      <td>0.001411</td>\n",
       "      <td>1.435011</td>\n",
       "      <td>0.151287</td>\n",
       "      <td>-0.000741</td>\n",
       "      <td>0.004791</td>\n",
       "      <td>0.515410</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>367</th>\n",
       "      <td>14043</td>\n",
       "      <td>EG8_00030</td>\n",
       "      <td>VII</td>\n",
       "      <td>472612</td>\n",
       "      <td>ERG4</td>\n",
       "      <td>upstream_gene_variant</td>\n",
       "      <td>A</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>243</td>\n",
       "      <td>75</td>\n",
       "      <td>NACL</td>\n",
       "      <td>-0.000383</td>\n",
       "      <td>0.001182</td>\n",
       "      <td>-0.323774</td>\n",
       "      <td>0.746110</td>\n",
       "      <td>-0.002699</td>\n",
       "      <td>0.001933</td>\n",
       "      <td>0.928813</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>368</th>\n",
       "      <td>14043</td>\n",
       "      <td>EG8_00030</td>\n",
       "      <td>VII</td>\n",
       "      <td>472612</td>\n",
       "      <td>ERG4</td>\n",
       "      <td>upstream_gene_variant</td>\n",
       "      <td>A</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>243</td>\n",
       "      <td>76</td>\n",
       "      <td>SC</td>\n",
       "      <td>-0.002316</td>\n",
       "      <td>0.001060</td>\n",
       "      <td>-2.183660</td>\n",
       "      <td>0.028990</td>\n",
       "      <td>-0.004394</td>\n",
       "      <td>-0.000237</td>\n",
       "      <td>0.225707</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>369</th>\n",
       "      <td>14043</td>\n",
       "      <td>EG8_00030</td>\n",
       "      <td>VII</td>\n",
       "      <td>472612</td>\n",
       "      <td>ERG4</td>\n",
       "      <td>upstream_gene_variant</td>\n",
       "      <td>A</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>243</td>\n",
       "      <td>77</td>\n",
       "      <td>TBF</td>\n",
       "      <td>0.000555</td>\n",
       "      <td>0.000846</td>\n",
       "      <td>0.656327</td>\n",
       "      <td>0.511616</td>\n",
       "      <td>-0.001103</td>\n",
       "      <td>0.002214</td>\n",
       "      <td>0.830142</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>370</th>\n",
       "      <td>14049</td>\n",
       "      <td>EG8_00031</td>\n",
       "      <td>VII</td>\n",
       "      <td>472612</td>\n",
       "      <td>ERG4</td>\n",
       "      <td>upstream_gene_variant</td>\n",
       "      <td>T</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>243</td>\n",
       "      <td>78</td>\n",
       "      <td>CAFF</td>\n",
       "      <td>-0.001028</td>\n",
       "      <td>0.001177</td>\n",
       "      <td>-0.873487</td>\n",
       "      <td>0.382400</td>\n",
       "      <td>-0.003335</td>\n",
       "      <td>0.001279</td>\n",
       "      <td>0.751325</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>371</th>\n",
       "      <td>14049</td>\n",
       "      <td>EG8_00031</td>\n",
       "      <td>VII</td>\n",
       "      <td>472612</td>\n",
       "      <td>ERG4</td>\n",
       "      <td>upstream_gene_variant</td>\n",
       "      <td>T</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>243</td>\n",
       "      <td>79</td>\n",
       "      <td>COCL</td>\n",
       "      <td>-0.005848</td>\n",
       "      <td>0.003879</td>\n",
       "      <td>-1.507645</td>\n",
       "      <td>0.131649</td>\n",
       "      <td>-0.013450</td>\n",
       "      <td>0.001755</td>\n",
       "      <td>0.483358</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>372</th>\n",
       "      <td>14049</td>\n",
       "      <td>EG8_00031</td>\n",
       "      <td>VII</td>\n",
       "      <td>472612</td>\n",
       "      <td>ERG4</td>\n",
       "      <td>upstream_gene_variant</td>\n",
       "      <td>T</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>243</td>\n",
       "      <td>80</td>\n",
       "      <td>LOV</td>\n",
       "      <td>0.007146</td>\n",
       "      <td>0.001517</td>\n",
       "      <td>4.712358</td>\n",
       "      <td>0.000002</td>\n",
       "      <td>0.004174</td>\n",
       "      <td>0.010119</td>\n",
       "      <td>0.000071</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>373</th>\n",
       "      <td>14049</td>\n",
       "      <td>EG8_00031</td>\n",
       "      <td>VII</td>\n",
       "      <td>472612</td>\n",
       "      <td>ERG4</td>\n",
       "      <td>upstream_gene_variant</td>\n",
       "      <td>T</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>243</td>\n",
       "      <td>81</td>\n",
       "      <td>NACL</td>\n",
       "      <td>0.001298</td>\n",
       "      <td>0.001212</td>\n",
       "      <td>1.071053</td>\n",
       "      <td>0.284148</td>\n",
       "      <td>-0.001077</td>\n",
       "      <td>0.003673</td>\n",
       "      <td>0.668798</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>374</th>\n",
       "      <td>14049</td>\n",
       "      <td>EG8_00031</td>\n",
       "      <td>VII</td>\n",
       "      <td>472612</td>\n",
       "      <td>ERG4</td>\n",
       "      <td>upstream_gene_variant</td>\n",
       "      <td>T</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>243</td>\n",
       "      <td>82</td>\n",
       "      <td>SC</td>\n",
       "      <td>-0.002652</td>\n",
       "      <td>0.001110</td>\n",
       "      <td>-2.390015</td>\n",
       "      <td>0.016850</td>\n",
       "      <td>-0.004827</td>\n",
       "      <td>-0.000477</td>\n",
       "      <td>0.160913</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>375</th>\n",
       "      <td>14049</td>\n",
       "      <td>EG8_00031</td>\n",
       "      <td>VII</td>\n",
       "      <td>472612</td>\n",
       "      <td>ERG4</td>\n",
       "      <td>upstream_gene_variant</td>\n",
       "      <td>T</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>243</td>\n",
       "      <td>83</td>\n",
       "      <td>TBF</td>\n",
       "      <td>0.000283</td>\n",
       "      <td>0.000842</td>\n",
       "      <td>0.335509</td>\n",
       "      <td>0.737242</td>\n",
       "      <td>-0.001368</td>\n",
       "      <td>0.001933</td>\n",
       "      <td>0.926144</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "     Unnamed: 0_x     var_id chrom  SNP_chr_pos  Gene            Consequence  \\\n",
       "364         14043  EG8_00030   VII       472612  ERG4  upstream_gene_variant   \n",
       "365         14043  EG8_00030   VII       472612  ERG4  upstream_gene_variant   \n",
       "366         14043  EG8_00030   VII       472612  ERG4  upstream_gene_variant   \n",
       "367         14043  EG8_00030   VII       472612  ERG4  upstream_gene_variant   \n",
       "368         14043  EG8_00030   VII       472612  ERG4  upstream_gene_variant   \n",
       "369         14043  EG8_00030   VII       472612  ERG4  upstream_gene_variant   \n",
       "370         14049  EG8_00031   VII       472612  ERG4  upstream_gene_variant   \n",
       "371         14049  EG8_00031   VII       472612  ERG4  upstream_gene_variant   \n",
       "372         14049  EG8_00031   VII       472612  ERG4  upstream_gene_variant   \n",
       "373         14049  EG8_00031   VII       472612  ERG4  upstream_gene_variant   \n",
       "374         14049  EG8_00031   VII       472612  ERG4  upstream_gene_variant   \n",
       "375         14049  EG8_00031   VII       472612  ERG4  upstream_gene_variant   \n",
       "\n",
       "    ALT CDS_position Protein_position Amino_acids  DISTANCE  Unnamed: 0_y  \\\n",
       "364   A            -                -           -       243            72   \n",
       "365   A            -                -           -       243            73   \n",
       "366   A            -                -           -       243            74   \n",
       "367   A            -                -           -       243            75   \n",
       "368   A            -                -           -       243            76   \n",
       "369   A            -                -           -       243            77   \n",
       "370   T            -                -           -       243            78   \n",
       "371   T            -                -           -       243            79   \n",
       "372   T            -                -           -       243            80   \n",
       "373   T            -                -           -       243            81   \n",
       "374   T            -                -           -       243            82   \n",
       "375   T            -                -           -       243            83   \n",
       "\n",
       "    condition      coef   std err         t      pval  Conf. Int. Low  \\\n",
       "364      CAFF -0.001099  0.001076 -1.021815  0.306871       -0.003207   \n",
       "365      COCL -0.003337  0.003359 -0.993492  0.320473       -0.009920   \n",
       "366       LOV  0.002025  0.001411  1.435011  0.151287       -0.000741   \n",
       "367      NACL -0.000383  0.001182 -0.323774  0.746110       -0.002699   \n",
       "368        SC -0.002316  0.001060 -2.183660  0.028990       -0.004394   \n",
       "369       TBF  0.000555  0.000846  0.656327  0.511616       -0.001103   \n",
       "370      CAFF -0.001028  0.001177 -0.873487  0.382400       -0.003335   \n",
       "371      COCL -0.005848  0.003879 -1.507645  0.131649       -0.013450   \n",
       "372       LOV  0.007146  0.001517  4.712358  0.000002        0.004174   \n",
       "373      NACL  0.001298  0.001212  1.071053  0.284148       -0.001077   \n",
       "374        SC -0.002652  0.001110 -2.390015  0.016850       -0.004827   \n",
       "375       TBF  0.000283  0.000842  0.335509  0.737242       -0.001368   \n",
       "\n",
       "     Conf. Int. Upp.      padj  \n",
       "364         0.001009  0.686635  \n",
       "365         0.003246  0.699971  \n",
       "366         0.004791  0.515410  \n",
       "367         0.001933  0.928813  \n",
       "368        -0.000237  0.225707  \n",
       "369         0.002214  0.830142  \n",
       "370         0.001279  0.751325  \n",
       "371         0.001755  0.483358  \n",
       "372         0.010119  0.000071  \n",
       "373         0.003673  0.668798  \n",
       "374        -0.000477  0.160913  \n",
       "375         0.001933  0.926144  "
      ]
     },
     "execution_count": 21,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "#ERG4 site with two upstream variants, one with a significant effect in lovastatin and one without\n",
    "\n",
    "chrom = 'VII'\n",
    "pos = 472612\n",
    "\n",
    "annotated_fits[(annotated_fits['chrom']==chrom)&(annotated_fits['SNP_chr_pos']==pos)]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 22,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Unnamed: 0_x</th>\n",
       "      <th>var_id</th>\n",
       "      <th>chrom</th>\n",
       "      <th>SNP_chr_pos</th>\n",
       "      <th>Gene</th>\n",
       "      <th>Consequence</th>\n",
       "      <th>ALT</th>\n",
       "      <th>CDS_position</th>\n",
       "      <th>Protein_position</th>\n",
       "      <th>Amino_acids</th>\n",
       "      <th>DISTANCE</th>\n",
       "      <th>Unnamed: 0_y</th>\n",
       "      <th>condition</th>\n",
       "      <th>coef</th>\n",
       "      <th>std err</th>\n",
       "      <th>t</th>\n",
       "      <th>pval</th>\n",
       "      <th>Conf. Int. Low</th>\n",
       "      <th>Conf. Int. Upp.</th>\n",
       "      <th>padj</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>946</th>\n",
       "      <td>14747</td>\n",
       "      <td>ERG_00285</td>\n",
       "      <td>VII</td>\n",
       "      <td>610071</td>\n",
       "      <td>ERG25</td>\n",
       "      <td>upstream_gene_variant</td>\n",
       "      <td>A</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>493</td>\n",
       "      <td>3667</td>\n",
       "      <td>CAFF</td>\n",
       "      <td>-0.003269</td>\n",
       "      <td>0.001526</td>\n",
       "      <td>-2.142775</td>\n",
       "      <td>0.032134</td>\n",
       "      <td>-0.006260</td>\n",
       "      <td>-0.000279</td>\n",
       "      <td>0.238992</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>947</th>\n",
       "      <td>14747</td>\n",
       "      <td>ERG_00285</td>\n",
       "      <td>VII</td>\n",
       "      <td>610071</td>\n",
       "      <td>ERG25</td>\n",
       "      <td>upstream_gene_variant</td>\n",
       "      <td>A</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>493</td>\n",
       "      <td>3668</td>\n",
       "      <td>COCL</td>\n",
       "      <td>-0.020771</td>\n",
       "      <td>0.004743</td>\n",
       "      <td>-4.379045</td>\n",
       "      <td>0.000012</td>\n",
       "      <td>-0.030068</td>\n",
       "      <td>-0.011474</td>\n",
       "      <td>0.000325</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>948</th>\n",
       "      <td>14747</td>\n",
       "      <td>ERG_00285</td>\n",
       "      <td>VII</td>\n",
       "      <td>610071</td>\n",
       "      <td>ERG25</td>\n",
       "      <td>upstream_gene_variant</td>\n",
       "      <td>A</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>493</td>\n",
       "      <td>3669</td>\n",
       "      <td>LOV</td>\n",
       "      <td>-0.003439</td>\n",
       "      <td>0.001894</td>\n",
       "      <td>-1.815888</td>\n",
       "      <td>0.069391</td>\n",
       "      <td>-0.007151</td>\n",
       "      <td>0.000273</td>\n",
       "      <td>0.354021</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>949</th>\n",
       "      <td>14747</td>\n",
       "      <td>ERG_00285</td>\n",
       "      <td>VII</td>\n",
       "      <td>610071</td>\n",
       "      <td>ERG25</td>\n",
       "      <td>upstream_gene_variant</td>\n",
       "      <td>A</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>493</td>\n",
       "      <td>3670</td>\n",
       "      <td>NACL</td>\n",
       "      <td>0.001857</td>\n",
       "      <td>0.001548</td>\n",
       "      <td>1.199645</td>\n",
       "      <td>0.230280</td>\n",
       "      <td>-0.001177</td>\n",
       "      <td>0.004892</td>\n",
       "      <td>0.612693</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>950</th>\n",
       "      <td>14747</td>\n",
       "      <td>ERG_00285</td>\n",
       "      <td>VII</td>\n",
       "      <td>610071</td>\n",
       "      <td>ERG25</td>\n",
       "      <td>upstream_gene_variant</td>\n",
       "      <td>A</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>493</td>\n",
       "      <td>3671</td>\n",
       "      <td>SC</td>\n",
       "      <td>0.001109</td>\n",
       "      <td>0.001533</td>\n",
       "      <td>0.723188</td>\n",
       "      <td>0.469566</td>\n",
       "      <td>-0.001896</td>\n",
       "      <td>0.004114</td>\n",
       "      <td>0.807993</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>951</th>\n",
       "      <td>14747</td>\n",
       "      <td>ERG_00285</td>\n",
       "      <td>VII</td>\n",
       "      <td>610071</td>\n",
       "      <td>ERG25</td>\n",
       "      <td>upstream_gene_variant</td>\n",
       "      <td>A</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>493</td>\n",
       "      <td>3672</td>\n",
       "      <td>TBF</td>\n",
       "      <td>0.002352</td>\n",
       "      <td>0.001170</td>\n",
       "      <td>2.010097</td>\n",
       "      <td>0.044424</td>\n",
       "      <td>0.000059</td>\n",
       "      <td>0.004646</td>\n",
       "      <td>0.282630</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>952</th>\n",
       "      <td>14753</td>\n",
       "      <td>ERG_00286</td>\n",
       "      <td>VII</td>\n",
       "      <td>610071</td>\n",
       "      <td>ERG25</td>\n",
       "      <td>upstream_gene_variant</td>\n",
       "      <td>T</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>493</td>\n",
       "      <td>3673</td>\n",
       "      <td>CAFF</td>\n",
       "      <td>-0.003152</td>\n",
       "      <td>0.003051</td>\n",
       "      <td>-1.033136</td>\n",
       "      <td>0.301543</td>\n",
       "      <td>-0.009133</td>\n",
       "      <td>0.002828</td>\n",
       "      <td>0.682883</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>953</th>\n",
       "      <td>14753</td>\n",
       "      <td>ERG_00286</td>\n",
       "      <td>VII</td>\n",
       "      <td>610071</td>\n",
       "      <td>ERG25</td>\n",
       "      <td>upstream_gene_variant</td>\n",
       "      <td>T</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>493</td>\n",
       "      <td>3674</td>\n",
       "      <td>COCL</td>\n",
       "      <td>-0.016885</td>\n",
       "      <td>0.010037</td>\n",
       "      <td>-1.682299</td>\n",
       "      <td>0.092514</td>\n",
       "      <td>-0.036558</td>\n",
       "      <td>0.002787</td>\n",
       "      <td>0.409315</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>954</th>\n",
       "      <td>14753</td>\n",
       "      <td>ERG_00286</td>\n",
       "      <td>VII</td>\n",
       "      <td>610071</td>\n",
       "      <td>ERG25</td>\n",
       "      <td>upstream_gene_variant</td>\n",
       "      <td>T</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>493</td>\n",
       "      <td>3675</td>\n",
       "      <td>LOV</td>\n",
       "      <td>-0.000591</td>\n",
       "      <td>0.003573</td>\n",
       "      <td>-0.165409</td>\n",
       "      <td>0.868622</td>\n",
       "      <td>-0.007595</td>\n",
       "      <td>0.006413</td>\n",
       "      <td>0.972669</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>955</th>\n",
       "      <td>14753</td>\n",
       "      <td>ERG_00286</td>\n",
       "      <td>VII</td>\n",
       "      <td>610071</td>\n",
       "      <td>ERG25</td>\n",
       "      <td>upstream_gene_variant</td>\n",
       "      <td>T</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>493</td>\n",
       "      <td>3676</td>\n",
       "      <td>NACL</td>\n",
       "      <td>0.000548</td>\n",
       "      <td>0.003514</td>\n",
       "      <td>0.156032</td>\n",
       "      <td>0.876008</td>\n",
       "      <td>-0.006339</td>\n",
       "      <td>0.007435</td>\n",
       "      <td>0.975015</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>956</th>\n",
       "      <td>14753</td>\n",
       "      <td>ERG_00286</td>\n",
       "      <td>VII</td>\n",
       "      <td>610071</td>\n",
       "      <td>ERG25</td>\n",
       "      <td>upstream_gene_variant</td>\n",
       "      <td>T</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>493</td>\n",
       "      <td>3677</td>\n",
       "      <td>SC</td>\n",
       "      <td>0.001990</td>\n",
       "      <td>0.002862</td>\n",
       "      <td>0.695213</td>\n",
       "      <td>0.486923</td>\n",
       "      <td>-0.003620</td>\n",
       "      <td>0.007600</td>\n",
       "      <td>0.818101</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>957</th>\n",
       "      <td>14753</td>\n",
       "      <td>ERG_00286</td>\n",
       "      <td>VII</td>\n",
       "      <td>610071</td>\n",
       "      <td>ERG25</td>\n",
       "      <td>upstream_gene_variant</td>\n",
       "      <td>T</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>493</td>\n",
       "      <td>3678</td>\n",
       "      <td>TBF</td>\n",
       "      <td>0.001324</td>\n",
       "      <td>0.002560</td>\n",
       "      <td>0.517067</td>\n",
       "      <td>0.605111</td>\n",
       "      <td>-0.003695</td>\n",
       "      <td>0.006342</td>\n",
       "      <td>0.871610</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "     Unnamed: 0_x     var_id chrom  SNP_chr_pos   Gene            Consequence  \\\n",
       "946         14747  ERG_00285   VII       610071  ERG25  upstream_gene_variant   \n",
       "947         14747  ERG_00285   VII       610071  ERG25  upstream_gene_variant   \n",
       "948         14747  ERG_00285   VII       610071  ERG25  upstream_gene_variant   \n",
       "949         14747  ERG_00285   VII       610071  ERG25  upstream_gene_variant   \n",
       "950         14747  ERG_00285   VII       610071  ERG25  upstream_gene_variant   \n",
       "951         14747  ERG_00285   VII       610071  ERG25  upstream_gene_variant   \n",
       "952         14753  ERG_00286   VII       610071  ERG25  upstream_gene_variant   \n",
       "953         14753  ERG_00286   VII       610071  ERG25  upstream_gene_variant   \n",
       "954         14753  ERG_00286   VII       610071  ERG25  upstream_gene_variant   \n",
       "955         14753  ERG_00286   VII       610071  ERG25  upstream_gene_variant   \n",
       "956         14753  ERG_00286   VII       610071  ERG25  upstream_gene_variant   \n",
       "957         14753  ERG_00286   VII       610071  ERG25  upstream_gene_variant   \n",
       "\n",
       "    ALT CDS_position Protein_position Amino_acids  DISTANCE  Unnamed: 0_y  \\\n",
       "946   A            -                -           -       493          3667   \n",
       "947   A            -                -           -       493          3668   \n",
       "948   A            -                -           -       493          3669   \n",
       "949   A            -                -           -       493          3670   \n",
       "950   A            -                -           -       493          3671   \n",
       "951   A            -                -           -       493          3672   \n",
       "952   T            -                -           -       493          3673   \n",
       "953   T            -                -           -       493          3674   \n",
       "954   T            -                -           -       493          3675   \n",
       "955   T            -                -           -       493          3676   \n",
       "956   T            -                -           -       493          3677   \n",
       "957   T            -                -           -       493          3678   \n",
       "\n",
       "    condition      coef   std err         t      pval  Conf. Int. Low  \\\n",
       "946      CAFF -0.003269  0.001526 -2.142775  0.032134       -0.006260   \n",
       "947      COCL -0.020771  0.004743 -4.379045  0.000012       -0.030068   \n",
       "948       LOV -0.003439  0.001894 -1.815888  0.069391       -0.007151   \n",
       "949      NACL  0.001857  0.001548  1.199645  0.230280       -0.001177   \n",
       "950        SC  0.001109  0.001533  0.723188  0.469566       -0.001896   \n",
       "951       TBF  0.002352  0.001170  2.010097  0.044424        0.000059   \n",
       "952      CAFF -0.003152  0.003051 -1.033136  0.301543       -0.009133   \n",
       "953      COCL -0.016885  0.010037 -1.682299  0.092514       -0.036558   \n",
       "954       LOV -0.000591  0.003573 -0.165409  0.868622       -0.007595   \n",
       "955      NACL  0.000548  0.003514  0.156032  0.876008       -0.006339   \n",
       "956        SC  0.001990  0.002862  0.695213  0.486923       -0.003620   \n",
       "957       TBF  0.001324  0.002560  0.517067  0.605111       -0.003695   \n",
       "\n",
       "     Conf. Int. Upp.      padj  \n",
       "946        -0.000279  0.238992  \n",
       "947        -0.011474  0.000325  \n",
       "948         0.000273  0.354021  \n",
       "949         0.004892  0.612693  \n",
       "950         0.004114  0.807993  \n",
       "951         0.004646  0.282630  \n",
       "952         0.002828  0.682883  \n",
       "953         0.002787  0.409315  \n",
       "954         0.006413  0.972669  \n",
       "955         0.007435  0.975015  \n",
       "956         0.007600  0.818101  \n",
       "957         0.006342  0.871610  "
      ]
     },
     "execution_count": 22,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "#ERG25 site with two upstream variants, one with a significant effect in cobalt chloride and one without \n",
    "#(though the cobalt effects are of similar magnitude)\n",
    "\n",
    "chrom = 'VII'\n",
    "pos =610071\n",
    "\n",
    "annotated_fits[(annotated_fits['chrom']==chrom)&(annotated_fits['SNP_chr_pos']==pos)]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 23,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Unnamed: 0_x</th>\n",
       "      <th>var_id</th>\n",
       "      <th>chrom</th>\n",
       "      <th>SNP_chr_pos</th>\n",
       "      <th>Gene</th>\n",
       "      <th>Consequence</th>\n",
       "      <th>ALT</th>\n",
       "      <th>CDS_position</th>\n",
       "      <th>Protein_position</th>\n",
       "      <th>Amino_acids</th>\n",
       "      <th>DISTANCE</th>\n",
       "      <th>Unnamed: 0_y</th>\n",
       "      <th>condition</th>\n",
       "      <th>coef</th>\n",
       "      <th>std err</th>\n",
       "      <th>t</th>\n",
       "      <th>pval</th>\n",
       "      <th>Conf. Int. Low</th>\n",
       "      <th>Conf. Int. Upp.</th>\n",
       "      <th>padj</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>1326</th>\n",
       "      <td>16747</td>\n",
       "      <td>ERG_00444</td>\n",
       "      <td>VII</td>\n",
       "      <td>848783</td>\n",
       "      <td>ERG1</td>\n",
       "      <td>upstream_gene_variant</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>361</td>\n",
       "      <td>3980</td>\n",
       "      <td>CAFF</td>\n",
       "      <td>-0.001571</td>\n",
       "      <td>0.000691</td>\n",
       "      <td>-2.273582</td>\n",
       "      <td>2.299332e-02</td>\n",
       "      <td>-0.002926</td>\n",
       "      <td>-0.000217</td>\n",
       "      <td>1.961712e-01</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1327</th>\n",
       "      <td>16747</td>\n",
       "      <td>ERG_00444</td>\n",
       "      <td>VII</td>\n",
       "      <td>848783</td>\n",
       "      <td>ERG1</td>\n",
       "      <td>upstream_gene_variant</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>361</td>\n",
       "      <td>3981</td>\n",
       "      <td>COCL</td>\n",
       "      <td>0.062549</td>\n",
       "      <td>0.002172</td>\n",
       "      <td>28.801341</td>\n",
       "      <td>1.196889e-181</td>\n",
       "      <td>0.058293</td>\n",
       "      <td>0.066806</td>\n",
       "      <td>1.705472e-179</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1328</th>\n",
       "      <td>16747</td>\n",
       "      <td>ERG_00444</td>\n",
       "      <td>VII</td>\n",
       "      <td>848783</td>\n",
       "      <td>ERG1</td>\n",
       "      <td>upstream_gene_variant</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>361</td>\n",
       "      <td>3982</td>\n",
       "      <td>LOV</td>\n",
       "      <td>0.003645</td>\n",
       "      <td>0.000894</td>\n",
       "      <td>4.075507</td>\n",
       "      <td>4.595049e-05</td>\n",
       "      <td>0.001892</td>\n",
       "      <td>0.005398</td>\n",
       "      <td>1.139496e-03</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1329</th>\n",
       "      <td>16747</td>\n",
       "      <td>ERG_00444</td>\n",
       "      <td>VII</td>\n",
       "      <td>848783</td>\n",
       "      <td>ERG1</td>\n",
       "      <td>upstream_gene_variant</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>361</td>\n",
       "      <td>3983</td>\n",
       "      <td>NACL</td>\n",
       "      <td>-0.001433</td>\n",
       "      <td>0.000690</td>\n",
       "      <td>-2.076500</td>\n",
       "      <td>3.785036e-02</td>\n",
       "      <td>-0.002785</td>\n",
       "      <td>-0.000080</td>\n",
       "      <td>2.603699e-01</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1330</th>\n",
       "      <td>16747</td>\n",
       "      <td>ERG_00444</td>\n",
       "      <td>VII</td>\n",
       "      <td>848783</td>\n",
       "      <td>ERG1</td>\n",
       "      <td>upstream_gene_variant</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>361</td>\n",
       "      <td>3984</td>\n",
       "      <td>SC</td>\n",
       "      <td>-0.000302</td>\n",
       "      <td>0.000678</td>\n",
       "      <td>-0.445465</td>\n",
       "      <td>6.559846e-01</td>\n",
       "      <td>-0.001630</td>\n",
       "      <td>0.001026</td>\n",
       "      <td>8.955965e-01</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1331</th>\n",
       "      <td>16747</td>\n",
       "      <td>ERG_00444</td>\n",
       "      <td>VII</td>\n",
       "      <td>848783</td>\n",
       "      <td>ERG1</td>\n",
       "      <td>upstream_gene_variant</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>361</td>\n",
       "      <td>3985</td>\n",
       "      <td>TBF</td>\n",
       "      <td>-0.100394</td>\n",
       "      <td>0.001044</td>\n",
       "      <td>-96.128673</td>\n",
       "      <td>0.000000e+00</td>\n",
       "      <td>-0.102441</td>\n",
       "      <td>-0.098347</td>\n",
       "      <td>0.000000e+00</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1332</th>\n",
       "      <td>16756</td>\n",
       "      <td>ERG_00445</td>\n",
       "      <td>VII</td>\n",
       "      <td>848783</td>\n",
       "      <td>ERG1</td>\n",
       "      <td>upstream_gene_variant</td>\n",
       "      <td>GC</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>360</td>\n",
       "      <td>3986</td>\n",
       "      <td>CAFF</td>\n",
       "      <td>-0.001859</td>\n",
       "      <td>0.001070</td>\n",
       "      <td>-1.737402</td>\n",
       "      <td>8.231933e-02</td>\n",
       "      <td>-0.003957</td>\n",
       "      <td>0.000238</td>\n",
       "      <td>3.850863e-01</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1333</th>\n",
       "      <td>16756</td>\n",
       "      <td>ERG_00445</td>\n",
       "      <td>VII</td>\n",
       "      <td>848783</td>\n",
       "      <td>ERG1</td>\n",
       "      <td>upstream_gene_variant</td>\n",
       "      <td>GC</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>360</td>\n",
       "      <td>3987</td>\n",
       "      <td>COCL</td>\n",
       "      <td>0.044502</td>\n",
       "      <td>0.003226</td>\n",
       "      <td>13.795166</td>\n",
       "      <td>2.992865e-43</td>\n",
       "      <td>0.038179</td>\n",
       "      <td>0.050824</td>\n",
       "      <td>2.035375e-41</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1334</th>\n",
       "      <td>16756</td>\n",
       "      <td>ERG_00445</td>\n",
       "      <td>VII</td>\n",
       "      <td>848783</td>\n",
       "      <td>ERG1</td>\n",
       "      <td>upstream_gene_variant</td>\n",
       "      <td>GC</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>360</td>\n",
       "      <td>3988</td>\n",
       "      <td>LOV</td>\n",
       "      <td>0.002868</td>\n",
       "      <td>0.001359</td>\n",
       "      <td>2.111320</td>\n",
       "      <td>3.474738e-02</td>\n",
       "      <td>0.000206</td>\n",
       "      <td>0.005531</td>\n",
       "      <td>2.495860e-01</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1335</th>\n",
       "      <td>16756</td>\n",
       "      <td>ERG_00445</td>\n",
       "      <td>VII</td>\n",
       "      <td>848783</td>\n",
       "      <td>ERG1</td>\n",
       "      <td>upstream_gene_variant</td>\n",
       "      <td>GC</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>360</td>\n",
       "      <td>3989</td>\n",
       "      <td>NACL</td>\n",
       "      <td>0.000702</td>\n",
       "      <td>0.001101</td>\n",
       "      <td>0.637268</td>\n",
       "      <td>5.239521e-01</td>\n",
       "      <td>-0.001456</td>\n",
       "      <td>0.002860</td>\n",
       "      <td>8.378120e-01</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1336</th>\n",
       "      <td>16756</td>\n",
       "      <td>ERG_00445</td>\n",
       "      <td>VII</td>\n",
       "      <td>848783</td>\n",
       "      <td>ERG1</td>\n",
       "      <td>upstream_gene_variant</td>\n",
       "      <td>GC</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>360</td>\n",
       "      <td>3990</td>\n",
       "      <td>SC</td>\n",
       "      <td>0.000785</td>\n",
       "      <td>0.001266</td>\n",
       "      <td>0.620234</td>\n",
       "      <td>5.351052e-01</td>\n",
       "      <td>-0.001696</td>\n",
       "      <td>0.003267</td>\n",
       "      <td>8.422578e-01</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1337</th>\n",
       "      <td>16756</td>\n",
       "      <td>ERG_00445</td>\n",
       "      <td>VII</td>\n",
       "      <td>848783</td>\n",
       "      <td>ERG1</td>\n",
       "      <td>upstream_gene_variant</td>\n",
       "      <td>GC</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>360</td>\n",
       "      <td>3991</td>\n",
       "      <td>TBF</td>\n",
       "      <td>-0.078456</td>\n",
       "      <td>0.001455</td>\n",
       "      <td>-53.915900</td>\n",
       "      <td>0.000000e+00</td>\n",
       "      <td>-0.081309</td>\n",
       "      <td>-0.075604</td>\n",
       "      <td>0.000000e+00</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "      Unnamed: 0_x     var_id chrom  SNP_chr_pos  Gene            Consequence  \\\n",
       "1326         16747  ERG_00444   VII       848783  ERG1  upstream_gene_variant   \n",
       "1327         16747  ERG_00444   VII       848783  ERG1  upstream_gene_variant   \n",
       "1328         16747  ERG_00444   VII       848783  ERG1  upstream_gene_variant   \n",
       "1329         16747  ERG_00444   VII       848783  ERG1  upstream_gene_variant   \n",
       "1330         16747  ERG_00444   VII       848783  ERG1  upstream_gene_variant   \n",
       "1331         16747  ERG_00444   VII       848783  ERG1  upstream_gene_variant   \n",
       "1332         16756  ERG_00445   VII       848783  ERG1  upstream_gene_variant   \n",
       "1333         16756  ERG_00445   VII       848783  ERG1  upstream_gene_variant   \n",
       "1334         16756  ERG_00445   VII       848783  ERG1  upstream_gene_variant   \n",
       "1335         16756  ERG_00445   VII       848783  ERG1  upstream_gene_variant   \n",
       "1336         16756  ERG_00445   VII       848783  ERG1  upstream_gene_variant   \n",
       "1337         16756  ERG_00445   VII       848783  ERG1  upstream_gene_variant   \n",
       "\n",
       "     ALT CDS_position Protein_position Amino_acids  DISTANCE  Unnamed: 0_y  \\\n",
       "1326   -            -                -           -       361          3980   \n",
       "1327   -            -                -           -       361          3981   \n",
       "1328   -            -                -           -       361          3982   \n",
       "1329   -            -                -           -       361          3983   \n",
       "1330   -            -                -           -       361          3984   \n",
       "1331   -            -                -           -       361          3985   \n",
       "1332  GC            -                -           -       360          3986   \n",
       "1333  GC            -                -           -       360          3987   \n",
       "1334  GC            -                -           -       360          3988   \n",
       "1335  GC            -                -           -       360          3989   \n",
       "1336  GC            -                -           -       360          3990   \n",
       "1337  GC            -                -           -       360          3991   \n",
       "\n",
       "     condition      coef   std err          t           pval  Conf. Int. Low  \\\n",
       "1326      CAFF -0.001571  0.000691  -2.273582   2.299332e-02       -0.002926   \n",
       "1327      COCL  0.062549  0.002172  28.801341  1.196889e-181        0.058293   \n",
       "1328       LOV  0.003645  0.000894   4.075507   4.595049e-05        0.001892   \n",
       "1329      NACL -0.001433  0.000690  -2.076500   3.785036e-02       -0.002785   \n",
       "1330        SC -0.000302  0.000678  -0.445465   6.559846e-01       -0.001630   \n",
       "1331       TBF -0.100394  0.001044 -96.128673   0.000000e+00       -0.102441   \n",
       "1332      CAFF -0.001859  0.001070  -1.737402   8.231933e-02       -0.003957   \n",
       "1333      COCL  0.044502  0.003226  13.795166   2.992865e-43        0.038179   \n",
       "1334       LOV  0.002868  0.001359   2.111320   3.474738e-02        0.000206   \n",
       "1335      NACL  0.000702  0.001101   0.637268   5.239521e-01       -0.001456   \n",
       "1336        SC  0.000785  0.001266   0.620234   5.351052e-01       -0.001696   \n",
       "1337       TBF -0.078456  0.001455 -53.915900   0.000000e+00       -0.081309   \n",
       "\n",
       "      Conf. Int. Upp.           padj  \n",
       "1326        -0.000217   1.961712e-01  \n",
       "1327         0.066806  1.705472e-179  \n",
       "1328         0.005398   1.139496e-03  \n",
       "1329        -0.000080   2.603699e-01  \n",
       "1330         0.001026   8.955965e-01  \n",
       "1331        -0.098347   0.000000e+00  \n",
       "1332         0.000238   3.850863e-01  \n",
       "1333         0.050824   2.035375e-41  \n",
       "1334         0.005531   2.495860e-01  \n",
       "1335         0.002860   8.378120e-01  \n",
       "1336         0.003267   8.422578e-01  \n",
       "1337        -0.075604   0.000000e+00  "
      ]
     },
     "execution_count": 23,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "#ERG1 site with two upstream variants, both with significant positive effects in cobalt chloride\n",
    "#and both with significant negative effects in terbinafine\n",
    "\n",
    "chrom = 'VII'\n",
    "pos =848783\n",
    "\n",
    "annotated_fits[(annotated_fits['chrom']==chrom)&(annotated_fits['SNP_chr_pos']==pos)]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 24,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>var_id</th>\n",
       "      <th>chrom</th>\n",
       "      <th>SNP_chr_pos</th>\n",
       "      <th>REF</th>\n",
       "      <th>ALT</th>\n",
       "      <th>guide_id</th>\n",
       "      <th>guide_strand</th>\n",
       "      <th>guide0_chr_pos</th>\n",
       "      <th>guide_cut_chr_pos</th>\n",
       "      <th>SNP_pos_in_guide</th>\n",
       "      <th>...</th>\n",
       "      <th>barcode_id</th>\n",
       "      <th>oligo_seq</th>\n",
       "      <th>oligo_id</th>\n",
       "      <th>chromosome</th>\n",
       "      <th>primer_F</th>\n",
       "      <th>primer_R</th>\n",
       "      <th>template_len</th>\n",
       "      <th>left_tm</th>\n",
       "      <th>right_tm</th>\n",
       "      <th>product_seq</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>637</th>\n",
       "      <td>ERG_00444</td>\n",
       "      <td>VII</td>\n",
       "      <td>848783</td>\n",
       "      <td>AC</td>\n",
       "      <td>[A]</td>\n",
       "      <td>guide_638</td>\n",
       "      <td>+</td>\n",
       "      <td>848789</td>\n",
       "      <td>848786</td>\n",
       "      <td>-7</td>\n",
       "      <td>...</td>\n",
       "      <td>002_012</td>\n",
       "      <td>GTTGCAGTTAGCTAACAGGCCATGCACAATGGCTTATGCATGCAGC...</td>\n",
       "      <td>ergosterol#guide_638#guide_638:REF2ALT:offset1...</td>\n",
       "      <td>chrVII</td>\n",
       "      <td>CCTACACGACGCTCTTCCGATCTCTTATACCCGACCGAGAGCAG</td>\n",
       "      <td>TTCAGACGTGTGCTCTTCCGATCTGCGAATGGGAATCGTTCTGC</td>\n",
       "      <td>199.0</td>\n",
       "      <td>59.732739</td>\n",
       "      <td>59.97189</td>\n",
       "      <td>CTTATACCCGACCGAGAGCAGCCACGGCATGCATCGACTACGAGCG...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>638</th>\n",
       "      <td>ERG_00444</td>\n",
       "      <td>VII</td>\n",
       "      <td>848783</td>\n",
       "      <td>AC</td>\n",
       "      <td>[A]</td>\n",
       "      <td>guide_639</td>\n",
       "      <td>-</td>\n",
       "      <td>848776</td>\n",
       "      <td>848780</td>\n",
       "      <td>-6</td>\n",
       "      <td>...</td>\n",
       "      <td>002_013</td>\n",
       "      <td>GTTGCAGTTAGCTAACAGGCCATGCCGTAGTAGTCTCGCATGCAGC...</td>\n",
       "      <td>ergosterol#guide_639#guide_639:REF2ALT:offset-...</td>\n",
       "      <td>chrVII</td>\n",
       "      <td>CCTACACGACGCTCTTCCGATCTCTTATACCCGACCGAGAGCAG</td>\n",
       "      <td>TTCAGACGTGTGCTCTTCCGATCTGCGAATGGGAATCGTTCTGC</td>\n",
       "      <td>199.0</td>\n",
       "      <td>59.732739</td>\n",
       "      <td>59.97189</td>\n",
       "      <td>CTTATACCCGACCGAGAGCAGCCACGGCATGCATCGACTACGAGCG...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>639</th>\n",
       "      <td>ERG_00445</td>\n",
       "      <td>VII</td>\n",
       "      <td>848783</td>\n",
       "      <td>AC</td>\n",
       "      <td>[GC]</td>\n",
       "      <td>guide_640</td>\n",
       "      <td>+</td>\n",
       "      <td>848789</td>\n",
       "      <td>848786</td>\n",
       "      <td>-7</td>\n",
       "      <td>...</td>\n",
       "      <td>020_024</td>\n",
       "      <td>GTTGCAGTTAGCTAACAGGCCATGCTACCGGCAGCTAGCATGCAGC...</td>\n",
       "      <td>ergosterol#guide_640#guide_640:REF2ALT:offset1...</td>\n",
       "      <td>chrVII</td>\n",
       "      <td>CCTACACGACGCTCTTCCGATCTCTTATACCCGACCGAGAGCAG</td>\n",
       "      <td>TTCAGACGTGTGCTCTTCCGATCTGCGAATGGGAATCGTTCTGC</td>\n",
       "      <td>199.0</td>\n",
       "      <td>59.732739</td>\n",
       "      <td>59.97189</td>\n",
       "      <td>CTTATACCCGACCGAGAGCAGCCACGGCATGCATCGACTACGAGCG...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>640</th>\n",
       "      <td>ERG_00445</td>\n",
       "      <td>VII</td>\n",
       "      <td>848783</td>\n",
       "      <td>AC</td>\n",
       "      <td>[GC]</td>\n",
       "      <td>guide_641</td>\n",
       "      <td>-</td>\n",
       "      <td>848776</td>\n",
       "      <td>848780</td>\n",
       "      <td>-6</td>\n",
       "      <td>...</td>\n",
       "      <td>020_025</td>\n",
       "      <td>GTTGCAGTTAGCTAACAGGCCATGCAGAATAGACGGCGCATGCAGC...</td>\n",
       "      <td>ergosterol#guide_641#guide_641:REF2ALT:offset-...</td>\n",
       "      <td>chrVII</td>\n",
       "      <td>CCTACACGACGCTCTTCCGATCTCTTATACCCGACCGAGAGCAG</td>\n",
       "      <td>TTCAGACGTGTGCTCTTCCGATCTGCGAATGGGAATCGTTCTGC</td>\n",
       "      <td>199.0</td>\n",
       "      <td>59.732739</td>\n",
       "      <td>59.97189</td>\n",
       "      <td>CTTATACCCGACCGAGAGCAGCCACGGCATGCATCGACTACGAGCG...</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>4 rows  50 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "        var_id chrom  SNP_chr_pos REF   ALT   guide_id guide_strand  \\\n",
       "637  ERG_00444   VII       848783  AC   [A]  guide_638            +   \n",
       "638  ERG_00444   VII       848783  AC   [A]  guide_639            -   \n",
       "639  ERG_00445   VII       848783  AC  [GC]  guide_640            +   \n",
       "640  ERG_00445   VII       848783  AC  [GC]  guide_641            -   \n",
       "\n",
       "     guide0_chr_pos  guide_cut_chr_pos  SNP_pos_in_guide  ...  barcode_id  \\\n",
       "637          848789             848786                -7  ...     002_012   \n",
       "638          848776             848780                -6  ...     002_013   \n",
       "639          848789             848786                -7  ...     020_024   \n",
       "640          848776             848780                -6  ...     020_025   \n",
       "\n",
       "                                             oligo_seq  \\\n",
       "637  GTTGCAGTTAGCTAACAGGCCATGCACAATGGCTTATGCATGCAGC...   \n",
       "638  GTTGCAGTTAGCTAACAGGCCATGCCGTAGTAGTCTCGCATGCAGC...   \n",
       "639  GTTGCAGTTAGCTAACAGGCCATGCTACCGGCAGCTAGCATGCAGC...   \n",
       "640  GTTGCAGTTAGCTAACAGGCCATGCAGAATAGACGGCGCATGCAGC...   \n",
       "\n",
       "                                              oligo_id chromosome  \\\n",
       "637  ergosterol#guide_638#guide_638:REF2ALT:offset1...     chrVII   \n",
       "638  ergosterol#guide_639#guide_639:REF2ALT:offset-...     chrVII   \n",
       "639  ergosterol#guide_640#guide_640:REF2ALT:offset1...     chrVII   \n",
       "640  ergosterol#guide_641#guide_641:REF2ALT:offset-...     chrVII   \n",
       "\n",
       "                                         primer_F  \\\n",
       "637  CCTACACGACGCTCTTCCGATCTCTTATACCCGACCGAGAGCAG   \n",
       "638  CCTACACGACGCTCTTCCGATCTCTTATACCCGACCGAGAGCAG   \n",
       "639  CCTACACGACGCTCTTCCGATCTCTTATACCCGACCGAGAGCAG   \n",
       "640  CCTACACGACGCTCTTCCGATCTCTTATACCCGACCGAGAGCAG   \n",
       "\n",
       "                                         primer_R  template_len    left_tm  \\\n",
       "637  TTCAGACGTGTGCTCTTCCGATCTGCGAATGGGAATCGTTCTGC         199.0  59.732739   \n",
       "638  TTCAGACGTGTGCTCTTCCGATCTGCGAATGGGAATCGTTCTGC         199.0  59.732739   \n",
       "639  TTCAGACGTGTGCTCTTCCGATCTGCGAATGGGAATCGTTCTGC         199.0  59.732739   \n",
       "640  TTCAGACGTGTGCTCTTCCGATCTGCGAATGGGAATCGTTCTGC         199.0  59.732739   \n",
       "\n",
       "     right_tm                                        product_seq  \n",
       "637  59.97189  CTTATACCCGACCGAGAGCAGCCACGGCATGCATCGACTACGAGCG...  \n",
       "638  59.97189  CTTATACCCGACCGAGAGCAGCCACGGCATGCATCGACTACGAGCG...  \n",
       "639  59.97189  CTTATACCCGACCGAGAGCAGCCACGGCATGCATCGACTACGAGCG...  \n",
       "640  59.97189  CTTATACCCGACCGAGAGCAGCCACGGCATGCATCGACTACGAGCG...  \n",
       "\n",
       "[4 rows x 50 columns]"
      ]
     },
     "execution_count": 24,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "oli_info[oli_info['var_id'].isin(['ERG_00444','ERG_00445'])]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 25,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Unnamed: 0_x</th>\n",
       "      <th>var_id</th>\n",
       "      <th>chrom</th>\n",
       "      <th>SNP_chr_pos</th>\n",
       "      <th>Gene</th>\n",
       "      <th>Consequence</th>\n",
       "      <th>ALT</th>\n",
       "      <th>CDS_position</th>\n",
       "      <th>Protein_position</th>\n",
       "      <th>Amino_acids</th>\n",
       "      <th>DISTANCE</th>\n",
       "      <th>Unnamed: 0_y</th>\n",
       "      <th>condition</th>\n",
       "      <th>coef</th>\n",
       "      <th>std err</th>\n",
       "      <th>t</th>\n",
       "      <th>pval</th>\n",
       "      <th>Conf. Int. Low</th>\n",
       "      <th>Conf. Int. Upp.</th>\n",
       "      <th>padj</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>1784</th>\n",
       "      <td>21055</td>\n",
       "      <td>ERG_00574</td>\n",
       "      <td>VIII</td>\n",
       "      <td>190196</td>\n",
       "      <td>NCP1</td>\n",
       "      <td>upstream_gene_variant</td>\n",
       "      <td>GG</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>346</td>\n",
       "      <td>4148</td>\n",
       "      <td>CAFF</td>\n",
       "      <td>-0.002521</td>\n",
       "      <td>0.001860</td>\n",
       "      <td>-1.355604</td>\n",
       "      <td>1.752282e-01</td>\n",
       "      <td>-0.006166</td>\n",
       "      <td>0.001124</td>\n",
       "      <td>5.456751e-01</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1785</th>\n",
       "      <td>21055</td>\n",
       "      <td>ERG_00574</td>\n",
       "      <td>VIII</td>\n",
       "      <td>190196</td>\n",
       "      <td>NCP1</td>\n",
       "      <td>upstream_gene_variant</td>\n",
       "      <td>GG</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>346</td>\n",
       "      <td>4149</td>\n",
       "      <td>COCL</td>\n",
       "      <td>0.003521</td>\n",
       "      <td>0.004998</td>\n",
       "      <td>0.704574</td>\n",
       "      <td>4.810768e-01</td>\n",
       "      <td>-0.006274</td>\n",
       "      <td>0.013317</td>\n",
       "      <td>8.160278e-01</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1786</th>\n",
       "      <td>21055</td>\n",
       "      <td>ERG_00574</td>\n",
       "      <td>VIII</td>\n",
       "      <td>190196</td>\n",
       "      <td>NCP1</td>\n",
       "      <td>upstream_gene_variant</td>\n",
       "      <td>GG</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>346</td>\n",
       "      <td>4150</td>\n",
       "      <td>LOV</td>\n",
       "      <td>-0.002644</td>\n",
       "      <td>0.002137</td>\n",
       "      <td>-1.237204</td>\n",
       "      <td>2.160142e-01</td>\n",
       "      <td>-0.006834</td>\n",
       "      <td>0.001545</td>\n",
       "      <td>5.977753e-01</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1787</th>\n",
       "      <td>21055</td>\n",
       "      <td>ERG_00574</td>\n",
       "      <td>VIII</td>\n",
       "      <td>190196</td>\n",
       "      <td>NCP1</td>\n",
       "      <td>upstream_gene_variant</td>\n",
       "      <td>GG</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>346</td>\n",
       "      <td>4151</td>\n",
       "      <td>NACL</td>\n",
       "      <td>-0.001430</td>\n",
       "      <td>0.001851</td>\n",
       "      <td>-0.772632</td>\n",
       "      <td>4.397424e-01</td>\n",
       "      <td>-0.005057</td>\n",
       "      <td>0.002197</td>\n",
       "      <td>7.897623e-01</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1788</th>\n",
       "      <td>21055</td>\n",
       "      <td>ERG_00574</td>\n",
       "      <td>VIII</td>\n",
       "      <td>190196</td>\n",
       "      <td>NCP1</td>\n",
       "      <td>upstream_gene_variant</td>\n",
       "      <td>GG</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>346</td>\n",
       "      <td>4152</td>\n",
       "      <td>SC</td>\n",
       "      <td>0.001149</td>\n",
       "      <td>0.001725</td>\n",
       "      <td>0.666115</td>\n",
       "      <td>5.053391e-01</td>\n",
       "      <td>-0.002232</td>\n",
       "      <td>0.004529</td>\n",
       "      <td>8.272109e-01</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1789</th>\n",
       "      <td>21055</td>\n",
       "      <td>ERG_00574</td>\n",
       "      <td>VIII</td>\n",
       "      <td>190196</td>\n",
       "      <td>NCP1</td>\n",
       "      <td>upstream_gene_variant</td>\n",
       "      <td>GG</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>346</td>\n",
       "      <td>4153</td>\n",
       "      <td>TBF</td>\n",
       "      <td>-0.001380</td>\n",
       "      <td>0.001399</td>\n",
       "      <td>-0.986421</td>\n",
       "      <td>3.239292e-01</td>\n",
       "      <td>-0.004122</td>\n",
       "      <td>0.001362</td>\n",
       "      <td>7.034137e-01</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1790</th>\n",
       "      <td>21062</td>\n",
       "      <td>ERG_00575</td>\n",
       "      <td>VIII</td>\n",
       "      <td>190196</td>\n",
       "      <td>NCP1</td>\n",
       "      <td>upstream_gene_variant</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>346</td>\n",
       "      <td>4154</td>\n",
       "      <td>CAFF</td>\n",
       "      <td>0.001221</td>\n",
       "      <td>0.000685</td>\n",
       "      <td>1.781216</td>\n",
       "      <td>7.488032e-02</td>\n",
       "      <td>-0.000123</td>\n",
       "      <td>0.002564</td>\n",
       "      <td>3.674500e-01</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1791</th>\n",
       "      <td>21062</td>\n",
       "      <td>ERG_00575</td>\n",
       "      <td>VIII</td>\n",
       "      <td>190196</td>\n",
       "      <td>NCP1</td>\n",
       "      <td>upstream_gene_variant</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>346</td>\n",
       "      <td>4155</td>\n",
       "      <td>COCL</td>\n",
       "      <td>0.001679</td>\n",
       "      <td>0.002432</td>\n",
       "      <td>0.690349</td>\n",
       "      <td>4.899764e-01</td>\n",
       "      <td>-0.003088</td>\n",
       "      <td>0.006446</td>\n",
       "      <td>8.198393e-01</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1792</th>\n",
       "      <td>21062</td>\n",
       "      <td>ERG_00575</td>\n",
       "      <td>VIII</td>\n",
       "      <td>190196</td>\n",
       "      <td>NCP1</td>\n",
       "      <td>upstream_gene_variant</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>346</td>\n",
       "      <td>4156</td>\n",
       "      <td>LOV</td>\n",
       "      <td>-0.002831</td>\n",
       "      <td>0.000889</td>\n",
       "      <td>-3.183045</td>\n",
       "      <td>1.457805e-03</td>\n",
       "      <td>-0.004575</td>\n",
       "      <td>-0.001088</td>\n",
       "      <td>2.522783e-02</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1793</th>\n",
       "      <td>21062</td>\n",
       "      <td>ERG_00575</td>\n",
       "      <td>VIII</td>\n",
       "      <td>190196</td>\n",
       "      <td>NCP1</td>\n",
       "      <td>upstream_gene_variant</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>346</td>\n",
       "      <td>4157</td>\n",
       "      <td>NACL</td>\n",
       "      <td>-0.001187</td>\n",
       "      <td>0.000670</td>\n",
       "      <td>-1.772269</td>\n",
       "      <td>7.635310e-02</td>\n",
       "      <td>-0.002499</td>\n",
       "      <td>0.000126</td>\n",
       "      <td>3.700982e-01</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1794</th>\n",
       "      <td>21062</td>\n",
       "      <td>ERG_00575</td>\n",
       "      <td>VIII</td>\n",
       "      <td>190196</td>\n",
       "      <td>NCP1</td>\n",
       "      <td>upstream_gene_variant</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>346</td>\n",
       "      <td>4158</td>\n",
       "      <td>SC</td>\n",
       "      <td>-0.001120</td>\n",
       "      <td>0.000677</td>\n",
       "      <td>-1.652974</td>\n",
       "      <td>9.833947e-02</td>\n",
       "      <td>-0.002447</td>\n",
       "      <td>0.000208</td>\n",
       "      <td>4.209796e-01</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1795</th>\n",
       "      <td>21062</td>\n",
       "      <td>ERG_00575</td>\n",
       "      <td>VIII</td>\n",
       "      <td>190196</td>\n",
       "      <td>NCP1</td>\n",
       "      <td>upstream_gene_variant</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>346</td>\n",
       "      <td>4159</td>\n",
       "      <td>TBF</td>\n",
       "      <td>-0.000125</td>\n",
       "      <td>0.000528</td>\n",
       "      <td>-0.235648</td>\n",
       "      <td>8.137064e-01</td>\n",
       "      <td>-0.001160</td>\n",
       "      <td>0.000911</td>\n",
       "      <td>9.569818e-01</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1796</th>\n",
       "      <td>21069</td>\n",
       "      <td>ERG_00576</td>\n",
       "      <td>VIII</td>\n",
       "      <td>190196</td>\n",
       "      <td>NCP1</td>\n",
       "      <td>upstream_gene_variant</td>\n",
       "      <td>TGTG</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>346</td>\n",
       "      <td>4160</td>\n",
       "      <td>CAFF</td>\n",
       "      <td>-0.000350</td>\n",
       "      <td>0.000645</td>\n",
       "      <td>-0.542675</td>\n",
       "      <td>5.873550e-01</td>\n",
       "      <td>-0.001615</td>\n",
       "      <td>0.000915</td>\n",
       "      <td>8.642604e-01</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1797</th>\n",
       "      <td>21069</td>\n",
       "      <td>ERG_00576</td>\n",
       "      <td>VIII</td>\n",
       "      <td>190196</td>\n",
       "      <td>NCP1</td>\n",
       "      <td>upstream_gene_variant</td>\n",
       "      <td>TGTG</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>346</td>\n",
       "      <td>4161</td>\n",
       "      <td>COCL</td>\n",
       "      <td>-0.003875</td>\n",
       "      <td>0.002116</td>\n",
       "      <td>-1.831390</td>\n",
       "      <td>6.704540e-02</td>\n",
       "      <td>-0.008022</td>\n",
       "      <td>0.000272</td>\n",
       "      <td>3.489082e-01</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1798</th>\n",
       "      <td>21069</td>\n",
       "      <td>ERG_00576</td>\n",
       "      <td>VIII</td>\n",
       "      <td>190196</td>\n",
       "      <td>NCP1</td>\n",
       "      <td>upstream_gene_variant</td>\n",
       "      <td>TGTG</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>346</td>\n",
       "      <td>4162</td>\n",
       "      <td>LOV</td>\n",
       "      <td>0.000978</td>\n",
       "      <td>0.000822</td>\n",
       "      <td>1.189322</td>\n",
       "      <td>2.343161e-01</td>\n",
       "      <td>-0.000634</td>\n",
       "      <td>0.002590</td>\n",
       "      <td>6.181372e-01</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1799</th>\n",
       "      <td>21069</td>\n",
       "      <td>ERG_00576</td>\n",
       "      <td>VIII</td>\n",
       "      <td>190196</td>\n",
       "      <td>NCP1</td>\n",
       "      <td>upstream_gene_variant</td>\n",
       "      <td>TGTG</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>346</td>\n",
       "      <td>4163</td>\n",
       "      <td>NACL</td>\n",
       "      <td>0.000190</td>\n",
       "      <td>0.000644</td>\n",
       "      <td>0.294982</td>\n",
       "      <td>7.680084e-01</td>\n",
       "      <td>-0.001072</td>\n",
       "      <td>0.001452</td>\n",
       "      <td>9.389094e-01</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1800</th>\n",
       "      <td>21069</td>\n",
       "      <td>ERG_00576</td>\n",
       "      <td>VIII</td>\n",
       "      <td>190196</td>\n",
       "      <td>NCP1</td>\n",
       "      <td>upstream_gene_variant</td>\n",
       "      <td>TGTG</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>346</td>\n",
       "      <td>4164</td>\n",
       "      <td>SC</td>\n",
       "      <td>-0.000289</td>\n",
       "      <td>0.000627</td>\n",
       "      <td>-0.460764</td>\n",
       "      <td>6.449693e-01</td>\n",
       "      <td>-0.001518</td>\n",
       "      <td>0.000940</td>\n",
       "      <td>8.914602e-01</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1801</th>\n",
       "      <td>21069</td>\n",
       "      <td>ERG_00576</td>\n",
       "      <td>VIII</td>\n",
       "      <td>190196</td>\n",
       "      <td>NCP1</td>\n",
       "      <td>upstream_gene_variant</td>\n",
       "      <td>TGTG</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>346</td>\n",
       "      <td>4165</td>\n",
       "      <td>TBF</td>\n",
       "      <td>-0.003378</td>\n",
       "      <td>0.000494</td>\n",
       "      <td>-6.837636</td>\n",
       "      <td>8.098222e-12</td>\n",
       "      <td>-0.004346</td>\n",
       "      <td>-0.002410</td>\n",
       "      <td>3.231011e-10</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "      Unnamed: 0_x     var_id chrom  SNP_chr_pos  Gene            Consequence  \\\n",
       "1784         21055  ERG_00574  VIII       190196  NCP1  upstream_gene_variant   \n",
       "1785         21055  ERG_00574  VIII       190196  NCP1  upstream_gene_variant   \n",
       "1786         21055  ERG_00574  VIII       190196  NCP1  upstream_gene_variant   \n",
       "1787         21055  ERG_00574  VIII       190196  NCP1  upstream_gene_variant   \n",
       "1788         21055  ERG_00574  VIII       190196  NCP1  upstream_gene_variant   \n",
       "1789         21055  ERG_00574  VIII       190196  NCP1  upstream_gene_variant   \n",
       "1790         21062  ERG_00575  VIII       190196  NCP1  upstream_gene_variant   \n",
       "1791         21062  ERG_00575  VIII       190196  NCP1  upstream_gene_variant   \n",
       "1792         21062  ERG_00575  VIII       190196  NCP1  upstream_gene_variant   \n",
       "1793         21062  ERG_00575  VIII       190196  NCP1  upstream_gene_variant   \n",
       "1794         21062  ERG_00575  VIII       190196  NCP1  upstream_gene_variant   \n",
       "1795         21062  ERG_00575  VIII       190196  NCP1  upstream_gene_variant   \n",
       "1796         21069  ERG_00576  VIII       190196  NCP1  upstream_gene_variant   \n",
       "1797         21069  ERG_00576  VIII       190196  NCP1  upstream_gene_variant   \n",
       "1798         21069  ERG_00576  VIII       190196  NCP1  upstream_gene_variant   \n",
       "1799         21069  ERG_00576  VIII       190196  NCP1  upstream_gene_variant   \n",
       "1800         21069  ERG_00576  VIII       190196  NCP1  upstream_gene_variant   \n",
       "1801         21069  ERG_00576  VIII       190196  NCP1  upstream_gene_variant   \n",
       "\n",
       "       ALT CDS_position Protein_position Amino_acids  DISTANCE  Unnamed: 0_y  \\\n",
       "1784    GG            -                -           -       346          4148   \n",
       "1785    GG            -                -           -       346          4149   \n",
       "1786    GG            -                -           -       346          4150   \n",
       "1787    GG            -                -           -       346          4151   \n",
       "1788    GG            -                -           -       346          4152   \n",
       "1789    GG            -                -           -       346          4153   \n",
       "1790     -            -                -           -       346          4154   \n",
       "1791     -            -                -           -       346          4155   \n",
       "1792     -            -                -           -       346          4156   \n",
       "1793     -            -                -           -       346          4157   \n",
       "1794     -            -                -           -       346          4158   \n",
       "1795     -            -                -           -       346          4159   \n",
       "1796  TGTG            -                -           -       346          4160   \n",
       "1797  TGTG            -                -           -       346          4161   \n",
       "1798  TGTG            -                -           -       346          4162   \n",
       "1799  TGTG            -                -           -       346          4163   \n",
       "1800  TGTG            -                -           -       346          4164   \n",
       "1801  TGTG            -                -           -       346          4165   \n",
       "\n",
       "     condition      coef   std err         t          pval  Conf. Int. Low  \\\n",
       "1784      CAFF -0.002521  0.001860 -1.355604  1.752282e-01       -0.006166   \n",
       "1785      COCL  0.003521  0.004998  0.704574  4.810768e-01       -0.006274   \n",
       "1786       LOV -0.002644  0.002137 -1.237204  2.160142e-01       -0.006834   \n",
       "1787      NACL -0.001430  0.001851 -0.772632  4.397424e-01       -0.005057   \n",
       "1788        SC  0.001149  0.001725  0.666115  5.053391e-01       -0.002232   \n",
       "1789       TBF -0.001380  0.001399 -0.986421  3.239292e-01       -0.004122   \n",
       "1790      CAFF  0.001221  0.000685  1.781216  7.488032e-02       -0.000123   \n",
       "1791      COCL  0.001679  0.002432  0.690349  4.899764e-01       -0.003088   \n",
       "1792       LOV -0.002831  0.000889 -3.183045  1.457805e-03       -0.004575   \n",
       "1793      NACL -0.001187  0.000670 -1.772269  7.635310e-02       -0.002499   \n",
       "1794        SC -0.001120  0.000677 -1.652974  9.833947e-02       -0.002447   \n",
       "1795       TBF -0.000125  0.000528 -0.235648  8.137064e-01       -0.001160   \n",
       "1796      CAFF -0.000350  0.000645 -0.542675  5.873550e-01       -0.001615   \n",
       "1797      COCL -0.003875  0.002116 -1.831390  6.704540e-02       -0.008022   \n",
       "1798       LOV  0.000978  0.000822  1.189322  2.343161e-01       -0.000634   \n",
       "1799      NACL  0.000190  0.000644  0.294982  7.680084e-01       -0.001072   \n",
       "1800        SC -0.000289  0.000627 -0.460764  6.449693e-01       -0.001518   \n",
       "1801       TBF -0.003378  0.000494 -6.837636  8.098222e-12       -0.004346   \n",
       "\n",
       "      Conf. Int. Upp.          padj  \n",
       "1784         0.001124  5.456751e-01  \n",
       "1785         0.013317  8.160278e-01  \n",
       "1786         0.001545  5.977753e-01  \n",
       "1787         0.002197  7.897623e-01  \n",
       "1788         0.004529  8.272109e-01  \n",
       "1789         0.001362  7.034137e-01  \n",
       "1790         0.002564  3.674500e-01  \n",
       "1791         0.006446  8.198393e-01  \n",
       "1792        -0.001088  2.522783e-02  \n",
       "1793         0.000126  3.700982e-01  \n",
       "1794         0.000208  4.209796e-01  \n",
       "1795         0.000911  9.569818e-01  \n",
       "1796         0.000915  8.642604e-01  \n",
       "1797         0.000272  3.489082e-01  \n",
       "1798         0.002590  6.181372e-01  \n",
       "1799         0.001452  9.389094e-01  \n",
       "1800         0.000940  8.914602e-01  \n",
       "1801        -0.002410  3.231011e-10  "
      ]
     },
     "execution_count": 25,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "#NCP1 site with three upstream variants, one with a significant effect in lovastatin\n",
    "#and the other two with no significant effects\n",
    "chrom = 'VIII'\n",
    "pos =190196\n",
    "\n",
    "annotated_fits[(annotated_fits['chrom']==chrom)&(annotated_fits['SNP_chr_pos']==pos)]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 26,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>var_id</th>\n",
       "      <th>chrom</th>\n",
       "      <th>SNP_chr_pos</th>\n",
       "      <th>REF</th>\n",
       "      <th>ALT</th>\n",
       "      <th>guide_id</th>\n",
       "      <th>guide_strand</th>\n",
       "      <th>guide0_chr_pos</th>\n",
       "      <th>guide_cut_chr_pos</th>\n",
       "      <th>SNP_pos_in_guide</th>\n",
       "      <th>...</th>\n",
       "      <th>barcode_id</th>\n",
       "      <th>oligo_seq</th>\n",
       "      <th>oligo_id</th>\n",
       "      <th>chromosome</th>\n",
       "      <th>primer_F</th>\n",
       "      <th>primer_R</th>\n",
       "      <th>template_len</th>\n",
       "      <th>left_tm</th>\n",
       "      <th>right_tm</th>\n",
       "      <th>product_seq</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>881</th>\n",
       "      <td>ERG_00574</td>\n",
       "      <td>VIII</td>\n",
       "      <td>190196</td>\n",
       "      <td>CG</td>\n",
       "      <td>[CGG]</td>\n",
       "      <td>guide_883</td>\n",
       "      <td>+</td>\n",
       "      <td>190196</td>\n",
       "      <td>190193</td>\n",
       "      <td>-1</td>\n",
       "      <td>...</td>\n",
       "      <td>014_012</td>\n",
       "      <td>GTTGCAGTTAGCTAACAGGCCATGCGGTCTTTCAACAGCATGCAGC...</td>\n",
       "      <td>ergosterol#guide_883#guide_883:REF2ALT:offset1...</td>\n",
       "      <td>chrVIII</td>\n",
       "      <td>CCTACACGACGCTCTTCCGATCTCCTCAGCCTCTATGCGTACG</td>\n",
       "      <td>TTCAGACGTGTGCTCTTCCGATCTCTAAACGAGCACACACGTGA</td>\n",
       "      <td>199.0</td>\n",
       "      <td>60.039978</td>\n",
       "      <td>58.516684</td>\n",
       "      <td>CCTCAGCCTCTATGCGTACGAGCCAGACCCTTGCTCCCAGTAGATA...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>882</th>\n",
       "      <td>ERG_00574</td>\n",
       "      <td>VIII</td>\n",
       "      <td>190196</td>\n",
       "      <td>CG</td>\n",
       "      <td>[CGG]</td>\n",
       "      <td>guide_884</td>\n",
       "      <td>+</td>\n",
       "      <td>190197</td>\n",
       "      <td>190194</td>\n",
       "      <td>-2</td>\n",
       "      <td>...</td>\n",
       "      <td>014_013</td>\n",
       "      <td>GTTGCAGTTAGCTAACAGGCCATGCACTGGTCCTCTGGCATGCAGC...</td>\n",
       "      <td>ergosterol#guide_884#guide_884:REF2ALT:offset1...</td>\n",
       "      <td>chrVIII</td>\n",
       "      <td>CCTACACGACGCTCTTCCGATCTCCTCAGCCTCTATGCGTACG</td>\n",
       "      <td>TTCAGACGTGTGCTCTTCCGATCTCTAAACGAGCACACACGTGA</td>\n",
       "      <td>199.0</td>\n",
       "      <td>60.039978</td>\n",
       "      <td>58.516684</td>\n",
       "      <td>CCTCAGCCTCTATGCGTACGAGCCAGACCCTTGCTCCCAGTAGATA...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>883</th>\n",
       "      <td>ERG_00574</td>\n",
       "      <td>VIII</td>\n",
       "      <td>190196</td>\n",
       "      <td>CG</td>\n",
       "      <td>[CGG]</td>\n",
       "      <td>guide_885</td>\n",
       "      <td>+</td>\n",
       "      <td>190198</td>\n",
       "      <td>190195</td>\n",
       "      <td>-3</td>\n",
       "      <td>...</td>\n",
       "      <td>014_014</td>\n",
       "      <td>GTTGCAGTTAGCTAACAGGCCATGCCGGTCATGAACTGCATGCAGC...</td>\n",
       "      <td>ergosterol#guide_885#guide_885:REF2ALT:offset1...</td>\n",
       "      <td>chrVIII</td>\n",
       "      <td>CCTACACGACGCTCTTCCGATCTCCTCAGCCTCTATGCGTACG</td>\n",
       "      <td>TTCAGACGTGTGCTCTTCCGATCTCTAAACGAGCACACACGTGA</td>\n",
       "      <td>199.0</td>\n",
       "      <td>60.039978</td>\n",
       "      <td>58.516684</td>\n",
       "      <td>CCTCAGCCTCTATGCGTACGAGCCAGACCCTTGCTCCCAGTAGATA...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>884</th>\n",
       "      <td>ERG_00574</td>\n",
       "      <td>VIII</td>\n",
       "      <td>190196</td>\n",
       "      <td>CG</td>\n",
       "      <td>[CGG]</td>\n",
       "      <td>guide_886</td>\n",
       "      <td>-</td>\n",
       "      <td>190196</td>\n",
       "      <td>190200</td>\n",
       "      <td>1</td>\n",
       "      <td>...</td>\n",
       "      <td>014_015</td>\n",
       "      <td>GTTGCAGTTAGCTAACAGGCCATGCGTAGACAGCTCTGCATGCAGC...</td>\n",
       "      <td>ergosterol#guide_886#guide_886:REF2ALT:offset-...</td>\n",
       "      <td>chrVIII</td>\n",
       "      <td>CCTACACGACGCTCTTCCGATCTCCTCAGCCTCTATGCGTACG</td>\n",
       "      <td>TTCAGACGTGTGCTCTTCCGATCTCTAAACGAGCACACACGTGA</td>\n",
       "      <td>199.0</td>\n",
       "      <td>60.039978</td>\n",
       "      <td>58.516684</td>\n",
       "      <td>CCTCAGCCTCTATGCGTACGAGCCAGACCCTTGCTCCCAGTAGATA...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>885</th>\n",
       "      <td>ERG_00575</td>\n",
       "      <td>VIII</td>\n",
       "      <td>190196</td>\n",
       "      <td>CG</td>\n",
       "      <td>[C]</td>\n",
       "      <td>guide_887</td>\n",
       "      <td>+</td>\n",
       "      <td>190196</td>\n",
       "      <td>190193</td>\n",
       "      <td>-1</td>\n",
       "      <td>...</td>\n",
       "      <td>007_036</td>\n",
       "      <td>GTTGCAGTTAGCTAACAGGCCATGCCTACTCGACCCTGCATGCAGC...</td>\n",
       "      <td>ergosterol#guide_887#guide_887:REF2ALT:offset1...</td>\n",
       "      <td>chrVIII</td>\n",
       "      <td>CCTACACGACGCTCTTCCGATCTCCTCAGCCTCTATGCGTACG</td>\n",
       "      <td>TTCAGACGTGTGCTCTTCCGATCTCTAAACGAGCACACACGTGA</td>\n",
       "      <td>199.0</td>\n",
       "      <td>60.039978</td>\n",
       "      <td>58.516684</td>\n",
       "      <td>CCTCAGCCTCTATGCGTACGAGCCAGACCCTTGCTCCCAGTAGATA...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>886</th>\n",
       "      <td>ERG_00575</td>\n",
       "      <td>VIII</td>\n",
       "      <td>190196</td>\n",
       "      <td>CG</td>\n",
       "      <td>[C]</td>\n",
       "      <td>guide_888</td>\n",
       "      <td>+</td>\n",
       "      <td>190197</td>\n",
       "      <td>190194</td>\n",
       "      <td>-2</td>\n",
       "      <td>...</td>\n",
       "      <td>007_037</td>\n",
       "      <td>GTTGCAGTTAGCTAACAGGCCATGCGCAACTTCCACTGCATGCAGC...</td>\n",
       "      <td>ergosterol#guide_888#guide_888:REF2ALT:offset1...</td>\n",
       "      <td>chrVIII</td>\n",
       "      <td>CCTACACGACGCTCTTCCGATCTCCTCAGCCTCTATGCGTACG</td>\n",
       "      <td>TTCAGACGTGTGCTCTTCCGATCTCTAAACGAGCACACACGTGA</td>\n",
       "      <td>199.0</td>\n",
       "      <td>60.039978</td>\n",
       "      <td>58.516684</td>\n",
       "      <td>CCTCAGCCTCTATGCGTACGAGCCAGACCCTTGCTCCCAGTAGATA...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>887</th>\n",
       "      <td>ERG_00575</td>\n",
       "      <td>VIII</td>\n",
       "      <td>190196</td>\n",
       "      <td>CG</td>\n",
       "      <td>[C]</td>\n",
       "      <td>guide_889</td>\n",
       "      <td>+</td>\n",
       "      <td>190198</td>\n",
       "      <td>190195</td>\n",
       "      <td>-3</td>\n",
       "      <td>...</td>\n",
       "      <td>007_038</td>\n",
       "      <td>GTTGCAGTTAGCTAACAGGCCATGCGTCCCACTCCGTGCATGCAGC...</td>\n",
       "      <td>ergosterol#guide_889#guide_889:REF2ALT:offset1...</td>\n",
       "      <td>chrVIII</td>\n",
       "      <td>CCTACACGACGCTCTTCCGATCTCCTCAGCCTCTATGCGTACG</td>\n",
       "      <td>TTCAGACGTGTGCTCTTCCGATCTCTAAACGAGCACACACGTGA</td>\n",
       "      <td>199.0</td>\n",
       "      <td>60.039978</td>\n",
       "      <td>58.516684</td>\n",
       "      <td>CCTCAGCCTCTATGCGTACGAGCCAGACCCTTGCTCCCAGTAGATA...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>888</th>\n",
       "      <td>ERG_00575</td>\n",
       "      <td>VIII</td>\n",
       "      <td>190196</td>\n",
       "      <td>CG</td>\n",
       "      <td>[C]</td>\n",
       "      <td>guide_890</td>\n",
       "      <td>-</td>\n",
       "      <td>190196</td>\n",
       "      <td>190200</td>\n",
       "      <td>1</td>\n",
       "      <td>...</td>\n",
       "      <td>007_039</td>\n",
       "      <td>GTTGCAGTTAGCTAACAGGCCATGCGCTGTCGATGTGGCATGCAGC...</td>\n",
       "      <td>ergosterol#guide_890#guide_890:REF2ALT:offset-...</td>\n",
       "      <td>chrVIII</td>\n",
       "      <td>CCTACACGACGCTCTTCCGATCTCCTCAGCCTCTATGCGTACG</td>\n",
       "      <td>TTCAGACGTGTGCTCTTCCGATCTCTAAACGAGCACACACGTGA</td>\n",
       "      <td>199.0</td>\n",
       "      <td>60.039978</td>\n",
       "      <td>58.516684</td>\n",
       "      <td>CCTCAGCCTCTATGCGTACGAGCCAGACCCTTGCTCCCAGTAGATA...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>889</th>\n",
       "      <td>ERG_00576</td>\n",
       "      <td>VIII</td>\n",
       "      <td>190196</td>\n",
       "      <td>CG</td>\n",
       "      <td>[CTGTG]</td>\n",
       "      <td>guide_891</td>\n",
       "      <td>+</td>\n",
       "      <td>190196</td>\n",
       "      <td>190193</td>\n",
       "      <td>-1</td>\n",
       "      <td>...</td>\n",
       "      <td>001_019</td>\n",
       "      <td>GTTGCAGTTAGCTAACAGGCCATGCGCTTAGTGTGCGGCATGCAGC...</td>\n",
       "      <td>ergosterol#guide_891#guide_891:REF2ALT:offset1...</td>\n",
       "      <td>chrVIII</td>\n",
       "      <td>CCTACACGACGCTCTTCCGATCTCCTCAGCCTCTATGCGTACG</td>\n",
       "      <td>TTCAGACGTGTGCTCTTCCGATCTCTAAACGAGCACACACGTGA</td>\n",
       "      <td>199.0</td>\n",
       "      <td>60.039978</td>\n",
       "      <td>58.516684</td>\n",
       "      <td>CCTCAGCCTCTATGCGTACGAGCCAGACCCTTGCTCCCAGTAGATA...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>890</th>\n",
       "      <td>ERG_00576</td>\n",
       "      <td>VIII</td>\n",
       "      <td>190196</td>\n",
       "      <td>CG</td>\n",
       "      <td>[CTGTG]</td>\n",
       "      <td>guide_892</td>\n",
       "      <td>+</td>\n",
       "      <td>190197</td>\n",
       "      <td>190194</td>\n",
       "      <td>-2</td>\n",
       "      <td>...</td>\n",
       "      <td>001_020</td>\n",
       "      <td>GTTGCAGTTAGCTAACAGGCCATGCTTATTTGCGGGTGCATGCAGC...</td>\n",
       "      <td>ergosterol#guide_892#guide_892:REF2ALT:offset1...</td>\n",
       "      <td>chrVIII</td>\n",
       "      <td>CCTACACGACGCTCTTCCGATCTCCTCAGCCTCTATGCGTACG</td>\n",
       "      <td>TTCAGACGTGTGCTCTTCCGATCTCTAAACGAGCACACACGTGA</td>\n",
       "      <td>199.0</td>\n",
       "      <td>60.039978</td>\n",
       "      <td>58.516684</td>\n",
       "      <td>CCTCAGCCTCTATGCGTACGAGCCAGACCCTTGCTCCCAGTAGATA...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>891</th>\n",
       "      <td>ERG_00576</td>\n",
       "      <td>VIII</td>\n",
       "      <td>190196</td>\n",
       "      <td>CG</td>\n",
       "      <td>[CTGTG]</td>\n",
       "      <td>guide_893</td>\n",
       "      <td>+</td>\n",
       "      <td>190198</td>\n",
       "      <td>190195</td>\n",
       "      <td>-3</td>\n",
       "      <td>...</td>\n",
       "      <td>001_021</td>\n",
       "      <td>GTTGCAGTTAGCTAACAGGCCATGCGGACTAGGATCAGCATGCAGC...</td>\n",
       "      <td>ergosterol#guide_893#guide_893:REF2ALT:offset1...</td>\n",
       "      <td>chrVIII</td>\n",
       "      <td>CCTACACGACGCTCTTCCGATCTCCTCAGCCTCTATGCGTACG</td>\n",
       "      <td>TTCAGACGTGTGCTCTTCCGATCTCTAAACGAGCACACACGTGA</td>\n",
       "      <td>199.0</td>\n",
       "      <td>60.039978</td>\n",
       "      <td>58.516684</td>\n",
       "      <td>CCTCAGCCTCTATGCGTACGAGCCAGACCCTTGCTCCCAGTAGATA...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>892</th>\n",
       "      <td>ERG_00576</td>\n",
       "      <td>VIII</td>\n",
       "      <td>190196</td>\n",
       "      <td>CG</td>\n",
       "      <td>[CTGTG]</td>\n",
       "      <td>guide_894</td>\n",
       "      <td>-</td>\n",
       "      <td>190196</td>\n",
       "      <td>190200</td>\n",
       "      <td>1</td>\n",
       "      <td>...</td>\n",
       "      <td>001_022</td>\n",
       "      <td>GTTGCAGTTAGCTAACAGGCCATGCCCAGTCTAGATCGCATGCAGC...</td>\n",
       "      <td>ergosterol#guide_894#guide_894:REF2ALT:offset-...</td>\n",
       "      <td>chrVIII</td>\n",
       "      <td>CCTACACGACGCTCTTCCGATCTCCTCAGCCTCTATGCGTACG</td>\n",
       "      <td>TTCAGACGTGTGCTCTTCCGATCTCTAAACGAGCACACACGTGA</td>\n",
       "      <td>199.0</td>\n",
       "      <td>60.039978</td>\n",
       "      <td>58.516684</td>\n",
       "      <td>CCTCAGCCTCTATGCGTACGAGCCAGACCCTTGCTCCCAGTAGATA...</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>12 rows  50 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "        var_id chrom  SNP_chr_pos REF      ALT   guide_id guide_strand  \\\n",
       "881  ERG_00574  VIII       190196  CG    [CGG]  guide_883            +   \n",
       "882  ERG_00574  VIII       190196  CG    [CGG]  guide_884            +   \n",
       "883  ERG_00574  VIII       190196  CG    [CGG]  guide_885            +   \n",
       "884  ERG_00574  VIII       190196  CG    [CGG]  guide_886            -   \n",
       "885  ERG_00575  VIII       190196  CG      [C]  guide_887            +   \n",
       "886  ERG_00575  VIII       190196  CG      [C]  guide_888            +   \n",
       "887  ERG_00575  VIII       190196  CG      [C]  guide_889            +   \n",
       "888  ERG_00575  VIII       190196  CG      [C]  guide_890            -   \n",
       "889  ERG_00576  VIII       190196  CG  [CTGTG]  guide_891            +   \n",
       "890  ERG_00576  VIII       190196  CG  [CTGTG]  guide_892            +   \n",
       "891  ERG_00576  VIII       190196  CG  [CTGTG]  guide_893            +   \n",
       "892  ERG_00576  VIII       190196  CG  [CTGTG]  guide_894            -   \n",
       "\n",
       "     guide0_chr_pos  guide_cut_chr_pos  SNP_pos_in_guide  ...  barcode_id  \\\n",
       "881          190196             190193                -1  ...     014_012   \n",
       "882          190197             190194                -2  ...     014_013   \n",
       "883          190198             190195                -3  ...     014_014   \n",
       "884          190196             190200                 1  ...     014_015   \n",
       "885          190196             190193                -1  ...     007_036   \n",
       "886          190197             190194                -2  ...     007_037   \n",
       "887          190198             190195                -3  ...     007_038   \n",
       "888          190196             190200                 1  ...     007_039   \n",
       "889          190196             190193                -1  ...     001_019   \n",
       "890          190197             190194                -2  ...     001_020   \n",
       "891          190198             190195                -3  ...     001_021   \n",
       "892          190196             190200                 1  ...     001_022   \n",
       "\n",
       "                                             oligo_seq  \\\n",
       "881  GTTGCAGTTAGCTAACAGGCCATGCGGTCTTTCAACAGCATGCAGC...   \n",
       "882  GTTGCAGTTAGCTAACAGGCCATGCACTGGTCCTCTGGCATGCAGC...   \n",
       "883  GTTGCAGTTAGCTAACAGGCCATGCCGGTCATGAACTGCATGCAGC...   \n",
       "884  GTTGCAGTTAGCTAACAGGCCATGCGTAGACAGCTCTGCATGCAGC...   \n",
       "885  GTTGCAGTTAGCTAACAGGCCATGCCTACTCGACCCTGCATGCAGC...   \n",
       "886  GTTGCAGTTAGCTAACAGGCCATGCGCAACTTCCACTGCATGCAGC...   \n",
       "887  GTTGCAGTTAGCTAACAGGCCATGCGTCCCACTCCGTGCATGCAGC...   \n",
       "888  GTTGCAGTTAGCTAACAGGCCATGCGCTGTCGATGTGGCATGCAGC...   \n",
       "889  GTTGCAGTTAGCTAACAGGCCATGCGCTTAGTGTGCGGCATGCAGC...   \n",
       "890  GTTGCAGTTAGCTAACAGGCCATGCTTATTTGCGGGTGCATGCAGC...   \n",
       "891  GTTGCAGTTAGCTAACAGGCCATGCGGACTAGGATCAGCATGCAGC...   \n",
       "892  GTTGCAGTTAGCTAACAGGCCATGCCCAGTCTAGATCGCATGCAGC...   \n",
       "\n",
       "                                              oligo_id chromosome  \\\n",
       "881  ergosterol#guide_883#guide_883:REF2ALT:offset1...    chrVIII   \n",
       "882  ergosterol#guide_884#guide_884:REF2ALT:offset1...    chrVIII   \n",
       "883  ergosterol#guide_885#guide_885:REF2ALT:offset1...    chrVIII   \n",
       "884  ergosterol#guide_886#guide_886:REF2ALT:offset-...    chrVIII   \n",
       "885  ergosterol#guide_887#guide_887:REF2ALT:offset1...    chrVIII   \n",
       "886  ergosterol#guide_888#guide_888:REF2ALT:offset1...    chrVIII   \n",
       "887  ergosterol#guide_889#guide_889:REF2ALT:offset1...    chrVIII   \n",
       "888  ergosterol#guide_890#guide_890:REF2ALT:offset-...    chrVIII   \n",
       "889  ergosterol#guide_891#guide_891:REF2ALT:offset1...    chrVIII   \n",
       "890  ergosterol#guide_892#guide_892:REF2ALT:offset1...    chrVIII   \n",
       "891  ergosterol#guide_893#guide_893:REF2ALT:offset1...    chrVIII   \n",
       "892  ergosterol#guide_894#guide_894:REF2ALT:offset-...    chrVIII   \n",
       "\n",
       "                                        primer_F  \\\n",
       "881  CCTACACGACGCTCTTCCGATCTCCTCAGCCTCTATGCGTACG   \n",
       "882  CCTACACGACGCTCTTCCGATCTCCTCAGCCTCTATGCGTACG   \n",
       "883  CCTACACGACGCTCTTCCGATCTCCTCAGCCTCTATGCGTACG   \n",
       "884  CCTACACGACGCTCTTCCGATCTCCTCAGCCTCTATGCGTACG   \n",
       "885  CCTACACGACGCTCTTCCGATCTCCTCAGCCTCTATGCGTACG   \n",
       "886  CCTACACGACGCTCTTCCGATCTCCTCAGCCTCTATGCGTACG   \n",
       "887  CCTACACGACGCTCTTCCGATCTCCTCAGCCTCTATGCGTACG   \n",
       "888  CCTACACGACGCTCTTCCGATCTCCTCAGCCTCTATGCGTACG   \n",
       "889  CCTACACGACGCTCTTCCGATCTCCTCAGCCTCTATGCGTACG   \n",
       "890  CCTACACGACGCTCTTCCGATCTCCTCAGCCTCTATGCGTACG   \n",
       "891  CCTACACGACGCTCTTCCGATCTCCTCAGCCTCTATGCGTACG   \n",
       "892  CCTACACGACGCTCTTCCGATCTCCTCAGCCTCTATGCGTACG   \n",
       "\n",
       "                                         primer_R  template_len    left_tm  \\\n",
       "881  TTCAGACGTGTGCTCTTCCGATCTCTAAACGAGCACACACGTGA         199.0  60.039978   \n",
       "882  TTCAGACGTGTGCTCTTCCGATCTCTAAACGAGCACACACGTGA         199.0  60.039978   \n",
       "883  TTCAGACGTGTGCTCTTCCGATCTCTAAACGAGCACACACGTGA         199.0  60.039978   \n",
       "884  TTCAGACGTGTGCTCTTCCGATCTCTAAACGAGCACACACGTGA         199.0  60.039978   \n",
       "885  TTCAGACGTGTGCTCTTCCGATCTCTAAACGAGCACACACGTGA         199.0  60.039978   \n",
       "886  TTCAGACGTGTGCTCTTCCGATCTCTAAACGAGCACACACGTGA         199.0  60.039978   \n",
       "887  TTCAGACGTGTGCTCTTCCGATCTCTAAACGAGCACACACGTGA         199.0  60.039978   \n",
       "888  TTCAGACGTGTGCTCTTCCGATCTCTAAACGAGCACACACGTGA         199.0  60.039978   \n",
       "889  TTCAGACGTGTGCTCTTCCGATCTCTAAACGAGCACACACGTGA         199.0  60.039978   \n",
       "890  TTCAGACGTGTGCTCTTCCGATCTCTAAACGAGCACACACGTGA         199.0  60.039978   \n",
       "891  TTCAGACGTGTGCTCTTCCGATCTCTAAACGAGCACACACGTGA         199.0  60.039978   \n",
       "892  TTCAGACGTGTGCTCTTCCGATCTCTAAACGAGCACACACGTGA         199.0  60.039978   \n",
       "\n",
       "      right_tm                                        product_seq  \n",
       "881  58.516684  CCTCAGCCTCTATGCGTACGAGCCAGACCCTTGCTCCCAGTAGATA...  \n",
       "882  58.516684  CCTCAGCCTCTATGCGTACGAGCCAGACCCTTGCTCCCAGTAGATA...  \n",
       "883  58.516684  CCTCAGCCTCTATGCGTACGAGCCAGACCCTTGCTCCCAGTAGATA...  \n",
       "884  58.516684  CCTCAGCCTCTATGCGTACGAGCCAGACCCTTGCTCCCAGTAGATA...  \n",
       "885  58.516684  CCTCAGCCTCTATGCGTACGAGCCAGACCCTTGCTCCCAGTAGATA...  \n",
       "886  58.516684  CCTCAGCCTCTATGCGTACGAGCCAGACCCTTGCTCCCAGTAGATA...  \n",
       "887  58.516684  CCTCAGCCTCTATGCGTACGAGCCAGACCCTTGCTCCCAGTAGATA...  \n",
       "888  58.516684  CCTCAGCCTCTATGCGTACGAGCCAGACCCTTGCTCCCAGTAGATA...  \n",
       "889  58.516684  CCTCAGCCTCTATGCGTACGAGCCAGACCCTTGCTCCCAGTAGATA...  \n",
       "890  58.516684  CCTCAGCCTCTATGCGTACGAGCCAGACCCTTGCTCCCAGTAGATA...  \n",
       "891  58.516684  CCTCAGCCTCTATGCGTACGAGCCAGACCCTTGCTCCCAGTAGATA...  \n",
       "892  58.516684  CCTCAGCCTCTATGCGTACGAGCCAGACCCTTGCTCCCAGTAGATA...  \n",
       "\n",
       "[12 rows x 50 columns]"
      ]
     },
     "execution_count": 26,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "oli_info[oli_info['var_id'].isin(['ERG_00574','ERG_00575','ERG_00576'])]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 27,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Unnamed: 0_x</th>\n",
       "      <th>var_id</th>\n",
       "      <th>chrom</th>\n",
       "      <th>SNP_chr_pos</th>\n",
       "      <th>Gene</th>\n",
       "      <th>Consequence</th>\n",
       "      <th>ALT</th>\n",
       "      <th>CDS_position</th>\n",
       "      <th>Protein_position</th>\n",
       "      <th>Amino_acids</th>\n",
       "      <th>DISTANCE</th>\n",
       "      <th>Unnamed: 0_y</th>\n",
       "      <th>condition</th>\n",
       "      <th>coef</th>\n",
       "      <th>std err</th>\n",
       "      <th>t</th>\n",
       "      <th>pval</th>\n",
       "      <th>Conf. Int. Low</th>\n",
       "      <th>Conf. Int. Upp.</th>\n",
       "      <th>padj</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>5271</th>\n",
       "      <td>32532</td>\n",
       "      <td>ERG_01689</td>\n",
       "      <td>XIII</td>\n",
       "      <td>20611</td>\n",
       "      <td>ERG13</td>\n",
       "      <td>upstream_gene_variant</td>\n",
       "      <td>A</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>76</td>\n",
       "      <td>6435</td>\n",
       "      <td>CAFF</td>\n",
       "      <td>0.001652</td>\n",
       "      <td>0.000991</td>\n",
       "      <td>1.666496</td>\n",
       "      <td>9.561785e-02</td>\n",
       "      <td>-0.000291</td>\n",
       "      <td>0.003594</td>\n",
       "      <td>4.168827e-01</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5272</th>\n",
       "      <td>32532</td>\n",
       "      <td>ERG_01689</td>\n",
       "      <td>XIII</td>\n",
       "      <td>20611</td>\n",
       "      <td>ERG13</td>\n",
       "      <td>upstream_gene_variant</td>\n",
       "      <td>A</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>76</td>\n",
       "      <td>6436</td>\n",
       "      <td>COCL</td>\n",
       "      <td>0.005671</td>\n",
       "      <td>0.003225</td>\n",
       "      <td>1.758513</td>\n",
       "      <td>7.866345e-02</td>\n",
       "      <td>-0.000650</td>\n",
       "      <td>0.011991</td>\n",
       "      <td>3.764189e-01</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5273</th>\n",
       "      <td>32532</td>\n",
       "      <td>ERG_01689</td>\n",
       "      <td>XIII</td>\n",
       "      <td>20611</td>\n",
       "      <td>ERG13</td>\n",
       "      <td>upstream_gene_variant</td>\n",
       "      <td>A</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>76</td>\n",
       "      <td>6437</td>\n",
       "      <td>LOV</td>\n",
       "      <td>0.010003</td>\n",
       "      <td>0.001267</td>\n",
       "      <td>7.895830</td>\n",
       "      <td>2.913668e-15</td>\n",
       "      <td>0.007520</td>\n",
       "      <td>0.012486</td>\n",
       "      <td>1.245524e-13</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5274</th>\n",
       "      <td>32532</td>\n",
       "      <td>ERG_01689</td>\n",
       "      <td>XIII</td>\n",
       "      <td>20611</td>\n",
       "      <td>ERG13</td>\n",
       "      <td>upstream_gene_variant</td>\n",
       "      <td>A</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>76</td>\n",
       "      <td>6438</td>\n",
       "      <td>NACL</td>\n",
       "      <td>0.001951</td>\n",
       "      <td>0.001016</td>\n",
       "      <td>1.920722</td>\n",
       "      <td>5.476964e-02</td>\n",
       "      <td>-0.000040</td>\n",
       "      <td>0.003941</td>\n",
       "      <td>3.143651e-01</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5275</th>\n",
       "      <td>32532</td>\n",
       "      <td>ERG_01689</td>\n",
       "      <td>XIII</td>\n",
       "      <td>20611</td>\n",
       "      <td>ERG13</td>\n",
       "      <td>upstream_gene_variant</td>\n",
       "      <td>A</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>76</td>\n",
       "      <td>6439</td>\n",
       "      <td>SC</td>\n",
       "      <td>0.000866</td>\n",
       "      <td>0.000949</td>\n",
       "      <td>0.911702</td>\n",
       "      <td>3.619278e-01</td>\n",
       "      <td>-0.000995</td>\n",
       "      <td>0.002726</td>\n",
       "      <td>7.344336e-01</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5276</th>\n",
       "      <td>32532</td>\n",
       "      <td>ERG_01689</td>\n",
       "      <td>XIII</td>\n",
       "      <td>20611</td>\n",
       "      <td>ERG13</td>\n",
       "      <td>upstream_gene_variant</td>\n",
       "      <td>A</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>76</td>\n",
       "      <td>6440</td>\n",
       "      <td>TBF</td>\n",
       "      <td>-0.000593</td>\n",
       "      <td>0.000718</td>\n",
       "      <td>-0.825954</td>\n",
       "      <td>4.088323e-01</td>\n",
       "      <td>-0.001999</td>\n",
       "      <td>0.000814</td>\n",
       "      <td>7.681620e-01</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5277</th>\n",
       "      <td>32538</td>\n",
       "      <td>ERG_01690</td>\n",
       "      <td>XIII</td>\n",
       "      <td>20611</td>\n",
       "      <td>ERG13</td>\n",
       "      <td>upstream_gene_variant</td>\n",
       "      <td>C</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>76</td>\n",
       "      <td>6441</td>\n",
       "      <td>CAFF</td>\n",
       "      <td>-0.002529</td>\n",
       "      <td>0.001142</td>\n",
       "      <td>-2.213578</td>\n",
       "      <td>2.686010e-02</td>\n",
       "      <td>-0.004767</td>\n",
       "      <td>-0.000290</td>\n",
       "      <td>2.156736e-01</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5278</th>\n",
       "      <td>32538</td>\n",
       "      <td>ERG_01690</td>\n",
       "      <td>XIII</td>\n",
       "      <td>20611</td>\n",
       "      <td>ERG13</td>\n",
       "      <td>upstream_gene_variant</td>\n",
       "      <td>C</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>76</td>\n",
       "      <td>6442</td>\n",
       "      <td>COCL</td>\n",
       "      <td>0.001580</td>\n",
       "      <td>0.003277</td>\n",
       "      <td>0.482114</td>\n",
       "      <td>6.297261e-01</td>\n",
       "      <td>-0.004844</td>\n",
       "      <td>0.008004</td>\n",
       "      <td>8.842543e-01</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5279</th>\n",
       "      <td>32538</td>\n",
       "      <td>ERG_01690</td>\n",
       "      <td>XIII</td>\n",
       "      <td>20611</td>\n",
       "      <td>ERG13</td>\n",
       "      <td>upstream_gene_variant</td>\n",
       "      <td>C</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>76</td>\n",
       "      <td>6443</td>\n",
       "      <td>LOV</td>\n",
       "      <td>0.007932</td>\n",
       "      <td>0.001272</td>\n",
       "      <td>6.235127</td>\n",
       "      <td>4.532523e-10</td>\n",
       "      <td>0.005439</td>\n",
       "      <td>0.010425</td>\n",
       "      <td>1.674422e-08</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5280</th>\n",
       "      <td>32538</td>\n",
       "      <td>ERG_01690</td>\n",
       "      <td>XIII</td>\n",
       "      <td>20611</td>\n",
       "      <td>ERG13</td>\n",
       "      <td>upstream_gene_variant</td>\n",
       "      <td>C</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>76</td>\n",
       "      <td>6444</td>\n",
       "      <td>NACL</td>\n",
       "      <td>-0.000104</td>\n",
       "      <td>0.001096</td>\n",
       "      <td>-0.094769</td>\n",
       "      <td>9.244982e-01</td>\n",
       "      <td>-0.002253</td>\n",
       "      <td>0.002045</td>\n",
       "      <td>9.871773e-01</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5281</th>\n",
       "      <td>32538</td>\n",
       "      <td>ERG_01690</td>\n",
       "      <td>XIII</td>\n",
       "      <td>20611</td>\n",
       "      <td>ERG13</td>\n",
       "      <td>upstream_gene_variant</td>\n",
       "      <td>C</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>76</td>\n",
       "      <td>6445</td>\n",
       "      <td>SC</td>\n",
       "      <td>0.000408</td>\n",
       "      <td>0.001055</td>\n",
       "      <td>0.386792</td>\n",
       "      <td>6.989110e-01</td>\n",
       "      <td>-0.001660</td>\n",
       "      <td>0.002476</td>\n",
       "      <td>9.103910e-01</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5282</th>\n",
       "      <td>32538</td>\n",
       "      <td>ERG_01690</td>\n",
       "      <td>XIII</td>\n",
       "      <td>20611</td>\n",
       "      <td>ERG13</td>\n",
       "      <td>upstream_gene_variant</td>\n",
       "      <td>C</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>76</td>\n",
       "      <td>6446</td>\n",
       "      <td>TBF</td>\n",
       "      <td>0.001207</td>\n",
       "      <td>0.000820</td>\n",
       "      <td>1.472372</td>\n",
       "      <td>1.409238e-01</td>\n",
       "      <td>-0.000400</td>\n",
       "      <td>0.002813</td>\n",
       "      <td>5.008206e-01</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "      Unnamed: 0_x     var_id chrom  SNP_chr_pos   Gene  \\\n",
       "5271         32532  ERG_01689  XIII        20611  ERG13   \n",
       "5272         32532  ERG_01689  XIII        20611  ERG13   \n",
       "5273         32532  ERG_01689  XIII        20611  ERG13   \n",
       "5274         32532  ERG_01689  XIII        20611  ERG13   \n",
       "5275         32532  ERG_01689  XIII        20611  ERG13   \n",
       "5276         32532  ERG_01689  XIII        20611  ERG13   \n",
       "5277         32538  ERG_01690  XIII        20611  ERG13   \n",
       "5278         32538  ERG_01690  XIII        20611  ERG13   \n",
       "5279         32538  ERG_01690  XIII        20611  ERG13   \n",
       "5280         32538  ERG_01690  XIII        20611  ERG13   \n",
       "5281         32538  ERG_01690  XIII        20611  ERG13   \n",
       "5282         32538  ERG_01690  XIII        20611  ERG13   \n",
       "\n",
       "                Consequence ALT CDS_position Protein_position Amino_acids  \\\n",
       "5271  upstream_gene_variant   A            -                -           -   \n",
       "5272  upstream_gene_variant   A            -                -           -   \n",
       "5273  upstream_gene_variant   A            -                -           -   \n",
       "5274  upstream_gene_variant   A            -                -           -   \n",
       "5275  upstream_gene_variant   A            -                -           -   \n",
       "5276  upstream_gene_variant   A            -                -           -   \n",
       "5277  upstream_gene_variant   C            -                -           -   \n",
       "5278  upstream_gene_variant   C            -                -           -   \n",
       "5279  upstream_gene_variant   C            -                -           -   \n",
       "5280  upstream_gene_variant   C            -                -           -   \n",
       "5281  upstream_gene_variant   C            -                -           -   \n",
       "5282  upstream_gene_variant   C            -                -           -   \n",
       "\n",
       "      DISTANCE  Unnamed: 0_y condition      coef   std err         t  \\\n",
       "5271        76          6435      CAFF  0.001652  0.000991  1.666496   \n",
       "5272        76          6436      COCL  0.005671  0.003225  1.758513   \n",
       "5273        76          6437       LOV  0.010003  0.001267  7.895830   \n",
       "5274        76          6438      NACL  0.001951  0.001016  1.920722   \n",
       "5275        76          6439        SC  0.000866  0.000949  0.911702   \n",
       "5276        76          6440       TBF -0.000593  0.000718 -0.825954   \n",
       "5277        76          6441      CAFF -0.002529  0.001142 -2.213578   \n",
       "5278        76          6442      COCL  0.001580  0.003277  0.482114   \n",
       "5279        76          6443       LOV  0.007932  0.001272  6.235127   \n",
       "5280        76          6444      NACL -0.000104  0.001096 -0.094769   \n",
       "5281        76          6445        SC  0.000408  0.001055  0.386792   \n",
       "5282        76          6446       TBF  0.001207  0.000820  1.472372   \n",
       "\n",
       "              pval  Conf. Int. Low  Conf. Int. Upp.          padj  \n",
       "5271  9.561785e-02       -0.000291         0.003594  4.168827e-01  \n",
       "5272  7.866345e-02       -0.000650         0.011991  3.764189e-01  \n",
       "5273  2.913668e-15        0.007520         0.012486  1.245524e-13  \n",
       "5274  5.476964e-02       -0.000040         0.003941  3.143651e-01  \n",
       "5275  3.619278e-01       -0.000995         0.002726  7.344336e-01  \n",
       "5276  4.088323e-01       -0.001999         0.000814  7.681620e-01  \n",
       "5277  2.686010e-02       -0.004767        -0.000290  2.156736e-01  \n",
       "5278  6.297261e-01       -0.004844         0.008004  8.842543e-01  \n",
       "5279  4.532523e-10        0.005439         0.010425  1.674422e-08  \n",
       "5280  9.244982e-01       -0.002253         0.002045  9.871773e-01  \n",
       "5281  6.989110e-01       -0.001660         0.002476  9.103910e-01  \n",
       "5282  1.409238e-01       -0.000400         0.002813  5.008206e-01  "
      ]
     },
     "execution_count": 27,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "#ERG13 site with two upstream variants, both with significant positive effects in lovastatin\n",
    "\n",
    "chrom = 'XIII'\n",
    "pos =20611\n",
    "\n",
    "annotated_fits[(annotated_fits['chrom']==chrom)&(annotated_fits['SNP_chr_pos']==pos)]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 28,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Unnamed: 0_x</th>\n",
       "      <th>var_id</th>\n",
       "      <th>chrom</th>\n",
       "      <th>SNP_chr_pos</th>\n",
       "      <th>Gene</th>\n",
       "      <th>Consequence</th>\n",
       "      <th>ALT</th>\n",
       "      <th>CDS_position</th>\n",
       "      <th>Protein_position</th>\n",
       "      <th>Amino_acids</th>\n",
       "      <th>DISTANCE</th>\n",
       "      <th>Unnamed: 0_y</th>\n",
       "      <th>condition</th>\n",
       "      <th>coef</th>\n",
       "      <th>std err</th>\n",
       "      <th>t</th>\n",
       "      <th>pval</th>\n",
       "      <th>Conf. Int. Low</th>\n",
       "      <th>Conf. Int. Upp.</th>\n",
       "      <th>padj</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>5750</th>\n",
       "      <td>32929</td>\n",
       "      <td>ERG_01887</td>\n",
       "      <td>XIII</td>\n",
       "      <td>118967</td>\n",
       "      <td>HMG1</td>\n",
       "      <td>upstream_gene_variant</td>\n",
       "      <td>A</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>69</td>\n",
       "      <td>6855</td>\n",
       "      <td>CAFF</td>\n",
       "      <td>-0.005981</td>\n",
       "      <td>0.002736</td>\n",
       "      <td>-2.185860</td>\n",
       "      <td>2.882820e-02</td>\n",
       "      <td>-0.011344</td>\n",
       "      <td>-0.000618</td>\n",
       "      <td>2.255488e-01</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5751</th>\n",
       "      <td>32929</td>\n",
       "      <td>ERG_01887</td>\n",
       "      <td>XIII</td>\n",
       "      <td>118967</td>\n",
       "      <td>HMG1</td>\n",
       "      <td>upstream_gene_variant</td>\n",
       "      <td>A</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>69</td>\n",
       "      <td>6856</td>\n",
       "      <td>COCL</td>\n",
       "      <td>0.005082</td>\n",
       "      <td>0.008237</td>\n",
       "      <td>0.616983</td>\n",
       "      <td>5.372474e-01</td>\n",
       "      <td>-0.011062</td>\n",
       "      <td>0.021226</td>\n",
       "      <td>8.422578e-01</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5752</th>\n",
       "      <td>32929</td>\n",
       "      <td>ERG_01887</td>\n",
       "      <td>XIII</td>\n",
       "      <td>118967</td>\n",
       "      <td>HMG1</td>\n",
       "      <td>upstream_gene_variant</td>\n",
       "      <td>A</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>69</td>\n",
       "      <td>6857</td>\n",
       "      <td>LOV</td>\n",
       "      <td>0.020362</td>\n",
       "      <td>0.003185</td>\n",
       "      <td>6.393992</td>\n",
       "      <td>1.623372e-10</td>\n",
       "      <td>0.014120</td>\n",
       "      <td>0.026604</td>\n",
       "      <td>6.097495e-09</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5753</th>\n",
       "      <td>32929</td>\n",
       "      <td>ERG_01887</td>\n",
       "      <td>XIII</td>\n",
       "      <td>118967</td>\n",
       "      <td>HMG1</td>\n",
       "      <td>upstream_gene_variant</td>\n",
       "      <td>A</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>69</td>\n",
       "      <td>6858</td>\n",
       "      <td>NACL</td>\n",
       "      <td>-0.002002</td>\n",
       "      <td>0.002962</td>\n",
       "      <td>-0.675804</td>\n",
       "      <td>4.991670e-01</td>\n",
       "      <td>-0.007807</td>\n",
       "      <td>0.003803</td>\n",
       "      <td>8.243139e-01</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5754</th>\n",
       "      <td>32929</td>\n",
       "      <td>ERG_01887</td>\n",
       "      <td>XIII</td>\n",
       "      <td>118967</td>\n",
       "      <td>HMG1</td>\n",
       "      <td>upstream_gene_variant</td>\n",
       "      <td>A</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>69</td>\n",
       "      <td>6859</td>\n",
       "      <td>SC</td>\n",
       "      <td>0.001541</td>\n",
       "      <td>0.002832</td>\n",
       "      <td>0.544084</td>\n",
       "      <td>5.863846e-01</td>\n",
       "      <td>-0.004010</td>\n",
       "      <td>0.007091</td>\n",
       "      <td>8.642217e-01</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5755</th>\n",
       "      <td>32929</td>\n",
       "      <td>ERG_01887</td>\n",
       "      <td>XIII</td>\n",
       "      <td>118967</td>\n",
       "      <td>HMG1</td>\n",
       "      <td>upstream_gene_variant</td>\n",
       "      <td>A</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>69</td>\n",
       "      <td>6860</td>\n",
       "      <td>TBF</td>\n",
       "      <td>0.001403</td>\n",
       "      <td>0.002502</td>\n",
       "      <td>0.560615</td>\n",
       "      <td>5.750612e-01</td>\n",
       "      <td>-0.003502</td>\n",
       "      <td>0.006307</td>\n",
       "      <td>8.593287e-01</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5756</th>\n",
       "      <td>32933</td>\n",
       "      <td>ERG_01888</td>\n",
       "      <td>XIII</td>\n",
       "      <td>118967</td>\n",
       "      <td>HMG1</td>\n",
       "      <td>upstream_gene_variant</td>\n",
       "      <td>T</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>69</td>\n",
       "      <td>6861</td>\n",
       "      <td>CAFF</td>\n",
       "      <td>0.010732</td>\n",
       "      <td>0.004132</td>\n",
       "      <td>2.597430</td>\n",
       "      <td>9.393843e-03</td>\n",
       "      <td>0.002634</td>\n",
       "      <td>0.018830</td>\n",
       "      <td>1.085309e-01</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5757</th>\n",
       "      <td>32933</td>\n",
       "      <td>ERG_01888</td>\n",
       "      <td>XIII</td>\n",
       "      <td>118967</td>\n",
       "      <td>HMG1</td>\n",
       "      <td>upstream_gene_variant</td>\n",
       "      <td>T</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>69</td>\n",
       "      <td>6862</td>\n",
       "      <td>COCL</td>\n",
       "      <td>-0.003330</td>\n",
       "      <td>0.010477</td>\n",
       "      <td>-0.317878</td>\n",
       "      <td>7.505783e-01</td>\n",
       "      <td>-0.023866</td>\n",
       "      <td>0.017205</td>\n",
       "      <td>9.300491e-01</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5758</th>\n",
       "      <td>32933</td>\n",
       "      <td>ERG_01888</td>\n",
       "      <td>XIII</td>\n",
       "      <td>118967</td>\n",
       "      <td>HMG1</td>\n",
       "      <td>upstream_gene_variant</td>\n",
       "      <td>T</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>69</td>\n",
       "      <td>6863</td>\n",
       "      <td>LOV</td>\n",
       "      <td>0.001800</td>\n",
       "      <td>0.004499</td>\n",
       "      <td>0.399974</td>\n",
       "      <td>6.891765e-01</td>\n",
       "      <td>-0.007019</td>\n",
       "      <td>0.010618</td>\n",
       "      <td>9.062533e-01</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5759</th>\n",
       "      <td>32933</td>\n",
       "      <td>ERG_01888</td>\n",
       "      <td>XIII</td>\n",
       "      <td>118967</td>\n",
       "      <td>HMG1</td>\n",
       "      <td>upstream_gene_variant</td>\n",
       "      <td>T</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>69</td>\n",
       "      <td>6864</td>\n",
       "      <td>NACL</td>\n",
       "      <td>-0.005376</td>\n",
       "      <td>0.004359</td>\n",
       "      <td>-1.233367</td>\n",
       "      <td>2.174420e-01</td>\n",
       "      <td>-0.013919</td>\n",
       "      <td>0.003167</td>\n",
       "      <td>5.993174e-01</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5760</th>\n",
       "      <td>32933</td>\n",
       "      <td>ERG_01888</td>\n",
       "      <td>XIII</td>\n",
       "      <td>118967</td>\n",
       "      <td>HMG1</td>\n",
       "      <td>upstream_gene_variant</td>\n",
       "      <td>T</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>69</td>\n",
       "      <td>6865</td>\n",
       "      <td>SC</td>\n",
       "      <td>-0.001694</td>\n",
       "      <td>0.004766</td>\n",
       "      <td>-0.355510</td>\n",
       "      <td>7.222080e-01</td>\n",
       "      <td>-0.011036</td>\n",
       "      <td>0.007647</td>\n",
       "      <td>9.199636e-01</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5761</th>\n",
       "      <td>32933</td>\n",
       "      <td>ERG_01888</td>\n",
       "      <td>XIII</td>\n",
       "      <td>118967</td>\n",
       "      <td>HMG1</td>\n",
       "      <td>upstream_gene_variant</td>\n",
       "      <td>T</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>69</td>\n",
       "      <td>6866</td>\n",
       "      <td>TBF</td>\n",
       "      <td>-0.003433</td>\n",
       "      <td>0.003873</td>\n",
       "      <td>-0.886360</td>\n",
       "      <td>3.754259e-01</td>\n",
       "      <td>-0.011025</td>\n",
       "      <td>0.004159</td>\n",
       "      <td>7.462975e-01</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "      Unnamed: 0_x     var_id chrom  SNP_chr_pos  Gene            Consequence  \\\n",
       "5750         32929  ERG_01887  XIII       118967  HMG1  upstream_gene_variant   \n",
       "5751         32929  ERG_01887  XIII       118967  HMG1  upstream_gene_variant   \n",
       "5752         32929  ERG_01887  XIII       118967  HMG1  upstream_gene_variant   \n",
       "5753         32929  ERG_01887  XIII       118967  HMG1  upstream_gene_variant   \n",
       "5754         32929  ERG_01887  XIII       118967  HMG1  upstream_gene_variant   \n",
       "5755         32929  ERG_01887  XIII       118967  HMG1  upstream_gene_variant   \n",
       "5756         32933  ERG_01888  XIII       118967  HMG1  upstream_gene_variant   \n",
       "5757         32933  ERG_01888  XIII       118967  HMG1  upstream_gene_variant   \n",
       "5758         32933  ERG_01888  XIII       118967  HMG1  upstream_gene_variant   \n",
       "5759         32933  ERG_01888  XIII       118967  HMG1  upstream_gene_variant   \n",
       "5760         32933  ERG_01888  XIII       118967  HMG1  upstream_gene_variant   \n",
       "5761         32933  ERG_01888  XIII       118967  HMG1  upstream_gene_variant   \n",
       "\n",
       "     ALT CDS_position Protein_position Amino_acids  DISTANCE  Unnamed: 0_y  \\\n",
       "5750   A            -                -           -        69          6855   \n",
       "5751   A            -                -           -        69          6856   \n",
       "5752   A            -                -           -        69          6857   \n",
       "5753   A            -                -           -        69          6858   \n",
       "5754   A            -                -           -        69          6859   \n",
       "5755   A            -                -           -        69          6860   \n",
       "5756   T            -                -           -        69          6861   \n",
       "5757   T            -                -           -        69          6862   \n",
       "5758   T            -                -           -        69          6863   \n",
       "5759   T            -                -           -        69          6864   \n",
       "5760   T            -                -           -        69          6865   \n",
       "5761   T            -                -           -        69          6866   \n",
       "\n",
       "     condition      coef   std err         t          pval  Conf. Int. Low  \\\n",
       "5750      CAFF -0.005981  0.002736 -2.185860  2.882820e-02       -0.011344   \n",
       "5751      COCL  0.005082  0.008237  0.616983  5.372474e-01       -0.011062   \n",
       "5752       LOV  0.020362  0.003185  6.393992  1.623372e-10        0.014120   \n",
       "5753      NACL -0.002002  0.002962 -0.675804  4.991670e-01       -0.007807   \n",
       "5754        SC  0.001541  0.002832  0.544084  5.863846e-01       -0.004010   \n",
       "5755       TBF  0.001403  0.002502  0.560615  5.750612e-01       -0.003502   \n",
       "5756      CAFF  0.010732  0.004132  2.597430  9.393843e-03        0.002634   \n",
       "5757      COCL -0.003330  0.010477 -0.317878  7.505783e-01       -0.023866   \n",
       "5758       LOV  0.001800  0.004499  0.399974  6.891765e-01       -0.007019   \n",
       "5759      NACL -0.005376  0.004359 -1.233367  2.174420e-01       -0.013919   \n",
       "5760        SC -0.001694  0.004766 -0.355510  7.222080e-01       -0.011036   \n",
       "5761       TBF -0.003433  0.003873 -0.886360  3.754259e-01       -0.011025   \n",
       "\n",
       "      Conf. Int. Upp.          padj  \n",
       "5750        -0.000618  2.255488e-01  \n",
       "5751         0.021226  8.422578e-01  \n",
       "5752         0.026604  6.097495e-09  \n",
       "5753         0.003803  8.243139e-01  \n",
       "5754         0.007091  8.642217e-01  \n",
       "5755         0.006307  8.593287e-01  \n",
       "5756         0.018830  1.085309e-01  \n",
       "5757         0.017205  9.300491e-01  \n",
       "5758         0.010618  9.062533e-01  \n",
       "5759         0.003167  5.993174e-01  \n",
       "5760         0.007647  9.199636e-01  \n",
       "5761         0.004159  7.462975e-01  "
      ]
     },
     "execution_count": 28,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "#HMG1 site with two upstream variants, one with a significant positive effect in lovastatin\n",
    "#the other with no significant effects\n",
    "\n",
    "chrom = 'XIII'\n",
    "pos =118967\n",
    "\n",
    "annotated_fits[(annotated_fits['chrom']==chrom)&(annotated_fits['SNP_chr_pos']==pos)]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 29,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Unnamed: 0_x</th>\n",
       "      <th>var_id</th>\n",
       "      <th>chrom</th>\n",
       "      <th>SNP_chr_pos</th>\n",
       "      <th>Gene</th>\n",
       "      <th>Consequence</th>\n",
       "      <th>ALT</th>\n",
       "      <th>CDS_position</th>\n",
       "      <th>Protein_position</th>\n",
       "      <th>Amino_acids</th>\n",
       "      <th>DISTANCE</th>\n",
       "      <th>Unnamed: 0_y</th>\n",
       "      <th>condition</th>\n",
       "      <th>coef</th>\n",
       "      <th>std err</th>\n",
       "      <th>t</th>\n",
       "      <th>pval</th>\n",
       "      <th>Conf. Int. Low</th>\n",
       "      <th>Conf. Int. Upp.</th>\n",
       "      <th>padj</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>8071</th>\n",
       "      <td>51186</td>\n",
       "      <td>ERG_02672</td>\n",
       "      <td>XVI</td>\n",
       "      <td>497884</td>\n",
       "      <td>ERG10</td>\n",
       "      <td>upstream_gene_variant</td>\n",
       "      <td>A</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>212</td>\n",
       "      <td>8207</td>\n",
       "      <td>CAFF</td>\n",
       "      <td>0.004485</td>\n",
       "      <td>0.000500</td>\n",
       "      <td>8.968342</td>\n",
       "      <td>3.061659e-19</td>\n",
       "      <td>0.003505</td>\n",
       "      <td>0.005465</td>\n",
       "      <td>1.395153e-17</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>8072</th>\n",
       "      <td>51186</td>\n",
       "      <td>ERG_02672</td>\n",
       "      <td>XVI</td>\n",
       "      <td>497884</td>\n",
       "      <td>ERG10</td>\n",
       "      <td>upstream_gene_variant</td>\n",
       "      <td>A</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>212</td>\n",
       "      <td>8208</td>\n",
       "      <td>COCL</td>\n",
       "      <td>0.000469</td>\n",
       "      <td>0.001735</td>\n",
       "      <td>0.270599</td>\n",
       "      <td>7.866999e-01</td>\n",
       "      <td>-0.002930</td>\n",
       "      <td>0.003869</td>\n",
       "      <td>9.477797e-01</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>8073</th>\n",
       "      <td>51186</td>\n",
       "      <td>ERG_02672</td>\n",
       "      <td>XVI</td>\n",
       "      <td>497884</td>\n",
       "      <td>ERG10</td>\n",
       "      <td>upstream_gene_variant</td>\n",
       "      <td>A</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>212</td>\n",
       "      <td>8209</td>\n",
       "      <td>LOV</td>\n",
       "      <td>-0.003719</td>\n",
       "      <td>0.000683</td>\n",
       "      <td>-5.442711</td>\n",
       "      <td>5.260164e-08</td>\n",
       "      <td>-0.005058</td>\n",
       "      <td>-0.002380</td>\n",
       "      <td>1.742454e-06</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>8074</th>\n",
       "      <td>51186</td>\n",
       "      <td>ERG_02672</td>\n",
       "      <td>XVI</td>\n",
       "      <td>497884</td>\n",
       "      <td>ERG10</td>\n",
       "      <td>upstream_gene_variant</td>\n",
       "      <td>A</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>212</td>\n",
       "      <td>8210</td>\n",
       "      <td>NACL</td>\n",
       "      <td>-0.002156</td>\n",
       "      <td>0.000471</td>\n",
       "      <td>-4.577842</td>\n",
       "      <td>4.703769e-06</td>\n",
       "      <td>-0.003080</td>\n",
       "      <td>-0.001233</td>\n",
       "      <td>1.323691e-04</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>8075</th>\n",
       "      <td>51186</td>\n",
       "      <td>ERG_02672</td>\n",
       "      <td>XVI</td>\n",
       "      <td>497884</td>\n",
       "      <td>ERG10</td>\n",
       "      <td>upstream_gene_variant</td>\n",
       "      <td>A</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>212</td>\n",
       "      <td>8211</td>\n",
       "      <td>SC</td>\n",
       "      <td>-0.000804</td>\n",
       "      <td>0.000478</td>\n",
       "      <td>-1.680440</td>\n",
       "      <td>9.287491e-02</td>\n",
       "      <td>-0.001741</td>\n",
       "      <td>0.000134</td>\n",
       "      <td>4.096807e-01</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>8076</th>\n",
       "      <td>51186</td>\n",
       "      <td>ERG_02672</td>\n",
       "      <td>XVI</td>\n",
       "      <td>497884</td>\n",
       "      <td>ERG10</td>\n",
       "      <td>upstream_gene_variant</td>\n",
       "      <td>A</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>212</td>\n",
       "      <td>8212</td>\n",
       "      <td>TBF</td>\n",
       "      <td>-0.000280</td>\n",
       "      <td>0.000360</td>\n",
       "      <td>-0.779146</td>\n",
       "      <td>4.358954e-01</td>\n",
       "      <td>-0.000986</td>\n",
       "      <td>0.000425</td>\n",
       "      <td>7.864612e-01</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>8077</th>\n",
       "      <td>51193</td>\n",
       "      <td>ERG_02673</td>\n",
       "      <td>XVI</td>\n",
       "      <td>497884</td>\n",
       "      <td>ERG10</td>\n",
       "      <td>upstream_gene_variant</td>\n",
       "      <td>C</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>212</td>\n",
       "      <td>8213</td>\n",
       "      <td>CAFF</td>\n",
       "      <td>-0.001096</td>\n",
       "      <td>0.001223</td>\n",
       "      <td>-0.896001</td>\n",
       "      <td>3.702543e-01</td>\n",
       "      <td>-0.003492</td>\n",
       "      <td>0.001301</td>\n",
       "      <td>7.423242e-01</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>8078</th>\n",
       "      <td>51193</td>\n",
       "      <td>ERG_02673</td>\n",
       "      <td>XVI</td>\n",
       "      <td>497884</td>\n",
       "      <td>ERG10</td>\n",
       "      <td>upstream_gene_variant</td>\n",
       "      <td>C</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>212</td>\n",
       "      <td>8214</td>\n",
       "      <td>COCL</td>\n",
       "      <td>0.004001</td>\n",
       "      <td>0.003943</td>\n",
       "      <td>1.014748</td>\n",
       "      <td>3.102285e-01</td>\n",
       "      <td>-0.003727</td>\n",
       "      <td>0.011730</td>\n",
       "      <td>6.907983e-01</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>8079</th>\n",
       "      <td>51193</td>\n",
       "      <td>ERG_02673</td>\n",
       "      <td>XVI</td>\n",
       "      <td>497884</td>\n",
       "      <td>ERG10</td>\n",
       "      <td>upstream_gene_variant</td>\n",
       "      <td>C</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>212</td>\n",
       "      <td>8215</td>\n",
       "      <td>LOV</td>\n",
       "      <td>0.000432</td>\n",
       "      <td>0.001576</td>\n",
       "      <td>0.273875</td>\n",
       "      <td>7.841814e-01</td>\n",
       "      <td>-0.002657</td>\n",
       "      <td>0.003521</td>\n",
       "      <td>9.458009e-01</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>8080</th>\n",
       "      <td>51193</td>\n",
       "      <td>ERG_02673</td>\n",
       "      <td>XVI</td>\n",
       "      <td>497884</td>\n",
       "      <td>ERG10</td>\n",
       "      <td>upstream_gene_variant</td>\n",
       "      <td>C</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>212</td>\n",
       "      <td>8216</td>\n",
       "      <td>NACL</td>\n",
       "      <td>0.002240</td>\n",
       "      <td>0.001219</td>\n",
       "      <td>1.838037</td>\n",
       "      <td>6.606002e-02</td>\n",
       "      <td>-0.000149</td>\n",
       "      <td>0.004629</td>\n",
       "      <td>3.463423e-01</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>8081</th>\n",
       "      <td>51193</td>\n",
       "      <td>ERG_02673</td>\n",
       "      <td>XVI</td>\n",
       "      <td>497884</td>\n",
       "      <td>ERG10</td>\n",
       "      <td>upstream_gene_variant</td>\n",
       "      <td>C</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>212</td>\n",
       "      <td>8217</td>\n",
       "      <td>SC</td>\n",
       "      <td>0.000201</td>\n",
       "      <td>0.001173</td>\n",
       "      <td>0.171019</td>\n",
       "      <td>8.642093e-01</td>\n",
       "      <td>-0.002098</td>\n",
       "      <td>0.002499</td>\n",
       "      <td>9.715505e-01</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>8082</th>\n",
       "      <td>51193</td>\n",
       "      <td>ERG_02673</td>\n",
       "      <td>XVI</td>\n",
       "      <td>497884</td>\n",
       "      <td>ERG10</td>\n",
       "      <td>upstream_gene_variant</td>\n",
       "      <td>C</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>212</td>\n",
       "      <td>8218</td>\n",
       "      <td>TBF</td>\n",
       "      <td>0.002131</td>\n",
       "      <td>0.001098</td>\n",
       "      <td>1.940765</td>\n",
       "      <td>5.228972e-02</td>\n",
       "      <td>-0.000021</td>\n",
       "      <td>0.004283</td>\n",
       "      <td>3.074033e-01</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "      Unnamed: 0_x     var_id chrom  SNP_chr_pos   Gene  \\\n",
       "8071         51186  ERG_02672   XVI       497884  ERG10   \n",
       "8072         51186  ERG_02672   XVI       497884  ERG10   \n",
       "8073         51186  ERG_02672   XVI       497884  ERG10   \n",
       "8074         51186  ERG_02672   XVI       497884  ERG10   \n",
       "8075         51186  ERG_02672   XVI       497884  ERG10   \n",
       "8076         51186  ERG_02672   XVI       497884  ERG10   \n",
       "8077         51193  ERG_02673   XVI       497884  ERG10   \n",
       "8078         51193  ERG_02673   XVI       497884  ERG10   \n",
       "8079         51193  ERG_02673   XVI       497884  ERG10   \n",
       "8080         51193  ERG_02673   XVI       497884  ERG10   \n",
       "8081         51193  ERG_02673   XVI       497884  ERG10   \n",
       "8082         51193  ERG_02673   XVI       497884  ERG10   \n",
       "\n",
       "                Consequence ALT CDS_position Protein_position Amino_acids  \\\n",
       "8071  upstream_gene_variant   A            -                -           -   \n",
       "8072  upstream_gene_variant   A            -                -           -   \n",
       "8073  upstream_gene_variant   A            -                -           -   \n",
       "8074  upstream_gene_variant   A            -                -           -   \n",
       "8075  upstream_gene_variant   A            -                -           -   \n",
       "8076  upstream_gene_variant   A            -                -           -   \n",
       "8077  upstream_gene_variant   C            -                -           -   \n",
       "8078  upstream_gene_variant   C            -                -           -   \n",
       "8079  upstream_gene_variant   C            -                -           -   \n",
       "8080  upstream_gene_variant   C            -                -           -   \n",
       "8081  upstream_gene_variant   C            -                -           -   \n",
       "8082  upstream_gene_variant   C            -                -           -   \n",
       "\n",
       "      DISTANCE  Unnamed: 0_y condition      coef   std err         t  \\\n",
       "8071       212          8207      CAFF  0.004485  0.000500  8.968342   \n",
       "8072       212          8208      COCL  0.000469  0.001735  0.270599   \n",
       "8073       212          8209       LOV -0.003719  0.000683 -5.442711   \n",
       "8074       212          8210      NACL -0.002156  0.000471 -4.577842   \n",
       "8075       212          8211        SC -0.000804  0.000478 -1.680440   \n",
       "8076       212          8212       TBF -0.000280  0.000360 -0.779146   \n",
       "8077       212          8213      CAFF -0.001096  0.001223 -0.896001   \n",
       "8078       212          8214      COCL  0.004001  0.003943  1.014748   \n",
       "8079       212          8215       LOV  0.000432  0.001576  0.273875   \n",
       "8080       212          8216      NACL  0.002240  0.001219  1.838037   \n",
       "8081       212          8217        SC  0.000201  0.001173  0.171019   \n",
       "8082       212          8218       TBF  0.002131  0.001098  1.940765   \n",
       "\n",
       "              pval  Conf. Int. Low  Conf. Int. Upp.          padj  \n",
       "8071  3.061659e-19        0.003505         0.005465  1.395153e-17  \n",
       "8072  7.866999e-01       -0.002930         0.003869  9.477797e-01  \n",
       "8073  5.260164e-08       -0.005058        -0.002380  1.742454e-06  \n",
       "8074  4.703769e-06       -0.003080        -0.001233  1.323691e-04  \n",
       "8075  9.287491e-02       -0.001741         0.000134  4.096807e-01  \n",
       "8076  4.358954e-01       -0.000986         0.000425  7.864612e-01  \n",
       "8077  3.702543e-01       -0.003492         0.001301  7.423242e-01  \n",
       "8078  3.102285e-01       -0.003727         0.011730  6.907983e-01  \n",
       "8079  7.841814e-01       -0.002657         0.003521  9.458009e-01  \n",
       "8080  6.606002e-02       -0.000149         0.004629  3.463423e-01  \n",
       "8081  8.642093e-01       -0.002098         0.002499  9.715505e-01  \n",
       "8082  5.228972e-02       -0.000021         0.004283  3.074033e-01  "
      ]
     },
     "execution_count": 29,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "#HMG1 site with two upstream variants, one with a significant effects in \n",
    "# lovastatin, sodium chloride, and caffeine\n",
    "chrom = 'XVI'\n",
    "pos =497884\n",
    "\n",
    "annotated_fits[(annotated_fits['chrom']==chrom)&(annotated_fits['SNP_chr_pos']==pos)]"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### Downstream Hits within sites w/ multiple edits"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 30,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Unnamed: 0_x</th>\n",
       "      <th>var_id</th>\n",
       "      <th>chrom</th>\n",
       "      <th>SNP_chr_pos</th>\n",
       "      <th>Gene</th>\n",
       "      <th>Consequence</th>\n",
       "      <th>ALT</th>\n",
       "      <th>CDS_position</th>\n",
       "      <th>Protein_position</th>\n",
       "      <th>Amino_acids</th>\n",
       "      <th>DISTANCE</th>\n",
       "      <th>Unnamed: 0_y</th>\n",
       "      <th>condition</th>\n",
       "      <th>coef</th>\n",
       "      <th>std err</th>\n",
       "      <th>t</th>\n",
       "      <th>pval</th>\n",
       "      <th>Conf. Int. Low</th>\n",
       "      <th>Conf. Int. Upp.</th>\n",
       "      <th>padj</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "Empty DataFrame\n",
       "Columns: [Unnamed: 0_x, var_id, chrom, SNP_chr_pos, Gene, Consequence, ALT, CDS_position, Protein_position, Amino_acids, DISTANCE, Unnamed: 0_y, condition, coef, std err, t, pval, Conf. Int. Low, Conf. Int. Upp., padj]\n",
       "Index: []"
      ]
     },
     "execution_count": 30,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "#displaying all synonymous variants which are hits and target a site with multiple edits \n",
    "annotated_fits[(annotated_fits['var_id'].isin(multi_edits))&(annotated_fits['padj']<.01)&(annotated_fits['Consequence']=='downstream_gene_variant')]"
   ]
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "CRISPEY_GxE",
   "language": "python",
   "name": "crispey_gxe"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.9.7"
  },
  "toc": {
   "base_numbering": 1,
   "nav_menu": {},
   "number_sections": true,
   "sideBar": true,
   "skip_h1_title": false,
   "title_cell": "Table of Contents",
   "title_sidebar": "Contents",
   "toc_cell": false,
   "toc_position": {},
   "toc_section_display": true,
   "toc_window_display": false
  }
 },
 "nbformat": 4,
 "nbformat_minor": 2
}
